// @polar-sh/sdk/models/errors/alreadycanceledsubscription.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlreadyCanceledSubscription$ = exports.AlreadyCanceledSubscription$outboundSchema = exports.AlreadyCanceledSubscription$inboundSchema = exports.AlreadyCanceledSubscriptionType$ = exports.AlreadyCanceledSubscriptionType$outboundSchema = exports.AlreadyCanceledSubscriptionType$inboundSchema = exports.AlreadyCanceledSubscription = exports.AlreadyCanceledSubscriptionType = void 0;
const z = __importStar(require("zod"));
exports.AlreadyCanceledSubscriptionType = {
    AlreadyCanceledSubscription: "AlreadyCanceledSubscription",
};
class AlreadyCanceledSubscription extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.type = err.type;
        this.detail = err.detail;
        this.name = "AlreadyCanceledSubscription";
    }
}
exports.AlreadyCanceledSubscription = AlreadyCanceledSubscription;
/** @internal */
exports.AlreadyCanceledSubscriptionType$inboundSchema = z.nativeEnum(exports.AlreadyCanceledSubscriptionType);
/** @internal */
exports.AlreadyCanceledSubscriptionType$outboundSchema = exports.AlreadyCanceledSubscriptionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AlreadyCanceledSubscriptionType$;
(function (AlreadyCanceledSubscriptionType$) {
    /** @deprecated use `AlreadyCanceledSubscriptionType$inboundSchema` instead. */
    AlreadyCanceledSubscriptionType$.inboundSchema = exports.AlreadyCanceledSubscriptionType$inboundSchema;
    /** @deprecated use `AlreadyCanceledSubscriptionType$outboundSchema` instead. */
    AlreadyCanceledSubscriptionType$.outboundSchema = exports.AlreadyCanceledSubscriptionType$outboundSchema;
})(AlreadyCanceledSubscriptionType$ || (exports.AlreadyCanceledSubscriptionType$ = AlreadyCanceledSubscriptionType$ = {}));
/** @internal */
exports.AlreadyCanceledSubscription$inboundSchema = z.object({
    type: z.literal("AlreadyCanceledSubscription"),
    detail: z.string(),
})
    .transform((v) => {
    return new AlreadyCanceledSubscription(v);
});
/** @internal */
exports.AlreadyCanceledSubscription$outboundSchema = z.instanceof(AlreadyCanceledSubscription)
    .transform(v => v.data$)
    .pipe(z.object({
    type: z.literal("AlreadyCanceledSubscription").default("AlreadyCanceledSubscription"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AlreadyCanceledSubscription$;
(function (AlreadyCanceledSubscription$) {
    /** @deprecated use `AlreadyCanceledSubscription$inboundSchema` instead. */
    AlreadyCanceledSubscription$.inboundSchema = exports.AlreadyCanceledSubscription$inboundSchema;
    /** @deprecated use `AlreadyCanceledSubscription$outboundSchema` instead. */
    AlreadyCanceledSubscription$.outboundSchema = exports.AlreadyCanceledSubscription$outboundSchema;
})(AlreadyCanceledSubscription$ || (exports.AlreadyCanceledSubscription$ = AlreadyCanceledSubscription$ = {}));
