// @polar-sh/sdk/models/errors/filenotfound.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileNotFound$ = exports.FileNotFound$outboundSchema = exports.FileNotFound$inboundSchema = exports.FileNotFoundType$ = exports.FileNotFoundType$outboundSchema = exports.FileNotFoundType$inboundSchema = exports.FileNotFound = exports.FileNotFoundType = void 0;
const z = __importStar(require("zod"));
exports.FileNotFoundType = {
    FileNotFound: "FileNotFound",
};
class FileNotFound extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.type = err.type;
        this.detail = err.detail;
        this.name = "FileNotFound";
    }
}
exports.FileNotFound = FileNotFound;
/** @internal */
exports.FileNotFoundType$inboundSchema = z.nativeEnum(exports.FileNotFoundType);
/** @internal */
exports.FileNotFoundType$outboundSchema = exports.FileNotFoundType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileNotFoundType$;
(function (FileNotFoundType$) {
    /** @deprecated use `FileNotFoundType$inboundSchema` instead. */
    FileNotFoundType$.inboundSchema = exports.FileNotFoundType$inboundSchema;
    /** @deprecated use `FileNotFoundType$outboundSchema` instead. */
    FileNotFoundType$.outboundSchema = exports.FileNotFoundType$outboundSchema;
})(FileNotFoundType$ || (exports.FileNotFoundType$ = FileNotFoundType$ = {}));
/** @internal */
exports.FileNotFound$inboundSchema = z.object({
    type: z.literal("FileNotFound"),
    detail: z.string(),
})
    .transform((v) => {
    return new FileNotFound(v);
});
/** @internal */
exports.FileNotFound$outboundSchema = z.instanceof(FileNotFound)
    .transform(v => v.data$)
    .pipe(z.object({
    type: z.literal("FileNotFound").default("FileNotFound"),
    detail: z.string(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileNotFound$;
(function (FileNotFound$) {
    /** @deprecated use `FileNotFound$inboundSchema` instead. */
    FileNotFound$.inboundSchema = exports.FileNotFound$inboundSchema;
    /** @deprecated use `FileNotFound$outboundSchema` instead. */
    FileNotFound$.outboundSchema = exports.FileNotFound$outboundSchema;
})(FileNotFound$ || (exports.FileNotFound$ = FileNotFound$ = {}));
