// @polar-sh/sdk/sdk/articles.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Articles = void 0;
const articlesCreate_js_1 = require("../funcs/articlesCreate.js");
const articlesDelete_js_1 = require("../funcs/articlesDelete.js");
const articlesExport_js_1 = require("../funcs/articlesExport.js");
const articlesGet_js_1 = require("../funcs/articlesGet.js");
const articlesList_js_1 = require("../funcs/articlesList.js");
const articlesPreview_js_1 = require("../funcs/articlesPreview.js");
const articlesReceivers_js_1 = require("../funcs/articlesReceivers.js");
const articlesSend_js_1 = require("../funcs/articlesSend.js");
const articlesUpdate_js_1 = require("../funcs/articlesUpdate.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const operations_js_1 = require("../types/operations.js");
class Articles extends sdks_js_1.ClientSDK {
    /**
     * List Articles
     *
     * @remarks
     * List articles.
     */
    async list(request, options) {
        return (0, operations_js_1.unwrapResultIterator)((0, articlesList_js_1.articlesList)(this, request, options));
    }
    /**
     * Create Article
     *
     * @remarks
     * Create an article.
     */
    async create(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesCreate_js_1.articlesCreate)(this, request, options));
    }
    /**
     * Export Articles
     *
     * @remarks
     * Export organization articles.
     */
    async export(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesExport_js_1.articlesExport)(this, request, options));
    }
    /**
     * Get Article
     *
     * @remarks
     * Get an article by ID.
     */
    async get(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesGet_js_1.articlesGet)(this, request, options));
    }
    /**
     * Update Article
     *
     * @remarks
     * Update an article.
     */
    async update(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesUpdate_js_1.articlesUpdate)(this, request, options));
    }
    /**
     * Delete Article
     *
     * @remarks
     * Delete an article.
     */
    async delete(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesDelete_js_1.articlesDelete)(this, request, options));
    }
    /**
     * Get Article Receivers Count
     *
     * @remarks
     * Get number of potential receivers for an article.
     */
    async receivers(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesReceivers_js_1.articlesReceivers)(this, request, options));
    }
    /**
     * Send Article Preview
     *
     * @remarks
     * Send an article preview by email.
     */
    async preview(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesPreview_js_1.articlesPreview)(this, request, options));
    }
    /**
     * Send Article
     *
     * @remarks
     * Send an article by email to all subscribers.
     */
    async send(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, articlesSend_js_1.articlesSend)(this, request, options));
    }
}
exports.Articles = Articles;
