// @polar-sh/sdk/sdk/polaradvertisements.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolarAdvertisements = void 0;
const usersAdvertisementsCreate_js_1 = require("../funcs/usersAdvertisementsCreate.js");
const usersAdvertisementsDelete_js_1 = require("../funcs/usersAdvertisementsDelete.js");
const usersAdvertisementsEnable_js_1 = require("../funcs/usersAdvertisementsEnable.js");
const usersAdvertisementsGet_js_1 = require("../funcs/usersAdvertisementsGet.js");
const usersAdvertisementsList_js_1 = require("../funcs/usersAdvertisementsList.js");
const usersAdvertisementsUpdate_js_1 = require("../funcs/usersAdvertisementsUpdate.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const operations_js_1 = require("../types/operations.js");
class PolarAdvertisements extends sdks_js_1.ClientSDK {
    /**
     * List Advertisements
     *
     * @remarks
     * List advertisement campaigns.
     */
    async list(request, options) {
        return (0, operations_js_1.unwrapResultIterator)((0, usersAdvertisementsList_js_1.usersAdvertisementsList)(this, request, options));
    }
    /**
     * Create Advertisement
     *
     * @remarks
     * Create an advertisement campaign.
     */
    async create(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, usersAdvertisementsCreate_js_1.usersAdvertisementsCreate)(this, request, options));
    }
    /**
     * Get Advertisement
     *
     * @remarks
     * Get an advertisement campaign by ID.
     */
    async get(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, usersAdvertisementsGet_js_1.usersAdvertisementsGet)(this, request, options));
    }
    /**
     * Update Advertisement
     *
     * @remarks
     * Update an advertisement campaign.
     */
    async update(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, usersAdvertisementsUpdate_js_1.usersAdvertisementsUpdate)(this, request, options));
    }
    /**
     * Delete Advertisement
     *
     * @remarks
     * Delete an advertisement campaign.
     *
     * It'll be automatically disabled on all granted benefits.
     */
    async delete(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, usersAdvertisementsDelete_js_1.usersAdvertisementsDelete)(this, request, options));
    }
    /**
     * Enable Advertisement
     *
     * @remarks
     * Enable an advertisement campaign on a granted benefit.
     */
    async enable(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, usersAdvertisementsEnable_js_1.usersAdvertisementsEnable)(this, request, options));
    }
}
exports.PolarAdvertisements = PolarAdvertisements;
