// @polar-sh/sdk/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Polar = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const advertisements_js_1 = require("./advertisements.js");
const articles_js_1 = require("./articles.js");
const benefits_js_1 = require("./benefits.js");
const checkoutlinks_js_1 = require("./checkoutlinks.js");
const checkouts_js_1 = require("./checkouts.js");
const externalorganizations_js_1 = require("./externalorganizations.js");
const files_js_1 = require("./files.js");
const licensekeys_js_1 = require("./licensekeys.js");
const metrics_js_1 = require("./metrics.js");
const oauth2_js_1 = require("./oauth2.js");
const orders_js_1 = require("./orders.js");
const organizations_js_1 = require("./organizations.js");
const products_js_1 = require("./products.js");
const repositories_js_1 = require("./repositories.js");
const subscriptions_js_1 = require("./subscriptions.js");
const users_js_1 = require("./users.js");
class Polar extends sdks_js_1.ClientSDK {
    get users() {
        var _a;
        return ((_a = this._users) !== null && _a !== void 0 ? _a : (this._users = new users_js_1.Users(this._options)));
    }
    get externalOrganizations() {
        var _a;
        return ((_a = this._externalOrganizations) !== null && _a !== void 0 ? _a : (this._externalOrganizations = new externalorganizations_js_1.ExternalOrganizations(this._options)));
    }
    get repositories() {
        var _a;
        return ((_a = this._repositories) !== null && _a !== void 0 ? _a : (this._repositories = new repositories_js_1.Repositories(this._options)));
    }
    get organizations() {
        var _a;
        return ((_a = this._organizations) !== null && _a !== void 0 ? _a : (this._organizations = new organizations_js_1.Organizations(this._options)));
    }
    get subscriptions() {
        var _a;
        return ((_a = this._subscriptions) !== null && _a !== void 0 ? _a : (this._subscriptions = new subscriptions_js_1.Subscriptions(this._options)));
    }
    get articles() {
        var _a;
        return ((_a = this._articles) !== null && _a !== void 0 ? _a : (this._articles = new articles_js_1.Articles(this._options)));
    }
    get advertisements() {
        var _a;
        return ((_a = this._advertisements) !== null && _a !== void 0 ? _a : (this._advertisements = new advertisements_js_1.Advertisements(this._options)));
    }
    get oauth2() {
        var _a;
        return ((_a = this._oauth2) !== null && _a !== void 0 ? _a : (this._oauth2 = new oauth2_js_1.Oauth2(this._options)));
    }
    get benefits() {
        var _a;
        return ((_a = this._benefits) !== null && _a !== void 0 ? _a : (this._benefits = new benefits_js_1.Benefits(this._options)));
    }
    get products() {
        var _a;
        return ((_a = this._products) !== null && _a !== void 0 ? _a : (this._products = new products_js_1.Products(this._options)));
    }
    get orders() {
        var _a;
        return ((_a = this._orders) !== null && _a !== void 0 ? _a : (this._orders = new orders_js_1.Orders(this._options)));
    }
    get checkouts() {
        var _a;
        return ((_a = this._checkouts) !== null && _a !== void 0 ? _a : (this._checkouts = new checkouts_js_1.Checkouts(this._options)));
    }
    get files() {
        var _a;
        return ((_a = this._files) !== null && _a !== void 0 ? _a : (this._files = new files_js_1.Files(this._options)));
    }
    get metrics() {
        var _a;
        return ((_a = this._metrics) !== null && _a !== void 0 ? _a : (this._metrics = new metrics_js_1.Metrics(this._options)));
    }
    get licenseKeys() {
        var _a;
        return ((_a = this._licenseKeys) !== null && _a !== void 0 ? _a : (this._licenseKeys = new licensekeys_js_1.LicenseKeys(this._options)));
    }
    get checkoutLinks() {
        var _a;
        return ((_a = this._checkoutLinks) !== null && _a !== void 0 ? _a : (this._checkoutLinks = new checkoutlinks_js_1.CheckoutLinks(this._options)));
    }
}
exports.Polar = Polar;
