// @polar-sh/sdk/sdk/users.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Users = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const downloadables_js_1 = require("./downloadables.js");
const polaradvertisements_js_1 = require("./polaradvertisements.js");
const polarbenefits_js_1 = require("./polarbenefits.js");
const polarlicensekeys_js_1 = require("./polarlicensekeys.js");
const polarorders_js_1 = require("./polarorders.js");
const polarsubscriptions_js_1 = require("./polarsubscriptions.js");
class Users extends sdks_js_1.ClientSDK {
    get benefits() {
        var _a;
        return ((_a = this._benefits) !== null && _a !== void 0 ? _a : (this._benefits = new polarbenefits_js_1.PolarBenefits(this._options)));
    }
    get orders() {
        var _a;
        return ((_a = this._orders) !== null && _a !== void 0 ? _a : (this._orders = new polarorders_js_1.PolarOrders(this._options)));
    }
    get subscriptions() {
        var _a;
        return ((_a = this._subscriptions) !== null && _a !== void 0 ? _a : (this._subscriptions = new polarsubscriptions_js_1.PolarSubscriptions(this._options)));
    }
    get advertisements() {
        var _a;
        return ((_a = this._advertisements) !== null && _a !== void 0 ? _a : (this._advertisements = new polaradvertisements_js_1.PolarAdvertisements(this._options)));
    }
    get downloadables() {
        var _a;
        return ((_a = this._downloadables) !== null && _a !== void 0 ? _a : (this._downloadables = new downloadables_js_1.Downloadables(this._options)));
    }
    get licenseKeys() {
        var _a;
        return ((_a = this._licenseKeys) !== null && _a !== void 0 ? _a : (this._licenseKeys = new polarlicensekeys_js_1.PolarLicenseKeys(this._options)));
    }
}
exports.Users = Users;
