// @polar-sh/sdk/models/components/advertisementcampaignlistresource.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvertisementCampaignListResource$ = exports.AdvertisementCampaignListResource$outboundSchema = exports.AdvertisementCampaignListResource$inboundSchema = void 0;
const z = __importStar(require("zod"));
const advertisementcampaign_js_1 = require("./advertisementcampaign.js");
const pagination_js_1 = require("./pagination.js");
/** @internal */
exports.AdvertisementCampaignListResource$inboundSchema = z.object({
    items: z.array(advertisementcampaign_js_1.AdvertisementCampaign$inboundSchema),
    pagination: pagination_js_1.Pagination$inboundSchema,
    dimensions: z.array(z.number().int()),
});
/** @internal */
exports.AdvertisementCampaignListResource$outboundSchema = z.object({
    items: z.array(advertisementcampaign_js_1.AdvertisementCampaign$outboundSchema),
    pagination: pagination_js_1.Pagination$outboundSchema,
    dimensions: z.array(z.number().int()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AdvertisementCampaignListResource$;
(function (AdvertisementCampaignListResource$) {
    /** @deprecated use `AdvertisementCampaignListResource$inboundSchema` instead. */
    AdvertisementCampaignListResource$.inboundSchema = exports.AdvertisementCampaignListResource$inboundSchema;
    /** @deprecated use `AdvertisementCampaignListResource$outboundSchema` instead. */
    AdvertisementCampaignListResource$.outboundSchema = exports.AdvertisementCampaignListResource$outboundSchema;
})(AdvertisementCampaignListResource$ || (exports.AdvertisementCampaignListResource$ = AdvertisementCampaignListResource$ = {}));
