// @polar-sh/sdk/models/components/advertisementsortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvertisementSortProperty$ = exports.AdvertisementSortProperty$outboundSchema = exports.AdvertisementSortProperty$inboundSchema = exports.AdvertisementSortProperty = void 0;
const z = __importStar(require("zod"));
exports.AdvertisementSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    GrantedAt: "granted_at",
    MinusGrantedAt: "-granted_at",
    Views: "views",
    MinusViews: "-views",
    Clicks: "clicks",
    MinusClicks: "-clicks",
};
/** @internal */
exports.AdvertisementSortProperty$inboundSchema = z.nativeEnum(exports.AdvertisementSortProperty);
/** @internal */
exports.AdvertisementSortProperty$outboundSchema = exports.AdvertisementSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AdvertisementSortProperty$;
(function (AdvertisementSortProperty$) {
    /** @deprecated use `AdvertisementSortProperty$inboundSchema` instead. */
    AdvertisementSortProperty$.inboundSchema = exports.AdvertisementSortProperty$inboundSchema;
    /** @deprecated use `AdvertisementSortProperty$outboundSchema` instead. */
    AdvertisementSortProperty$.outboundSchema = exports.AdvertisementSortProperty$outboundSchema;
})(AdvertisementSortProperty$ || (exports.AdvertisementSortProperty$ = AdvertisementSortProperty$ = {}));
