// @polar-sh/sdk/models/components/articlereceivers.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArticleReceivers$ = exports.ArticleReceivers$outboundSchema = exports.ArticleReceivers$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.ArticleReceivers$inboundSchema = z.object({
    free_subscribers: z.number().int(),
    premium_subscribers: z.number().int(),
    organization_members: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "free_subscribers": "freeSubscribers",
        "premium_subscribers": "premiumSubscribers",
        "organization_members": "organizationMembers",
    });
});
/** @internal */
exports.ArticleReceivers$outboundSchema = z.object({
    freeSubscribers: z.number().int(),
    premiumSubscribers: z.number().int(),
    organizationMembers: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        freeSubscribers: "free_subscribers",
        premiumSubscribers: "premium_subscribers",
        organizationMembers: "organization_members",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArticleReceivers$;
(function (ArticleReceivers$) {
    /** @deprecated use `ArticleReceivers$inboundSchema` instead. */
    ArticleReceivers$.inboundSchema = exports.ArticleReceivers$inboundSchema;
    /** @deprecated use `ArticleReceivers$outboundSchema` instead. */
    ArticleReceivers$.outboundSchema = exports.ArticleReceivers$outboundSchema;
})(ArticleReceivers$ || (exports.ArticleReceivers$ = ArticleReceivers$ = {}));
