// @polar-sh/sdk/models/components/articlevisibility.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArticleVisibility$ = exports.ArticleVisibility$outboundSchema = exports.ArticleVisibility$inboundSchema = exports.ArticleVisibility = void 0;
const z = __importStar(require("zod"));
exports.ArticleVisibility = {
    Public: "public",
    Hidden: "hidden",
    Private: "private",
};
/** @internal */
exports.ArticleVisibility$inboundSchema = z.nativeEnum(exports.ArticleVisibility);
/** @internal */
exports.ArticleVisibility$outboundSchema = exports.ArticleVisibility$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArticleVisibility$;
(function (ArticleVisibility$) {
    /** @deprecated use `ArticleVisibility$inboundSchema` instead. */
    ArticleVisibility$.inboundSchema = exports.ArticleVisibility$inboundSchema;
    /** @deprecated use `ArticleVisibility$outboundSchema` instead. */
    ArticleVisibility$.outboundSchema = exports.ArticleVisibility$outboundSchema;
})(ArticleVisibility$ || (exports.ArticleVisibility$ = ArticleVisibility$ = {}));
