// @polar-sh/sdk/models/components/article.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Article$ = exports.Article$outboundSchema = exports.Article$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const articlevisibility_js_1 = require("./articlevisibility.js");
const bylineprofile_js_1 = require("./bylineprofile.js");
const organization_js_1 = require("./organization.js");
/** @internal */
exports.Article$inboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    title: z.string(),
    body: z.string(),
    byline: bylineprofile_js_1.BylineProfile$inboundSchema,
    visibility: articlevisibility_js_1.ArticleVisibility$inboundSchema,
    user_id: z.nullable(z.string()),
    organization_id: z.string(),
    organization: organization_js_1.Organization$inboundSchema,
    published_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    paid_subscribers_only: z.nullable(z.boolean()),
    paid_subscribers_only_ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    is_preview: z.boolean(),
    is_pinned: z.boolean(),
    notify_subscribers: z.nullable(z.boolean()),
    notifications_sent_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    email_sent_to_count: z.nullable(z.number().int()),
    og_image_url: z.nullable(z.string()),
    og_description: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "user_id": "userId",
        "organization_id": "organizationId",
        "published_at": "publishedAt",
        "paid_subscribers_only": "paidSubscribersOnly",
        "paid_subscribers_only_ends_at": "paidSubscribersOnlyEndsAt",
        "is_preview": "isPreview",
        "is_pinned": "isPinned",
        "notify_subscribers": "notifySubscribers",
        "notifications_sent_at": "notificationsSentAt",
        "email_sent_to_count": "emailSentToCount",
        "og_image_url": "ogImageUrl",
        "og_description": "ogDescription",
    });
});
/** @internal */
exports.Article$outboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    title: z.string(),
    body: z.string(),
    byline: bylineprofile_js_1.BylineProfile$outboundSchema,
    visibility: articlevisibility_js_1.ArticleVisibility$outboundSchema,
    userId: z.nullable(z.string()),
    organizationId: z.string(),
    organization: organization_js_1.Organization$outboundSchema,
    publishedAt: z.nullable(z.date().transform(v => v.toISOString())),
    paidSubscribersOnly: z.nullable(z.boolean()),
    paidSubscribersOnlyEndsAt: z.nullable(z.date().transform(v => v.toISOString())),
    isPreview: z.boolean(),
    isPinned: z.boolean(),
    notifySubscribers: z.nullable(z.boolean()),
    notificationsSentAt: z.nullable(z.date().transform(v => v.toISOString())),
    emailSentToCount: z.nullable(z.number().int()),
    ogImageUrl: z.nullable(z.string()),
    ogDescription: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userId: "user_id",
        organizationId: "organization_id",
        publishedAt: "published_at",
        paidSubscribersOnly: "paid_subscribers_only",
        paidSubscribersOnlyEndsAt: "paid_subscribers_only_ends_at",
        isPreview: "is_preview",
        isPinned: "is_pinned",
        notifySubscribers: "notify_subscribers",
        notificationsSentAt: "notifications_sent_at",
        emailSentToCount: "email_sent_to_count",
        ogImageUrl: "og_image_url",
        ogDescription: "og_description",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Article$;
(function (Article$) {
    /** @deprecated use `Article$inboundSchema` instead. */
    Article$.inboundSchema = exports.Article$inboundSchema;
    /** @deprecated use `Article$outboundSchema` instead. */
    Article$.outboundSchema = exports.Article$outboundSchema;
})(Article$ || (exports.Article$ = Article$ = {}));
