// @polar-sh/sdk/models/components/assignee.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Assignee$ = exports.Assignee$outboundSchema = exports.Assignee$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.Assignee$inboundSchema = z.object({
    id: z.number().int(),
    login: z.string(),
    html_url: z.string(),
    avatar_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "html_url": "htmlUrl",
        "avatar_url": "avatarUrl",
    });
});
/** @internal */
exports.Assignee$outboundSchema = z.object({
    id: z.number().int(),
    login: z.string(),
    htmlUrl: z.string(),
    avatarUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        htmlUrl: "html_url",
        avatarUrl: "avatar_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Assignee$;
(function (Assignee$) {
    /** @deprecated use `Assignee$inboundSchema` instead. */
    Assignee$.inboundSchema = exports.Assignee$inboundSchema;
    /** @deprecated use `Assignee$outboundSchema` instead. */
    Assignee$.outboundSchema = exports.Assignee$outboundSchema;
})(Assignee$ || (exports.Assignee$ = Assignee$ = {}));
