// @polar-sh/sdk/models/components/benefitadsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitAdsUpdate$ = exports.BenefitAdsUpdate$outboundSchema = exports.BenefitAdsUpdate$inboundSchema = exports.BenefitAdsUpdateType$ = exports.BenefitAdsUpdateType$outboundSchema = exports.BenefitAdsUpdateType$inboundSchema = exports.BenefitAdsUpdateType = void 0;
const z = __importStar(require("zod"));
const benefitadsproperties_js_1 = require("./benefitadsproperties.js");
exports.BenefitAdsUpdateType = {
    Ads: "ads",
};
/** @internal */
exports.BenefitAdsUpdateType$inboundSchema = z.nativeEnum(exports.BenefitAdsUpdateType);
/** @internal */
exports.BenefitAdsUpdateType$outboundSchema = exports.BenefitAdsUpdateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitAdsUpdateType$;
(function (BenefitAdsUpdateType$) {
    /** @deprecated use `BenefitAdsUpdateType$inboundSchema` instead. */
    BenefitAdsUpdateType$.inboundSchema = exports.BenefitAdsUpdateType$inboundSchema;
    /** @deprecated use `BenefitAdsUpdateType$outboundSchema` instead. */
    BenefitAdsUpdateType$.outboundSchema = exports.BenefitAdsUpdateType$outboundSchema;
})(BenefitAdsUpdateType$ || (exports.BenefitAdsUpdateType$ = BenefitAdsUpdateType$ = {}));
/** @internal */
exports.BenefitAdsUpdate$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("ads").optional(),
    properties: z.nullable(benefitadsproperties_js_1.BenefitAdsProperties$inboundSchema).optional(),
});
/** @internal */
exports.BenefitAdsUpdate$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("ads").default("ads"),
    properties: z.nullable(benefitadsproperties_js_1.BenefitAdsProperties$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitAdsUpdate$;
(function (BenefitAdsUpdate$) {
    /** @deprecated use `BenefitAdsUpdate$inboundSchema` instead. */
    BenefitAdsUpdate$.inboundSchema = exports.BenefitAdsUpdate$inboundSchema;
    /** @deprecated use `BenefitAdsUpdate$outboundSchema` instead. */
    BenefitAdsUpdate$.outboundSchema = exports.BenefitAdsUpdate$outboundSchema;
})(BenefitAdsUpdate$ || (exports.BenefitAdsUpdate$ = BenefitAdsUpdate$ = {}));
