// @polar-sh/sdk/models/components/benefitlicensekeysupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeysUpdate$ = exports.BenefitLicenseKeysUpdate$outboundSchema = exports.BenefitLicenseKeysUpdate$inboundSchema = exports.BenefitLicenseKeysUpdateType$ = exports.BenefitLicenseKeysUpdateType$outboundSchema = exports.BenefitLicenseKeysUpdateType$inboundSchema = exports.BenefitLicenseKeysUpdateType = void 0;
const z = __importStar(require("zod"));
const benefitlicensekeyscreateproperties_js_1 = require("./benefitlicensekeyscreateproperties.js");
exports.BenefitLicenseKeysUpdateType = {
    LicenseKeys: "license_keys",
};
/** @internal */
exports.BenefitLicenseKeysUpdateType$inboundSchema = z.nativeEnum(exports.BenefitLicenseKeysUpdateType);
/** @internal */
exports.BenefitLicenseKeysUpdateType$outboundSchema = exports.BenefitLicenseKeysUpdateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysUpdateType$;
(function (BenefitLicenseKeysUpdateType$) {
    /** @deprecated use `BenefitLicenseKeysUpdateType$inboundSchema` instead. */
    BenefitLicenseKeysUpdateType$.inboundSchema = exports.BenefitLicenseKeysUpdateType$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysUpdateType$outboundSchema` instead. */
    BenefitLicenseKeysUpdateType$.outboundSchema = exports.BenefitLicenseKeysUpdateType$outboundSchema;
})(BenefitLicenseKeysUpdateType$ || (exports.BenefitLicenseKeysUpdateType$ = BenefitLicenseKeysUpdateType$ = {}));
/** @internal */
exports.BenefitLicenseKeysUpdate$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("license_keys").optional(),
    properties: z.nullable(benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitLicenseKeysUpdate$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("license_keys").default("license_keys"),
    properties: z.nullable(benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysUpdate$;
(function (BenefitLicenseKeysUpdate$) {
    /** @deprecated use `BenefitLicenseKeysUpdate$inboundSchema` instead. */
    BenefitLicenseKeysUpdate$.inboundSchema = exports.BenefitLicenseKeysUpdate$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysUpdate$outboundSchema` instead. */
    BenefitLicenseKeysUpdate$.outboundSchema = exports.BenefitLicenseKeysUpdate$outboundSchema;
})(BenefitLicenseKeysUpdate$ || (exports.BenefitLicenseKeysUpdate$ = BenefitLicenseKeysUpdate$ = {}));
