// @polar-sh/sdk/models/components/benefitlicensekeyssubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeysSubscriber$ = exports.BenefitLicenseKeysSubscriber$outboundSchema = exports.BenefitLicenseKeysSubscriber$inboundSchema = exports.BenefitLicenseKeysSubscriberType$ = exports.BenefitLicenseKeysSubscriberType$outboundSchema = exports.BenefitLicenseKeysSubscriberType$inboundSchema = exports.BenefitLicenseKeysSubscriberType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitgrantlicensekeys_js_1 = require("./benefitgrantlicensekeys.js");
const benefitlicensekeyssubscriberproperties_js_1 = require("./benefitlicensekeyssubscriberproperties.js");
exports.BenefitLicenseKeysSubscriberType = {
    LicenseKeys: "license_keys",
};
/** @internal */
exports.BenefitLicenseKeysSubscriberType$inboundSchema = z.nativeEnum(exports.BenefitLicenseKeysSubscriberType);
/** @internal */
exports.BenefitLicenseKeysSubscriberType$outboundSchema = exports.BenefitLicenseKeysSubscriberType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysSubscriberType$;
(function (BenefitLicenseKeysSubscriberType$) {
    /** @deprecated use `BenefitLicenseKeysSubscriberType$inboundSchema` instead. */
    BenefitLicenseKeysSubscriberType$.inboundSchema = exports.BenefitLicenseKeysSubscriberType$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysSubscriberType$outboundSchema` instead. */
    BenefitLicenseKeysSubscriberType$.outboundSchema = exports.BenefitLicenseKeysSubscriberType$outboundSchema;
})(BenefitLicenseKeysSubscriberType$ || (exports.BenefitLicenseKeysSubscriberType$ = BenefitLicenseKeysSubscriberType$ = {}));
/** @internal */
exports.BenefitLicenseKeysSubscriber$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("license_keys").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitlicensekeyssubscriberproperties_js_1.BenefitLicenseKeysSubscriberProperties$inboundSchema,
    grants: z.array(benefitgrantlicensekeys_js_1.BenefitGrantLicenseKeys$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitLicenseKeysSubscriber$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("license_keys").default("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitlicensekeyssubscriberproperties_js_1.BenefitLicenseKeysSubscriberProperties$outboundSchema,
    grants: z.array(benefitgrantlicensekeys_js_1.BenefitGrantLicenseKeys$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysSubscriber$;
(function (BenefitLicenseKeysSubscriber$) {
    /** @deprecated use `BenefitLicenseKeysSubscriber$inboundSchema` instead. */
    BenefitLicenseKeysSubscriber$.inboundSchema = exports.BenefitLicenseKeysSubscriber$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysSubscriber$outboundSchema` instead. */
    BenefitLicenseKeysSubscriber$.outboundSchema = exports.BenefitLicenseKeysSubscriber$outboundSchema;
})(BenefitLicenseKeysSubscriber$ || (exports.BenefitLicenseKeysSubscriber$ = BenefitLicenseKeysSubscriber$ = {}));
