// @polar-sh/sdk/models/components/benefitlicensekeys.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeys$ = exports.BenefitLicenseKeys$outboundSchema = exports.BenefitLicenseKeys$inboundSchema = exports.BenefitLicenseKeysType$ = exports.BenefitLicenseKeysType$outboundSchema = exports.BenefitLicenseKeysType$inboundSchema = exports.BenefitLicenseKeysType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitlicensekeysproperties_js_1 = require("./benefitlicensekeysproperties.js");
exports.BenefitLicenseKeysType = {
    LicenseKeys: "license_keys",
};
/** @internal */
exports.BenefitLicenseKeysType$inboundSchema = z.nativeEnum(exports.BenefitLicenseKeysType);
/** @internal */
exports.BenefitLicenseKeysType$outboundSchema = exports.BenefitLicenseKeysType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysType$;
(function (BenefitLicenseKeysType$) {
    /** @deprecated use `BenefitLicenseKeysType$inboundSchema` instead. */
    BenefitLicenseKeysType$.inboundSchema = exports.BenefitLicenseKeysType$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysType$outboundSchema` instead. */
    BenefitLicenseKeysType$.outboundSchema = exports.BenefitLicenseKeysType$outboundSchema;
})(BenefitLicenseKeysType$ || (exports.BenefitLicenseKeysType$ = BenefitLicenseKeysType$ = {}));
/** @internal */
exports.BenefitLicenseKeys$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("license_keys").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitlicensekeysproperties_js_1.BenefitLicenseKeysProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitLicenseKeys$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("license_keys").default("license_keys"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitlicensekeysproperties_js_1.BenefitLicenseKeysProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeys$;
(function (BenefitLicenseKeys$) {
    /** @deprecated use `BenefitLicenseKeys$inboundSchema` instead. */
    BenefitLicenseKeys$.inboundSchema = exports.BenefitLicenseKeys$inboundSchema;
    /** @deprecated use `BenefitLicenseKeys$outboundSchema` instead. */
    BenefitLicenseKeys$.outboundSchema = exports.BenefitLicenseKeys$outboundSchema;
})(BenefitLicenseKeys$ || (exports.BenefitLicenseKeys$ = BenefitLicenseKeys$ = {}));
