// @polar-sh/sdk/models/components/benefitcustomupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitCustomUpdate$ = exports.BenefitCustomUpdate$outboundSchema = exports.BenefitCustomUpdate$inboundSchema = exports.BenefitCustomUpdateType$ = exports.BenefitCustomUpdateType$outboundSchema = exports.BenefitCustomUpdateType$inboundSchema = exports.BenefitCustomUpdateType = void 0;
const z = __importStar(require("zod"));
const benefitcustomproperties_js_1 = require("./benefitcustomproperties.js");
exports.BenefitCustomUpdateType = {
    Custom: "custom",
};
/** @internal */
exports.BenefitCustomUpdateType$inboundSchema = z.nativeEnum(exports.BenefitCustomUpdateType);
/** @internal */
exports.BenefitCustomUpdateType$outboundSchema = exports.BenefitCustomUpdateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomUpdateType$;
(function (BenefitCustomUpdateType$) {
    /** @deprecated use `BenefitCustomUpdateType$inboundSchema` instead. */
    BenefitCustomUpdateType$.inboundSchema = exports.BenefitCustomUpdateType$inboundSchema;
    /** @deprecated use `BenefitCustomUpdateType$outboundSchema` instead. */
    BenefitCustomUpdateType$.outboundSchema = exports.BenefitCustomUpdateType$outboundSchema;
})(BenefitCustomUpdateType$ || (exports.BenefitCustomUpdateType$ = BenefitCustomUpdateType$ = {}));
/** @internal */
exports.BenefitCustomUpdate$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("custom").optional(),
    properties: z.nullable(benefitcustomproperties_js_1.BenefitCustomProperties$inboundSchema).optional(),
});
/** @internal */
exports.BenefitCustomUpdate$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("custom").default("custom"),
    properties: z.nullable(benefitcustomproperties_js_1.BenefitCustomProperties$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomUpdate$;
(function (BenefitCustomUpdate$) {
    /** @deprecated use `BenefitCustomUpdate$inboundSchema` instead. */
    BenefitCustomUpdate$.inboundSchema = exports.BenefitCustomUpdate$inboundSchema;
    /** @deprecated use `BenefitCustomUpdate$outboundSchema` instead. */
    BenefitCustomUpdate$.outboundSchema = exports.BenefitCustomUpdate$outboundSchema;
})(BenefitCustomUpdate$ || (exports.BenefitCustomUpdate$ = BenefitCustomUpdate$ = {}));
