// @polar-sh/sdk/models/components/benefitdownloadablescreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadablesCreate$ = exports.BenefitDownloadablesCreate$outboundSchema = exports.BenefitDownloadablesCreate$inboundSchema = exports.BenefitDownloadablesCreateType$ = exports.BenefitDownloadablesCreateType$outboundSchema = exports.BenefitDownloadablesCreateType$inboundSchema = exports.BenefitDownloadablesCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitdownloadablescreateproperties_js_1 = require("./benefitdownloadablescreateproperties.js");
exports.BenefitDownloadablesCreateType = {
    Downloadables: "downloadables",
};
/** @internal */
exports.BenefitDownloadablesCreateType$inboundSchema = z.nativeEnum(exports.BenefitDownloadablesCreateType);
/** @internal */
exports.BenefitDownloadablesCreateType$outboundSchema = exports.BenefitDownloadablesCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesCreateType$;
(function (BenefitDownloadablesCreateType$) {
    /** @deprecated use `BenefitDownloadablesCreateType$inboundSchema` instead. */
    BenefitDownloadablesCreateType$.inboundSchema = exports.BenefitDownloadablesCreateType$inboundSchema;
    /** @deprecated use `BenefitDownloadablesCreateType$outboundSchema` instead. */
    BenefitDownloadablesCreateType$.outboundSchema = exports.BenefitDownloadablesCreateType$outboundSchema;
})(BenefitDownloadablesCreateType$ || (exports.BenefitDownloadablesCreateType$ = BenefitDownloadablesCreateType$ = {}));
/** @internal */
exports.BenefitDownloadablesCreate$inboundSchema = z.object({
    type: z.literal("downloadables").optional(),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitdownloadablescreateproperties_js_1.BenefitDownloadablesCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDownloadablesCreate$outboundSchema = z.object({
    type: z.literal("downloadables").default("downloadables"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitdownloadablescreateproperties_js_1.BenefitDownloadablesCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesCreate$;
(function (BenefitDownloadablesCreate$) {
    /** @deprecated use `BenefitDownloadablesCreate$inboundSchema` instead. */
    BenefitDownloadablesCreate$.inboundSchema = exports.BenefitDownloadablesCreate$inboundSchema;
    /** @deprecated use `BenefitDownloadablesCreate$outboundSchema` instead. */
    BenefitDownloadablesCreate$.outboundSchema = exports.BenefitDownloadablesCreate$outboundSchema;
})(BenefitDownloadablesCreate$ || (exports.BenefitDownloadablesCreate$ = BenefitDownloadablesCreate$ = {}));
