// @polar-sh/sdk/models/components/benefitgithubrepositoryproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositoryProperties$ = exports.BenefitGitHubRepositoryProperties$outboundSchema = exports.BenefitGitHubRepositoryProperties$inboundSchema = exports.Permission$ = exports.Permission$outboundSchema = exports.Permission$inboundSchema = exports.Permission = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/**
 * The permission level to grant. Read more about roles and their permissions on [GitHub documentation](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role).
 */
exports.Permission = {
    Pull: "pull",
    Triage: "triage",
    Push: "push",
    Maintain: "maintain",
    Admin: "admin",
};
/** @internal */
exports.Permission$inboundSchema = z
    .nativeEnum(exports.Permission);
/** @internal */
exports.Permission$outboundSchema = exports.Permission$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Permission$;
(function (Permission$) {
    /** @deprecated use `Permission$inboundSchema` instead. */
    Permission$.inboundSchema = exports.Permission$inboundSchema;
    /** @deprecated use `Permission$outboundSchema` instead. */
    Permission$.outboundSchema = exports.Permission$outboundSchema;
})(Permission$ || (exports.Permission$ = Permission$ = {}));
/** @internal */
exports.BenefitGitHubRepositoryProperties$inboundSchema = z.object({
    repository_id: z.nullable(z.string()),
    repository_owner: z.string(),
    repository_name: z.string(),
    permission: exports.Permission$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "repository_id": "repositoryId",
        "repository_owner": "repositoryOwner",
        "repository_name": "repositoryName",
    });
});
/** @internal */
exports.BenefitGitHubRepositoryProperties$outboundSchema = z.object({
    repositoryId: z.nullable(z.string()),
    repositoryOwner: z.string(),
    repositoryName: z.string(),
    permission: exports.Permission$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        repositoryId: "repository_id",
        repositoryOwner: "repository_owner",
        repositoryName: "repository_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryProperties$;
(function (BenefitGitHubRepositoryProperties$) {
    /** @deprecated use `BenefitGitHubRepositoryProperties$inboundSchema` instead. */
    BenefitGitHubRepositoryProperties$.inboundSchema = exports.BenefitGitHubRepositoryProperties$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryProperties$outboundSchema` instead. */
    BenefitGitHubRepositoryProperties$.outboundSchema = exports.BenefitGitHubRepositoryProperties$outboundSchema;
})(BenefitGitHubRepositoryProperties$ || (exports.BenefitGitHubRepositoryProperties$ = BenefitGitHubRepositoryProperties$ = {}));
