// @polar-sh/sdk/models/components/benefitdownloadablessubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadablesSubscriber$ = exports.BenefitDownloadablesSubscriber$outboundSchema = exports.BenefitDownloadablesSubscriber$inboundSchema = exports.BenefitDownloadablesSubscriberType$ = exports.BenefitDownloadablesSubscriberType$outboundSchema = exports.BenefitDownloadablesSubscriberType$inboundSchema = exports.BenefitDownloadablesSubscriberType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitdownloadablessubscriberproperties_js_1 = require("./benefitdownloadablessubscriberproperties.js");
exports.BenefitDownloadablesSubscriberType = {
    Downloadables: "downloadables",
};
/** @internal */
exports.BenefitDownloadablesSubscriberType$inboundSchema = z.nativeEnum(exports.BenefitDownloadablesSubscriberType);
/** @internal */
exports.BenefitDownloadablesSubscriberType$outboundSchema = exports.BenefitDownloadablesSubscriberType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesSubscriberType$;
(function (BenefitDownloadablesSubscriberType$) {
    /** @deprecated use `BenefitDownloadablesSubscriberType$inboundSchema` instead. */
    BenefitDownloadablesSubscriberType$.inboundSchema = exports.BenefitDownloadablesSubscriberType$inboundSchema;
    /** @deprecated use `BenefitDownloadablesSubscriberType$outboundSchema` instead. */
    BenefitDownloadablesSubscriberType$.outboundSchema = exports.BenefitDownloadablesSubscriberType$outboundSchema;
})(BenefitDownloadablesSubscriberType$ || (exports.BenefitDownloadablesSubscriberType$ = BenefitDownloadablesSubscriberType$ = {}));
/** @internal */
exports.BenefitDownloadablesSubscriber$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("downloadables").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitdownloadablessubscriberproperties_js_1.BenefitDownloadablesSubscriberProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDownloadablesSubscriber$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("downloadables").default("downloadables"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitdownloadablessubscriberproperties_js_1.BenefitDownloadablesSubscriberProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesSubscriber$;
(function (BenefitDownloadablesSubscriber$) {
    /** @deprecated use `BenefitDownloadablesSubscriber$inboundSchema` instead. */
    BenefitDownloadablesSubscriber$.inboundSchema = exports.BenefitDownloadablesSubscriber$inboundSchema;
    /** @deprecated use `BenefitDownloadablesSubscriber$outboundSchema` instead. */
    BenefitDownloadablesSubscriber$.outboundSchema = exports.BenefitDownloadablesSubscriber$outboundSchema;
})(BenefitDownloadablesSubscriber$ || (exports.BenefitDownloadablesSubscriber$ = BenefitDownloadablesSubscriber$ = {}));
