// @polar-sh/sdk/models/components/benefitgithubrepositorycreateproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositoryCreateProperties$ = exports.BenefitGitHubRepositoryCreateProperties$outboundSchema = exports.BenefitGitHubRepositoryCreateProperties$inboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$ = exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/**
 * The permission level to grant. Read more about roles and their permissions on [GitHub documentation](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role).
 */
exports.BenefitGitHubRepositoryCreatePropertiesPermission = {
    Pull: "pull",
    Triage: "triage",
    Push: "push",
    Maintain: "maintain",
    Admin: "admin",
};
/** @internal */
exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema = z
    .nativeEnum(exports.BenefitGitHubRepositoryCreatePropertiesPermission);
/** @internal */
exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreatePropertiesPermission$;
(function (BenefitGitHubRepositoryCreatePropertiesPermission$) {
    /** @deprecated use `BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema` instead. */
    BenefitGitHubRepositoryCreatePropertiesPermission$.inboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema` instead. */
    BenefitGitHubRepositoryCreatePropertiesPermission$.outboundSchema = exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema;
})(BenefitGitHubRepositoryCreatePropertiesPermission$ || (exports.BenefitGitHubRepositoryCreatePropertiesPermission$ = BenefitGitHubRepositoryCreatePropertiesPermission$ = {}));
/** @internal */
exports.BenefitGitHubRepositoryCreateProperties$inboundSchema = z.object({
    repository_id: z.nullable(z.string()).optional(),
    repository_owner: z.nullable(z.string()).optional(),
    repository_name: z.nullable(z.string()).optional(),
    permission: exports.BenefitGitHubRepositoryCreatePropertiesPermission$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "repository_id": "repositoryId",
        "repository_owner": "repositoryOwner",
        "repository_name": "repositoryName",
    });
});
/** @internal */
exports.BenefitGitHubRepositoryCreateProperties$outboundSchema = z.object({
    repositoryId: z.nullable(z.string()).optional(),
    repositoryOwner: z.nullable(z.string()).optional(),
    repositoryName: z.nullable(z.string()).optional(),
    permission: exports.BenefitGitHubRepositoryCreatePropertiesPermission$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        repositoryId: "repository_id",
        repositoryOwner: "repository_owner",
        repositoryName: "repository_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreateProperties$;
(function (BenefitGitHubRepositoryCreateProperties$) {
    /** @deprecated use `BenefitGitHubRepositoryCreateProperties$inboundSchema` instead. */
    BenefitGitHubRepositoryCreateProperties$.inboundSchema = exports.BenefitGitHubRepositoryCreateProperties$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreateProperties$outboundSchema` instead. */
    BenefitGitHubRepositoryCreateProperties$.outboundSchema = exports.BenefitGitHubRepositoryCreateProperties$outboundSchema;
})(BenefitGitHubRepositoryCreateProperties$ || (exports.BenefitGitHubRepositoryCreateProperties$ = BenefitGitHubRepositoryCreateProperties$ = {}));
