// @polar-sh/sdk/models/components/benefitcustom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitCustom$ = exports.BenefitCustom$outboundSchema = exports.BenefitCustom$inboundSchema = exports.BenefitCustomType$ = exports.BenefitCustomType$outboundSchema = exports.BenefitCustomType$inboundSchema = exports.BenefitCustomType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitcustomproperties_js_1 = require("./benefitcustomproperties.js");
exports.BenefitCustomType = {
    Custom: "custom",
};
/** @internal */
exports.BenefitCustomType$inboundSchema = z.nativeEnum(exports.BenefitCustomType);
/** @internal */
exports.BenefitCustomType$outboundSchema = exports.BenefitCustomType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomType$;
(function (BenefitCustomType$) {
    /** @deprecated use `BenefitCustomType$inboundSchema` instead. */
    BenefitCustomType$.inboundSchema = exports.BenefitCustomType$inboundSchema;
    /** @deprecated use `BenefitCustomType$outboundSchema` instead. */
    BenefitCustomType$.outboundSchema = exports.BenefitCustomType$outboundSchema;
})(BenefitCustomType$ || (exports.BenefitCustomType$ = BenefitCustomType$ = {}));
/** @internal */
exports.BenefitCustom$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("custom").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitcustomproperties_js_1.BenefitCustomProperties$inboundSchema,
    is_tax_applicable: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
        "is_tax_applicable": "isTaxApplicable",
    });
});
/** @internal */
exports.BenefitCustom$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("custom").default("custom"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitcustomproperties_js_1.BenefitCustomProperties$outboundSchema,
    isTaxApplicable: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
        isTaxApplicable: "is_tax_applicable",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustom$;
(function (BenefitCustom$) {
    /** @deprecated use `BenefitCustom$inboundSchema` instead. */
    BenefitCustom$.inboundSchema = exports.BenefitCustom$inboundSchema;
    /** @deprecated use `BenefitCustom$outboundSchema` instead. */
    BenefitCustom$.outboundSchema = exports.BenefitCustom$outboundSchema;
})(BenefitCustom$ || (exports.BenefitCustom$ = BenefitCustom$ = {}));
