// @polar-sh/sdk/models/components/benefitlicensekeyexpirationproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeyExpirationProperties$ = exports.BenefitLicenseKeyExpirationProperties$outboundSchema = exports.BenefitLicenseKeyExpirationProperties$inboundSchema = exports.Timeframe$ = exports.Timeframe$outboundSchema = exports.Timeframe$inboundSchema = exports.Timeframe = void 0;
const z = __importStar(require("zod"));
exports.Timeframe = {
    Year: "year",
    Month: "month",
    Day: "day",
};
/** @internal */
exports.Timeframe$inboundSchema = z
    .nativeEnum(exports.Timeframe);
/** @internal */
exports.Timeframe$outboundSchema = exports.Timeframe$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Timeframe$;
(function (Timeframe$) {
    /** @deprecated use `Timeframe$inboundSchema` instead. */
    Timeframe$.inboundSchema = exports.Timeframe$inboundSchema;
    /** @deprecated use `Timeframe$outboundSchema` instead. */
    Timeframe$.outboundSchema = exports.Timeframe$outboundSchema;
})(Timeframe$ || (exports.Timeframe$ = Timeframe$ = {}));
/** @internal */
exports.BenefitLicenseKeyExpirationProperties$inboundSchema = z.object({
    ttl: z.number().int(),
    timeframe: exports.Timeframe$inboundSchema,
});
/** @internal */
exports.BenefitLicenseKeyExpirationProperties$outboundSchema = z.object({
    ttl: z.number().int(),
    timeframe: exports.Timeframe$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeyExpirationProperties$;
(function (BenefitLicenseKeyExpirationProperties$) {
    /** @deprecated use `BenefitLicenseKeyExpirationProperties$inboundSchema` instead. */
    BenefitLicenseKeyExpirationProperties$.inboundSchema = exports.BenefitLicenseKeyExpirationProperties$inboundSchema;
    /** @deprecated use `BenefitLicenseKeyExpirationProperties$outboundSchema` instead. */
    BenefitLicenseKeyExpirationProperties$.outboundSchema = exports.BenefitLicenseKeyExpirationProperties$outboundSchema;
})(BenefitLicenseKeyExpirationProperties$ || (exports.BenefitLicenseKeyExpirationProperties$ = BenefitLicenseKeyExpirationProperties$ = {}));
