// @polar-sh/sdk/models/components/benefitdownloadablesupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadablesUpdate$ = exports.BenefitDownloadablesUpdate$outboundSchema = exports.BenefitDownloadablesUpdate$inboundSchema = exports.BenefitDownloadablesUpdateType$ = exports.BenefitDownloadablesUpdateType$outboundSchema = exports.BenefitDownloadablesUpdateType$inboundSchema = exports.BenefitDownloadablesUpdateType = void 0;
const z = __importStar(require("zod"));
const benefitdownloadablescreateproperties_js_1 = require("./benefitdownloadablescreateproperties.js");
exports.BenefitDownloadablesUpdateType = {
    Downloadables: "downloadables",
};
/** @internal */
exports.BenefitDownloadablesUpdateType$inboundSchema = z.nativeEnum(exports.BenefitDownloadablesUpdateType);
/** @internal */
exports.BenefitDownloadablesUpdateType$outboundSchema = exports.BenefitDownloadablesUpdateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesUpdateType$;
(function (BenefitDownloadablesUpdateType$) {
    /** @deprecated use `BenefitDownloadablesUpdateType$inboundSchema` instead. */
    BenefitDownloadablesUpdateType$.inboundSchema = exports.BenefitDownloadablesUpdateType$inboundSchema;
    /** @deprecated use `BenefitDownloadablesUpdateType$outboundSchema` instead. */
    BenefitDownloadablesUpdateType$.outboundSchema = exports.BenefitDownloadablesUpdateType$outboundSchema;
})(BenefitDownloadablesUpdateType$ || (exports.BenefitDownloadablesUpdateType$ = BenefitDownloadablesUpdateType$ = {}));
/** @internal */
exports.BenefitDownloadablesUpdate$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("downloadables").optional(),
    properties: z.nullable(benefitdownloadablescreateproperties_js_1.BenefitDownloadablesCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitDownloadablesUpdate$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("downloadables").default("downloadables"),
    properties: z.nullable(benefitdownloadablescreateproperties_js_1.BenefitDownloadablesCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesUpdate$;
(function (BenefitDownloadablesUpdate$) {
    /** @deprecated use `BenefitDownloadablesUpdate$inboundSchema` instead. */
    BenefitDownloadablesUpdate$.inboundSchema = exports.BenefitDownloadablesUpdate$inboundSchema;
    /** @deprecated use `BenefitDownloadablesUpdate$outboundSchema` instead. */
    BenefitDownloadablesUpdate$.outboundSchema = exports.BenefitDownloadablesUpdate$outboundSchema;
})(BenefitDownloadablesUpdate$ || (exports.BenefitDownloadablesUpdate$ = BenefitDownloadablesUpdate$ = {}));
