// @polar-sh/sdk/models/components/benefittype.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitType$ = exports.BenefitType$outboundSchema = exports.BenefitType$inboundSchema = exports.BenefitType = void 0;
const z = __importStar(require("zod"));
exports.BenefitType = {
    Custom: "custom",
    Articles: "articles",
    Ads: "ads",
    Discord: "discord",
    GithubRepository: "github_repository",
    Downloadables: "downloadables",
    LicenseKeys: "license_keys",
};
/** @internal */
exports.BenefitType$inboundSchema = z
    .nativeEnum(exports.BenefitType);
/** @internal */
exports.BenefitType$outboundSchema = exports.BenefitType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitType$;
(function (BenefitType$) {
    /** @deprecated use `BenefitType$inboundSchema` instead. */
    BenefitType$.inboundSchema = exports.BenefitType$inboundSchema;
    /** @deprecated use `BenefitType$outboundSchema` instead. */
    BenefitType$.outboundSchema = exports.BenefitType$outboundSchema;
})(BenefitType$ || (exports.BenefitType$ = BenefitType$ = {}));
