// @polar-sh/sdk/models/components/benefitdownloadables.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadables$ = exports.BenefitDownloadables$outboundSchema = exports.BenefitDownloadables$inboundSchema = exports.BenefitDownloadablesType$ = exports.BenefitDownloadablesType$outboundSchema = exports.BenefitDownloadablesType$inboundSchema = exports.BenefitDownloadablesType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitdownloadablesproperties_js_1 = require("./benefitdownloadablesproperties.js");
exports.BenefitDownloadablesType = {
    Downloadables: "downloadables",
};
/** @internal */
exports.BenefitDownloadablesType$inboundSchema = z.nativeEnum(exports.BenefitDownloadablesType);
/** @internal */
exports.BenefitDownloadablesType$outboundSchema = exports.BenefitDownloadablesType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesType$;
(function (BenefitDownloadablesType$) {
    /** @deprecated use `BenefitDownloadablesType$inboundSchema` instead. */
    BenefitDownloadablesType$.inboundSchema = exports.BenefitDownloadablesType$inboundSchema;
    /** @deprecated use `BenefitDownloadablesType$outboundSchema` instead. */
    BenefitDownloadablesType$.outboundSchema = exports.BenefitDownloadablesType$outboundSchema;
})(BenefitDownloadablesType$ || (exports.BenefitDownloadablesType$ = BenefitDownloadablesType$ = {}));
/** @internal */
exports.BenefitDownloadables$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("downloadables").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitdownloadablesproperties_js_1.BenefitDownloadablesProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDownloadables$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("downloadables").default("downloadables"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitdownloadablesproperties_js_1.BenefitDownloadablesProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadables$;
(function (BenefitDownloadables$) {
    /** @deprecated use `BenefitDownloadables$inboundSchema` instead. */
    BenefitDownloadables$.inboundSchema = exports.BenefitDownloadables$inboundSchema;
    /** @deprecated use `BenefitDownloadables$outboundSchema` instead. */
    BenefitDownloadables$.outboundSchema = exports.BenefitDownloadables$outboundSchema;
})(BenefitDownloadables$ || (exports.BenefitDownloadables$ = BenefitDownloadables$ = {}));
