// @polar-sh/sdk/models/components/benefitdiscordupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordUpdate$ = exports.BenefitDiscordUpdate$outboundSchema = exports.BenefitDiscordUpdate$inboundSchema = exports.BenefitDiscordUpdateType$ = exports.BenefitDiscordUpdateType$outboundSchema = exports.BenefitDiscordUpdateType$inboundSchema = exports.BenefitDiscordUpdateType = void 0;
const z = __importStar(require("zod"));
const benefitdiscordcreateproperties_js_1 = require("./benefitdiscordcreateproperties.js");
exports.BenefitDiscordUpdateType = {
    Discord: "discord",
};
/** @internal */
exports.BenefitDiscordUpdateType$inboundSchema = z.nativeEnum(exports.BenefitDiscordUpdateType);
/** @internal */
exports.BenefitDiscordUpdateType$outboundSchema = exports.BenefitDiscordUpdateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordUpdateType$;
(function (BenefitDiscordUpdateType$) {
    /** @deprecated use `BenefitDiscordUpdateType$inboundSchema` instead. */
    BenefitDiscordUpdateType$.inboundSchema = exports.BenefitDiscordUpdateType$inboundSchema;
    /** @deprecated use `BenefitDiscordUpdateType$outboundSchema` instead. */
    BenefitDiscordUpdateType$.outboundSchema = exports.BenefitDiscordUpdateType$outboundSchema;
})(BenefitDiscordUpdateType$ || (exports.BenefitDiscordUpdateType$ = BenefitDiscordUpdateType$ = {}));
/** @internal */
exports.BenefitDiscordUpdate$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("discord").optional(),
    properties: z.nullable(benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitDiscordUpdate$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("discord").default("discord"),
    properties: z.nullable(benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordUpdate$;
(function (BenefitDiscordUpdate$) {
    /** @deprecated use `BenefitDiscordUpdate$inboundSchema` instead. */
    BenefitDiscordUpdate$.inboundSchema = exports.BenefitDiscordUpdate$inboundSchema;
    /** @deprecated use `BenefitDiscordUpdate$outboundSchema` instead. */
    BenefitDiscordUpdate$.outboundSchema = exports.BenefitDiscordUpdate$outboundSchema;
})(BenefitDiscordUpdate$ || (exports.BenefitDiscordUpdate$ = BenefitDiscordUpdate$ = {}));
