// @polar-sh/sdk/models/components/benefitcustomcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitCustomCreate$ = exports.BenefitCustomCreate$outboundSchema = exports.BenefitCustomCreate$inboundSchema = exports.BenefitCustomCreateType$ = exports.BenefitCustomCreateType$outboundSchema = exports.BenefitCustomCreateType$inboundSchema = exports.BenefitCustomCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitcustomcreateproperties_js_1 = require("./benefitcustomcreateproperties.js");
exports.BenefitCustomCreateType = {
    Custom: "custom",
};
/** @internal */
exports.BenefitCustomCreateType$inboundSchema = z.nativeEnum(exports.BenefitCustomCreateType);
/** @internal */
exports.BenefitCustomCreateType$outboundSchema = exports.BenefitCustomCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomCreateType$;
(function (BenefitCustomCreateType$) {
    /** @deprecated use `BenefitCustomCreateType$inboundSchema` instead. */
    BenefitCustomCreateType$.inboundSchema = exports.BenefitCustomCreateType$inboundSchema;
    /** @deprecated use `BenefitCustomCreateType$outboundSchema` instead. */
    BenefitCustomCreateType$.outboundSchema = exports.BenefitCustomCreateType$outboundSchema;
})(BenefitCustomCreateType$ || (exports.BenefitCustomCreateType$ = BenefitCustomCreateType$ = {}));
/** @internal */
exports.BenefitCustomCreate$inboundSchema = z.object({
    type: z.literal("custom").optional(),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    is_tax_applicable: z.boolean(),
    properties: benefitcustomcreateproperties_js_1.BenefitCustomCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "is_tax_applicable": "isTaxApplicable",
    });
});
/** @internal */
exports.BenefitCustomCreate$outboundSchema = z.object({
    type: z.literal("custom").default("custom"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    isTaxApplicable: z.boolean(),
    properties: benefitcustomcreateproperties_js_1.BenefitCustomCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        isTaxApplicable: "is_tax_applicable",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomCreate$;
(function (BenefitCustomCreate$) {
    /** @deprecated use `BenefitCustomCreate$inboundSchema` instead. */
    BenefitCustomCreate$.inboundSchema = exports.BenefitCustomCreate$inboundSchema;
    /** @deprecated use `BenefitCustomCreate$outboundSchema` instead. */
    BenefitCustomCreate$.outboundSchema = exports.BenefitCustomCreate$outboundSchema;
})(BenefitCustomCreate$ || (exports.BenefitCustomCreate$ = BenefitCustomCreate$ = {}));
