// @polar-sh/sdk/models/components/benefitarticlesupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitArticlesUpdate$ = exports.BenefitArticlesUpdate$outboundSchema = exports.BenefitArticlesUpdate$inboundSchema = exports.BenefitArticlesUpdateType$ = exports.BenefitArticlesUpdateType$outboundSchema = exports.BenefitArticlesUpdateType$inboundSchema = exports.BenefitArticlesUpdateType = void 0;
const z = __importStar(require("zod"));
exports.BenefitArticlesUpdateType = {
    Articles: "articles",
};
/** @internal */
exports.BenefitArticlesUpdateType$inboundSchema = z.nativeEnum(exports.BenefitArticlesUpdateType);
/** @internal */
exports.BenefitArticlesUpdateType$outboundSchema = exports.BenefitArticlesUpdateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitArticlesUpdateType$;
(function (BenefitArticlesUpdateType$) {
    /** @deprecated use `BenefitArticlesUpdateType$inboundSchema` instead. */
    BenefitArticlesUpdateType$.inboundSchema = exports.BenefitArticlesUpdateType$inboundSchema;
    /** @deprecated use `BenefitArticlesUpdateType$outboundSchema` instead. */
    BenefitArticlesUpdateType$.outboundSchema = exports.BenefitArticlesUpdateType$outboundSchema;
})(BenefitArticlesUpdateType$ || (exports.BenefitArticlesUpdateType$ = BenefitArticlesUpdateType$ = {}));
/** @internal */
exports.BenefitArticlesUpdate$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("articles").optional(),
});
/** @internal */
exports.BenefitArticlesUpdate$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("articles").default("articles"),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitArticlesUpdate$;
(function (BenefitArticlesUpdate$) {
    /** @deprecated use `BenefitArticlesUpdate$inboundSchema` instead. */
    BenefitArticlesUpdate$.inboundSchema = exports.BenefitArticlesUpdate$inboundSchema;
    /** @deprecated use `BenefitArticlesUpdate$outboundSchema` instead. */
    BenefitArticlesUpdate$.outboundSchema = exports.BenefitArticlesUpdate$outboundSchema;
})(BenefitArticlesUpdate$ || (exports.BenefitArticlesUpdate$ = BenefitArticlesUpdate$ = {}));
