// @polar-sh/sdk/models/components/benefitgithubrepositorysubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositorySubscriber$ = exports.BenefitGitHubRepositorySubscriber$outboundSchema = exports.BenefitGitHubRepositorySubscriber$inboundSchema = exports.BenefitGitHubRepositorySubscriberType$ = exports.BenefitGitHubRepositorySubscriberType$outboundSchema = exports.BenefitGitHubRepositorySubscriberType$inboundSchema = exports.BenefitGitHubRepositorySubscriberType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitgithubrepositorysubscriberproperties_js_1 = require("./benefitgithubrepositorysubscriberproperties.js");
exports.BenefitGitHubRepositorySubscriberType = {
    GithubRepository: "github_repository",
};
/** @internal */
exports.BenefitGitHubRepositorySubscriberType$inboundSchema = z.nativeEnum(exports.BenefitGitHubRepositorySubscriberType);
/** @internal */
exports.BenefitGitHubRepositorySubscriberType$outboundSchema = exports.BenefitGitHubRepositorySubscriberType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositorySubscriberType$;
(function (BenefitGitHubRepositorySubscriberType$) {
    /** @deprecated use `BenefitGitHubRepositorySubscriberType$inboundSchema` instead. */
    BenefitGitHubRepositorySubscriberType$.inboundSchema = exports.BenefitGitHubRepositorySubscriberType$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositorySubscriberType$outboundSchema` instead. */
    BenefitGitHubRepositorySubscriberType$.outboundSchema = exports.BenefitGitHubRepositorySubscriberType$outboundSchema;
})(BenefitGitHubRepositorySubscriberType$ || (exports.BenefitGitHubRepositorySubscriberType$ = BenefitGitHubRepositorySubscriberType$ = {}));
/** @internal */
exports.BenefitGitHubRepositorySubscriber$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("github_repository").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitgithubrepositorysubscriberproperties_js_1.BenefitGitHubRepositorySubscriberProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitGitHubRepositorySubscriber$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("github_repository").default("github_repository"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitgithubrepositorysubscriberproperties_js_1.BenefitGitHubRepositorySubscriberProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositorySubscriber$;
(function (BenefitGitHubRepositorySubscriber$) {
    /** @deprecated use `BenefitGitHubRepositorySubscriber$inboundSchema` instead. */
    BenefitGitHubRepositorySubscriber$.inboundSchema = exports.BenefitGitHubRepositorySubscriber$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositorySubscriber$outboundSchema` instead. */
    BenefitGitHubRepositorySubscriber$.outboundSchema = exports.BenefitGitHubRepositorySubscriber$outboundSchema;
})(BenefitGitHubRepositorySubscriber$ || (exports.BenefitGitHubRepositorySubscriber$ = BenefitGitHubRepositorySubscriber$ = {}));
