// @polar-sh/sdk/models/components/benefitcustomsubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitCustomSubscriber$ = exports.BenefitCustomSubscriber$outboundSchema = exports.BenefitCustomSubscriber$inboundSchema = exports.BenefitCustomSubscriberType$ = exports.BenefitCustomSubscriberType$outboundSchema = exports.BenefitCustomSubscriberType$inboundSchema = exports.BenefitCustomSubscriberType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitcustomsubscriberproperties_js_1 = require("./benefitcustomsubscriberproperties.js");
const benefitgrantsubscriber_js_1 = require("./benefitgrantsubscriber.js");
exports.BenefitCustomSubscriberType = {
    Custom: "custom",
};
/** @internal */
exports.BenefitCustomSubscriberType$inboundSchema = z.nativeEnum(exports.BenefitCustomSubscriberType);
/** @internal */
exports.BenefitCustomSubscriberType$outboundSchema = exports.BenefitCustomSubscriberType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomSubscriberType$;
(function (BenefitCustomSubscriberType$) {
    /** @deprecated use `BenefitCustomSubscriberType$inboundSchema` instead. */
    BenefitCustomSubscriberType$.inboundSchema = exports.BenefitCustomSubscriberType$inboundSchema;
    /** @deprecated use `BenefitCustomSubscriberType$outboundSchema` instead. */
    BenefitCustomSubscriberType$.outboundSchema = exports.BenefitCustomSubscriberType$outboundSchema;
})(BenefitCustomSubscriberType$ || (exports.BenefitCustomSubscriberType$ = BenefitCustomSubscriberType$ = {}));
/** @internal */
exports.BenefitCustomSubscriber$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("custom").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    grants: z.array(benefitgrantsubscriber_js_1.BenefitGrantSubscriber$inboundSchema),
    properties: benefitcustomsubscriberproperties_js_1.BenefitCustomSubscriberProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitCustomSubscriber$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("custom").default("custom"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    grants: z.array(benefitgrantsubscriber_js_1.BenefitGrantSubscriber$outboundSchema),
    properties: benefitcustomsubscriberproperties_js_1.BenefitCustomSubscriberProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCustomSubscriber$;
(function (BenefitCustomSubscriber$) {
    /** @deprecated use `BenefitCustomSubscriber$inboundSchema` instead. */
    BenefitCustomSubscriber$.inboundSchema = exports.BenefitCustomSubscriber$inboundSchema;
    /** @deprecated use `BenefitCustomSubscriber$outboundSchema` instead. */
    BenefitCustomSubscriber$.outboundSchema = exports.BenefitCustomSubscriber$outboundSchema;
})(BenefitCustomSubscriber$ || (exports.BenefitCustomSubscriber$ = BenefitCustomSubscriber$ = {}));
