// @polar-sh/sdk/models/components/benefitdiscordproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordProperties$ = exports.BenefitDiscordProperties$outboundSchema = exports.BenefitDiscordProperties$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.BenefitDiscordProperties$inboundSchema = z.object({
    guild_id: z.string(),
    role_id: z.string(),
    guild_token: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "guild_id": "guildId",
        "role_id": "roleId",
        "guild_token": "guildToken",
    });
});
/** @internal */
exports.BenefitDiscordProperties$outboundSchema = z.object({
    guildId: z.string(),
    roleId: z.string(),
    guildToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        guildId: "guild_id",
        roleId: "role_id",
        guildToken: "guild_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordProperties$;
(function (BenefitDiscordProperties$) {
    /** @deprecated use `BenefitDiscordProperties$inboundSchema` instead. */
    BenefitDiscordProperties$.inboundSchema = exports.BenefitDiscordProperties$inboundSchema;
    /** @deprecated use `BenefitDiscordProperties$outboundSchema` instead. */
    BenefitDiscordProperties$.outboundSchema = exports.BenefitDiscordProperties$outboundSchema;
})(BenefitDiscordProperties$ || (exports.BenefitDiscordProperties$ = BenefitDiscordProperties$ = {}));
