// @polar-sh/sdk/models/components/benefitgithubrepositorycreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositoryCreate$ = exports.BenefitGitHubRepositoryCreate$outboundSchema = exports.BenefitGitHubRepositoryCreate$inboundSchema = exports.BenefitGitHubRepositoryCreateType$ = exports.BenefitGitHubRepositoryCreateType$outboundSchema = exports.BenefitGitHubRepositoryCreateType$inboundSchema = exports.BenefitGitHubRepositoryCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitgithubrepositorycreateproperties_js_1 = require("./benefitgithubrepositorycreateproperties.js");
exports.BenefitGitHubRepositoryCreateType = {
    GithubRepository: "github_repository",
};
/** @internal */
exports.BenefitGitHubRepositoryCreateType$inboundSchema = z.nativeEnum(exports.BenefitGitHubRepositoryCreateType);
/** @internal */
exports.BenefitGitHubRepositoryCreateType$outboundSchema = exports.BenefitGitHubRepositoryCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreateType$;
(function (BenefitGitHubRepositoryCreateType$) {
    /** @deprecated use `BenefitGitHubRepositoryCreateType$inboundSchema` instead. */
    BenefitGitHubRepositoryCreateType$.inboundSchema = exports.BenefitGitHubRepositoryCreateType$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreateType$outboundSchema` instead. */
    BenefitGitHubRepositoryCreateType$.outboundSchema = exports.BenefitGitHubRepositoryCreateType$outboundSchema;
})(BenefitGitHubRepositoryCreateType$ || (exports.BenefitGitHubRepositoryCreateType$ = BenefitGitHubRepositoryCreateType$ = {}));
/** @internal */
exports.BenefitGitHubRepositoryCreate$inboundSchema = z.object({
    type: z.literal("github_repository").optional(),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitGitHubRepositoryCreate$outboundSchema = z.object({
    type: z.literal("github_repository").default("github_repository"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryCreate$;
(function (BenefitGitHubRepositoryCreate$) {
    /** @deprecated use `BenefitGitHubRepositoryCreate$inboundSchema` instead. */
    BenefitGitHubRepositoryCreate$.inboundSchema = exports.BenefitGitHubRepositoryCreate$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryCreate$outboundSchema` instead. */
    BenefitGitHubRepositoryCreate$.outboundSchema = exports.BenefitGitHubRepositoryCreate$outboundSchema;
})(BenefitGitHubRepositoryCreate$ || (exports.BenefitGitHubRepositoryCreate$ = BenefitGitHubRepositoryCreate$ = {}));
