// @polar-sh/sdk/models/components/benefitdiscordcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordCreate$ = exports.BenefitDiscordCreate$outboundSchema = exports.BenefitDiscordCreate$inboundSchema = exports.BenefitDiscordCreateType$ = exports.BenefitDiscordCreateType$outboundSchema = exports.BenefitDiscordCreateType$inboundSchema = exports.BenefitDiscordCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitdiscordcreateproperties_js_1 = require("./benefitdiscordcreateproperties.js");
exports.BenefitDiscordCreateType = {
    Discord: "discord",
};
/** @internal */
exports.BenefitDiscordCreateType$inboundSchema = z.nativeEnum(exports.BenefitDiscordCreateType);
/** @internal */
exports.BenefitDiscordCreateType$outboundSchema = exports.BenefitDiscordCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordCreateType$;
(function (BenefitDiscordCreateType$) {
    /** @deprecated use `BenefitDiscordCreateType$inboundSchema` instead. */
    BenefitDiscordCreateType$.inboundSchema = exports.BenefitDiscordCreateType$inboundSchema;
    /** @deprecated use `BenefitDiscordCreateType$outboundSchema` instead. */
    BenefitDiscordCreateType$.outboundSchema = exports.BenefitDiscordCreateType$outboundSchema;
})(BenefitDiscordCreateType$ || (exports.BenefitDiscordCreateType$ = BenefitDiscordCreateType$ = {}));
/** @internal */
exports.BenefitDiscordCreate$inboundSchema = z.object({
    type: z.literal("discord").optional(),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDiscordCreate$outboundSchema = z.object({
    type: z.literal("discord").default("discord"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitdiscordcreateproperties_js_1.BenefitDiscordCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordCreate$;
(function (BenefitDiscordCreate$) {
    /** @deprecated use `BenefitDiscordCreate$inboundSchema` instead. */
    BenefitDiscordCreate$.inboundSchema = exports.BenefitDiscordCreate$inboundSchema;
    /** @deprecated use `BenefitDiscordCreate$outboundSchema` instead. */
    BenefitDiscordCreate$.outboundSchema = exports.BenefitDiscordCreate$outboundSchema;
})(BenefitDiscordCreate$ || (exports.BenefitDiscordCreate$ = BenefitDiscordCreate$ = {}));
