// @polar-sh/sdk/models/components/benefitlicensekeyscreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitLicenseKeysCreate$ = exports.BenefitLicenseKeysCreate$outboundSchema = exports.BenefitLicenseKeysCreate$inboundSchema = exports.BenefitLicenseKeysCreateType$ = exports.BenefitLicenseKeysCreateType$outboundSchema = exports.BenefitLicenseKeysCreateType$inboundSchema = exports.BenefitLicenseKeysCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitlicensekeyscreateproperties_js_1 = require("./benefitlicensekeyscreateproperties.js");
exports.BenefitLicenseKeysCreateType = {
    LicenseKeys: "license_keys",
};
/** @internal */
exports.BenefitLicenseKeysCreateType$inboundSchema = z.nativeEnum(exports.BenefitLicenseKeysCreateType);
/** @internal */
exports.BenefitLicenseKeysCreateType$outboundSchema = exports.BenefitLicenseKeysCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysCreateType$;
(function (BenefitLicenseKeysCreateType$) {
    /** @deprecated use `BenefitLicenseKeysCreateType$inboundSchema` instead. */
    BenefitLicenseKeysCreateType$.inboundSchema = exports.BenefitLicenseKeysCreateType$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysCreateType$outboundSchema` instead. */
    BenefitLicenseKeysCreateType$.outboundSchema = exports.BenefitLicenseKeysCreateType$outboundSchema;
})(BenefitLicenseKeysCreateType$ || (exports.BenefitLicenseKeysCreateType$ = BenefitLicenseKeysCreateType$ = {}));
/** @internal */
exports.BenefitLicenseKeysCreate$inboundSchema = z.object({
    type: z.literal("license_keys").optional(),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitLicenseKeysCreate$outboundSchema = z.object({
    type: z.literal("license_keys").default("license_keys"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitlicensekeyscreateproperties_js_1.BenefitLicenseKeysCreateProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitLicenseKeysCreate$;
(function (BenefitLicenseKeysCreate$) {
    /** @deprecated use `BenefitLicenseKeysCreate$inboundSchema` instead. */
    BenefitLicenseKeysCreate$.inboundSchema = exports.BenefitLicenseKeysCreate$inboundSchema;
    /** @deprecated use `BenefitLicenseKeysCreate$outboundSchema` instead. */
    BenefitLicenseKeysCreate$.outboundSchema = exports.BenefitLicenseKeysCreate$outboundSchema;
})(BenefitLicenseKeysCreate$ || (exports.BenefitLicenseKeysCreate$ = BenefitLicenseKeysCreate$ = {}));
