// @polar-sh/sdk/models/components/benefitadscreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitAdsCreate$ = exports.BenefitAdsCreate$outboundSchema = exports.BenefitAdsCreate$inboundSchema = exports.BenefitAdsCreateType$ = exports.BenefitAdsCreateType$outboundSchema = exports.BenefitAdsCreateType$inboundSchema = exports.BenefitAdsCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitadsproperties_js_1 = require("./benefitadsproperties.js");
exports.BenefitAdsCreateType = {
    Ads: "ads",
};
/** @internal */
exports.BenefitAdsCreateType$inboundSchema = z.nativeEnum(exports.BenefitAdsCreateType);
/** @internal */
exports.BenefitAdsCreateType$outboundSchema = exports.BenefitAdsCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitAdsCreateType$;
(function (BenefitAdsCreateType$) {
    /** @deprecated use `BenefitAdsCreateType$inboundSchema` instead. */
    BenefitAdsCreateType$.inboundSchema = exports.BenefitAdsCreateType$inboundSchema;
    /** @deprecated use `BenefitAdsCreateType$outboundSchema` instead. */
    BenefitAdsCreateType$.outboundSchema = exports.BenefitAdsCreateType$outboundSchema;
})(BenefitAdsCreateType$ || (exports.BenefitAdsCreateType$ = BenefitAdsCreateType$ = {}));
/** @internal */
exports.BenefitAdsCreate$inboundSchema = z.object({
    type: z.literal("ads").optional(),
    description: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    properties: benefitadsproperties_js_1.BenefitAdsProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitAdsCreate$outboundSchema = z.object({
    type: z.literal("ads").default("ads"),
    description: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    properties: benefitadsproperties_js_1.BenefitAdsProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitAdsCreate$;
(function (BenefitAdsCreate$) {
    /** @deprecated use `BenefitAdsCreate$inboundSchema` instead. */
    BenefitAdsCreate$.inboundSchema = exports.BenefitAdsCreate$inboundSchema;
    /** @deprecated use `BenefitAdsCreate$outboundSchema` instead. */
    BenefitAdsCreate$.outboundSchema = exports.BenefitAdsCreate$outboundSchema;
})(BenefitAdsCreate$ || (exports.BenefitAdsCreate$ = BenefitAdsCreate$ = {}));
