// @polar-sh/sdk/models/components/benefitgithubrepositoryupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGitHubRepositoryUpdate$ = exports.BenefitGitHubRepositoryUpdate$outboundSchema = exports.BenefitGitHubRepositoryUpdate$inboundSchema = exports.BenefitGitHubRepositoryUpdateType$ = exports.BenefitGitHubRepositoryUpdateType$outboundSchema = exports.BenefitGitHubRepositoryUpdateType$inboundSchema = exports.BenefitGitHubRepositoryUpdateType = void 0;
const z = __importStar(require("zod"));
const benefitgithubrepositorycreateproperties_js_1 = require("./benefitgithubrepositorycreateproperties.js");
exports.BenefitGitHubRepositoryUpdateType = {
    GithubRepository: "github_repository",
};
/** @internal */
exports.BenefitGitHubRepositoryUpdateType$inboundSchema = z.nativeEnum(exports.BenefitGitHubRepositoryUpdateType);
/** @internal */
exports.BenefitGitHubRepositoryUpdateType$outboundSchema = exports.BenefitGitHubRepositoryUpdateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryUpdateType$;
(function (BenefitGitHubRepositoryUpdateType$) {
    /** @deprecated use `BenefitGitHubRepositoryUpdateType$inboundSchema` instead. */
    BenefitGitHubRepositoryUpdateType$.inboundSchema = exports.BenefitGitHubRepositoryUpdateType$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryUpdateType$outboundSchema` instead. */
    BenefitGitHubRepositoryUpdateType$.outboundSchema = exports.BenefitGitHubRepositoryUpdateType$outboundSchema;
})(BenefitGitHubRepositoryUpdateType$ || (exports.BenefitGitHubRepositoryUpdateType$ = BenefitGitHubRepositoryUpdateType$ = {}));
/** @internal */
exports.BenefitGitHubRepositoryUpdate$inboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("github_repository").optional(),
    properties: z.nullable(benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.BenefitGitHubRepositoryUpdate$outboundSchema = z.object({
    description: z.nullable(z.string()).optional(),
    type: z.literal("github_repository").default("github_repository"),
    properties: z.nullable(benefitgithubrepositorycreateproperties_js_1.BenefitGitHubRepositoryCreateProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGitHubRepositoryUpdate$;
(function (BenefitGitHubRepositoryUpdate$) {
    /** @deprecated use `BenefitGitHubRepositoryUpdate$inboundSchema` instead. */
    BenefitGitHubRepositoryUpdate$.inboundSchema = exports.BenefitGitHubRepositoryUpdate$inboundSchema;
    /** @deprecated use `BenefitGitHubRepositoryUpdate$outboundSchema` instead. */
    BenefitGitHubRepositoryUpdate$.outboundSchema = exports.BenefitGitHubRepositoryUpdate$outboundSchema;
})(BenefitGitHubRepositoryUpdate$ || (exports.BenefitGitHubRepositoryUpdate$ = BenefitGitHubRepositoryUpdate$ = {}));
