// @polar-sh/sdk/models/components/benefitdiscord.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDiscordInput$ = exports.BenefitDiscordInput$outboundSchema = exports.BenefitDiscordInput$inboundSchema = exports.BenefitDiscord$ = exports.BenefitDiscord$outboundSchema = exports.BenefitDiscord$inboundSchema = exports.BenefitDiscordType$ = exports.BenefitDiscordType$outboundSchema = exports.BenefitDiscordType$inboundSchema = exports.BenefitDiscordType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitdiscordproperties_js_1 = require("./benefitdiscordproperties.js");
const benefitdiscordpropertiesinput_js_1 = require("./benefitdiscordpropertiesinput.js");
exports.BenefitDiscordType = {
    Discord: "discord",
};
/** @internal */
exports.BenefitDiscordType$inboundSchema = z.nativeEnum(exports.BenefitDiscordType);
/** @internal */
exports.BenefitDiscordType$outboundSchema = exports.BenefitDiscordType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordType$;
(function (BenefitDiscordType$) {
    /** @deprecated use `BenefitDiscordType$inboundSchema` instead. */
    BenefitDiscordType$.inboundSchema = exports.BenefitDiscordType$inboundSchema;
    /** @deprecated use `BenefitDiscordType$outboundSchema` instead. */
    BenefitDiscordType$.outboundSchema = exports.BenefitDiscordType$outboundSchema;
})(BenefitDiscordType$ || (exports.BenefitDiscordType$ = BenefitDiscordType$ = {}));
/** @internal */
exports.BenefitDiscord$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("discord").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitdiscordproperties_js_1.BenefitDiscordProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDiscord$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("discord").default("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitdiscordproperties_js_1.BenefitDiscordProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscord$;
(function (BenefitDiscord$) {
    /** @deprecated use `BenefitDiscord$inboundSchema` instead. */
    BenefitDiscord$.inboundSchema = exports.BenefitDiscord$inboundSchema;
    /** @deprecated use `BenefitDiscord$outboundSchema` instead. */
    BenefitDiscord$.outboundSchema = exports.BenefitDiscord$outboundSchema;
})(BenefitDiscord$ || (exports.BenefitDiscord$ = BenefitDiscord$ = {}));
/** @internal */
exports.BenefitDiscordInput$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("discord").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    properties: benefitdiscordpropertiesinput_js_1.BenefitDiscordPropertiesInput$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitDiscordInput$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("discord").default("discord"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    properties: benefitdiscordpropertiesinput_js_1.BenefitDiscordPropertiesInput$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDiscordInput$;
(function (BenefitDiscordInput$) {
    /** @deprecated use `BenefitDiscordInput$inboundSchema` instead. */
    BenefitDiscordInput$.inboundSchema = exports.BenefitDiscordInput$inboundSchema;
    /** @deprecated use `BenefitDiscordInput$outboundSchema` instead. */
    BenefitDiscordInput$.outboundSchema = exports.BenefitDiscordInput$outboundSchema;
})(BenefitDiscordInput$ || (exports.BenefitDiscordInput$ = BenefitDiscordInput$ = {}));
