// @polar-sh/sdk/models/components/checkoutlinkcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinkCreate$ = exports.CheckoutLinkCreate$outboundSchema = exports.CheckoutLinkCreate$inboundSchema = exports.CheckoutLinkCreatePaymentProcessor$ = exports.CheckoutLinkCreatePaymentProcessor$outboundSchema = exports.CheckoutLinkCreatePaymentProcessor$inboundSchema = exports.CheckoutLinkCreatePaymentProcessor = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/**
 * Payment processor to use. Currently only Stripe is supported.
 */
exports.CheckoutLinkCreatePaymentProcessor = {
    Stripe: "stripe",
};
/** @internal */
exports.CheckoutLinkCreatePaymentProcessor$inboundSchema = z.nativeEnum(exports.CheckoutLinkCreatePaymentProcessor);
/** @internal */
exports.CheckoutLinkCreatePaymentProcessor$outboundSchema = exports.CheckoutLinkCreatePaymentProcessor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkCreatePaymentProcessor$;
(function (CheckoutLinkCreatePaymentProcessor$) {
    /** @deprecated use `CheckoutLinkCreatePaymentProcessor$inboundSchema` instead. */
    CheckoutLinkCreatePaymentProcessor$.inboundSchema = exports.CheckoutLinkCreatePaymentProcessor$inboundSchema;
    /** @deprecated use `CheckoutLinkCreatePaymentProcessor$outboundSchema` instead. */
    CheckoutLinkCreatePaymentProcessor$.outboundSchema = exports.CheckoutLinkCreatePaymentProcessor$outboundSchema;
})(CheckoutLinkCreatePaymentProcessor$ || (exports.CheckoutLinkCreatePaymentProcessor$ = CheckoutLinkCreatePaymentProcessor$ = {}));
/** @internal */
exports.CheckoutLinkCreate$inboundSchema = z.object({
    metadata: z.record(z.string()).optional(),
    payment_processor: z.literal("stripe").optional(),
    product_price_id: z.string(),
    success_url: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "payment_processor": "paymentProcessor",
        "product_price_id": "productPriceId",
        "success_url": "successUrl",
    });
});
/** @internal */
exports.CheckoutLinkCreate$outboundSchema = z.object({
    metadata: z.record(z.string()).optional(),
    paymentProcessor: z.literal("stripe").default("stripe"),
    productPriceId: z.string(),
    successUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        paymentProcessor: "payment_processor",
        productPriceId: "product_price_id",
        successUrl: "success_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkCreate$;
(function (CheckoutLinkCreate$) {
    /** @deprecated use `CheckoutLinkCreate$inboundSchema` instead. */
    CheckoutLinkCreate$.inboundSchema = exports.CheckoutLinkCreate$inboundSchema;
    /** @deprecated use `CheckoutLinkCreate$outboundSchema` instead. */
    CheckoutLinkCreate$.outboundSchema = exports.CheckoutLinkCreate$outboundSchema;
})(CheckoutLinkCreate$ || (exports.CheckoutLinkCreate$ = CheckoutLinkCreate$ = {}));
