// @polar-sh/sdk/models/components/donation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Donation$ = exports.Donation$outboundSchema = exports.Donation$inboundSchema = exports.Donor$ = exports.Donor$outboundSchema = exports.Donor$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const donationorganization_js_1 = require("./donationorganization.js");
const donationuser_js_1 = require("./donationuser.js");
const issue_js_1 = require("./issue.js");
/** @internal */
exports.Donor$inboundSchema = z
    .union([donationuser_js_1.DonationUser$inboundSchema, donationorganization_js_1.DonationOrganization$inboundSchema]);
/** @internal */
exports.Donor$outboundSchema = z.union([donationuser_js_1.DonationUser$outboundSchema, donationorganization_js_1.DonationOrganization$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Donor$;
(function (Donor$) {
    /** @deprecated use `Donor$inboundSchema` instead. */
    Donor$.inboundSchema = exports.Donor$inboundSchema;
    /** @deprecated use `Donor$outboundSchema` instead. */
    Donor$.outboundSchema = exports.Donor$outboundSchema;
})(Donor$ || (exports.Donor$ = Donor$ = {}));
/** @internal */
exports.Donation$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount: z.number().int(),
    currency: z.string(),
    message: z.nullable(z.string()),
    donor: z.nullable(z.union([donationuser_js_1.DonationUser$inboundSchema, donationorganization_js_1.DonationOrganization$inboundSchema])),
    email: z.string(),
    issue: z.nullable(issue_js_1.Issue$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
    });
});
/** @internal */
exports.Donation$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amount: z.number().int(),
    currency: z.string(),
    message: z.nullable(z.string()),
    donor: z.nullable(z.union([donationuser_js_1.DonationUser$outboundSchema, donationorganization_js_1.DonationOrganization$outboundSchema])),
    email: z.string(),
    issue: z.nullable(issue_js_1.Issue$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Donation$;
(function (Donation$) {
    /** @deprecated use `Donation$inboundSchema` instead. */
    Donation$.inboundSchema = exports.Donation$inboundSchema;
    /** @deprecated use `Donation$outboundSchema` instead. */
    Donation$.outboundSchema = exports.Donation$outboundSchema;
})(Donation$ || (exports.Donation$ = Donation$ = {}));
