// @polar-sh/sdk/models/components/donationuser.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DonationUser$ = exports.DonationUser$outboundSchema = exports.DonationUser$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.DonationUser$inboundSchema = z.object({
    id: z.string(),
    public_name: z.string(),
    avatar_url: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "public_name": "publicName",
        "avatar_url": "avatarUrl",
    });
});
/** @internal */
exports.DonationUser$outboundSchema = z.object({
    id: z.string(),
    publicName: z.string(),
    avatarUrl: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        publicName: "public_name",
        avatarUrl: "avatar_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DonationUser$;
(function (DonationUser$) {
    /** @deprecated use `DonationUser$inboundSchema` instead. */
    DonationUser$.inboundSchema = exports.DonationUser$inboundSchema;
    /** @deprecated use `DonationUser$outboundSchema` instead. */
    DonationUser$.outboundSchema = exports.DonationUser$outboundSchema;
})(DonationUser$ || (exports.DonationUser$ = DonationUser$ = {}));
