// @polar-sh/sdk/models/components/downloadablefileread.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DownloadableFileRead$ = exports.DownloadableFileRead$outboundSchema = exports.DownloadableFileRead$inboundSchema = exports.DownloadableFileReadService$ = exports.DownloadableFileReadService$outboundSchema = exports.DownloadableFileReadService$inboundSchema = exports.DownloadableFileReadService = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.DownloadableFileReadService = {
    Downloadable: "downloadable",
};
/** @internal */
exports.DownloadableFileReadService$inboundSchema = z.nativeEnum(exports.DownloadableFileReadService);
/** @internal */
exports.DownloadableFileReadService$outboundSchema = exports.DownloadableFileReadService$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DownloadableFileReadService$;
(function (DownloadableFileReadService$) {
    /** @deprecated use `DownloadableFileReadService$inboundSchema` instead. */
    DownloadableFileReadService$.inboundSchema = exports.DownloadableFileReadService$inboundSchema;
    /** @deprecated use `DownloadableFileReadService$outboundSchema` instead. */
    DownloadableFileReadService$.outboundSchema = exports.DownloadableFileReadService$outboundSchema;
})(DownloadableFileReadService$ || (exports.DownloadableFileReadService$ = DownloadableFileReadService$ = {}));
/** @internal */
exports.DownloadableFileRead$inboundSchema = z.object({
    id: z.string(),
    organization_id: z.string(),
    name: z.string(),
    path: z.string(),
    mime_type: z.string(),
    size: z.number().int(),
    storage_version: z.nullable(z.string()),
    checksum_etag: z.nullable(z.string()),
    checksum_sha256_base64: z.nullable(z.string()),
    checksum_sha256_hex: z.nullable(z.string()),
    last_modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    version: z.nullable(z.string()),
    service: z.literal("downloadable").optional(),
    is_uploaded: z.boolean(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    size_readable: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "mime_type": "mimeType",
        "storage_version": "storageVersion",
        "checksum_etag": "checksumEtag",
        "checksum_sha256_base64": "checksumSha256Base64",
        "checksum_sha256_hex": "checksumSha256Hex",
        "last_modified_at": "lastModifiedAt",
        "is_uploaded": "isUploaded",
        "created_at": "createdAt",
        "size_readable": "sizeReadable",
    });
});
/** @internal */
exports.DownloadableFileRead$outboundSchema = z.object({
    id: z.string(),
    organizationId: z.string(),
    name: z.string(),
    path: z.string(),
    mimeType: z.string(),
    size: z.number().int(),
    storageVersion: z.nullable(z.string()),
    checksumEtag: z.nullable(z.string()),
    checksumSha256Base64: z.nullable(z.string()),
    checksumSha256Hex: z.nullable(z.string()),
    lastModifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    version: z.nullable(z.string()),
    service: z.literal("downloadable").default("downloadable"),
    isUploaded: z.boolean(),
    createdAt: z.date().transform(v => v.toISOString()),
    sizeReadable: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        mimeType: "mime_type",
        storageVersion: "storage_version",
        checksumEtag: "checksum_etag",
        checksumSha256Base64: "checksum_sha256_base64",
        checksumSha256Hex: "checksum_sha256_hex",
        lastModifiedAt: "last_modified_at",
        isUploaded: "is_uploaded",
        createdAt: "created_at",
        sizeReadable: "size_readable",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DownloadableFileRead$;
(function (DownloadableFileRead$) {
    /** @deprecated use `DownloadableFileRead$inboundSchema` instead. */
    DownloadableFileRead$.inboundSchema = exports.DownloadableFileRead$inboundSchema;
    /** @deprecated use `DownloadableFileRead$outboundSchema` instead. */
    DownloadableFileRead$.outboundSchema = exports.DownloadableFileRead$outboundSchema;
})(DownloadableFileRead$ || (exports.DownloadableFileRead$ = DownloadableFileRead$ = {}));
