// @polar-sh/sdk/models/components/externalorganization.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalOrganization$ = exports.ExternalOrganization$outboundSchema = exports.ExternalOrganization$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.ExternalOrganization$inboundSchema = z.object({
    id: z.string(),
    platform: z.literal("github").optional(),
    name: z.string(),
    avatar_url: z.string(),
    is_personal: z.boolean(),
    bio: z.nullable(z.string()),
    pretty_name: z.nullable(z.string()),
    company: z.nullable(z.string()),
    blog: z.nullable(z.string()),
    location: z.nullable(z.string()),
    email: z.nullable(z.string()),
    twitter_username: z.nullable(z.string()),
    organization_id: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "avatar_url": "avatarUrl",
        "is_personal": "isPersonal",
        "pretty_name": "prettyName",
        "twitter_username": "twitterUsername",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ExternalOrganization$outboundSchema = z.object({
    id: z.string(),
    platform: z.literal("github").default("github"),
    name: z.string(),
    avatarUrl: z.string(),
    isPersonal: z.boolean(),
    bio: z.nullable(z.string()),
    prettyName: z.nullable(z.string()),
    company: z.nullable(z.string()),
    blog: z.nullable(z.string()),
    location: z.nullable(z.string()),
    email: z.nullable(z.string()),
    twitterUsername: z.nullable(z.string()),
    organizationId: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        avatarUrl: "avatar_url",
        isPersonal: "is_personal",
        prettyName: "pretty_name",
        twitterUsername: "twitter_username",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExternalOrganization$;
(function (ExternalOrganization$) {
    /** @deprecated use `ExternalOrganization$inboundSchema` instead. */
    ExternalOrganization$.inboundSchema = exports.ExternalOrganization$inboundSchema;
    /** @deprecated use `ExternalOrganization$outboundSchema` instead. */
    ExternalOrganization$.outboundSchema = exports.ExternalOrganization$outboundSchema;
})(ExternalOrganization$ || (exports.ExternalOrganization$ = ExternalOrganization$ = {}));
