// @polar-sh/sdk/models/components/funding.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Funding$ = exports.Funding$outboundSchema = exports.Funding$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const currencyamount_js_1 = require("./currencyamount.js");
/** @internal */
exports.Funding$inboundSchema = z.object({
    funding_goal: z.nullable(currencyamount_js_1.CurrencyAmount$inboundSchema).optional(),
    pledges_sum: z.nullable(currencyamount_js_1.CurrencyAmount$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "funding_goal": "fundingGoal",
        "pledges_sum": "pledgesSum",
    });
});
/** @internal */
exports.Funding$outboundSchema = z.object({
    fundingGoal: z.nullable(currencyamount_js_1.CurrencyAmount$outboundSchema).optional(),
    pledgesSum: z.nullable(currencyamount_js_1.CurrencyAmount$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fundingGoal: "funding_goal",
        pledgesSum: "pledges_sum",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Funding$;
(function (Funding$) {
    /** @deprecated use `Funding$inboundSchema` instead. */
    Funding$.inboundSchema = exports.Funding$inboundSchema;
    /** @deprecated use `Funding$outboundSchema` instead. */
    Funding$.outboundSchema = exports.Funding$outboundSchema;
})(Funding$ || (exports.Funding$ = Funding$ = {}));
