// @polar-sh/sdk/models/components/licensekeyactivationbase.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseKeyActivationBase$ = exports.LicenseKeyActivationBase$outboundSchema = exports.LicenseKeyActivationBase$inboundSchema = exports.Meta$ = exports.Meta$outboundSchema = exports.Meta$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.Meta$inboundSchema = z
    .object({});
/** @internal */
exports.Meta$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Meta$;
(function (Meta$) {
    /** @deprecated use `Meta$inboundSchema` instead. */
    Meta$.inboundSchema = exports.Meta$inboundSchema;
    /** @deprecated use `Meta$outboundSchema` instead. */
    Meta$.outboundSchema = exports.Meta$outboundSchema;
})(Meta$ || (exports.Meta$ = Meta$ = {}));
/** @internal */
exports.LicenseKeyActivationBase$inboundSchema = z.object({
    id: z.string(),
    license_key_id: z.string(),
    label: z.string(),
    meta: z.lazy(() => exports.Meta$inboundSchema),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "license_key_id": "licenseKeyId",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
    });
});
/** @internal */
exports.LicenseKeyActivationBase$outboundSchema = z.object({
    id: z.string(),
    licenseKeyId: z.string(),
    label: z.string(),
    meta: z.lazy(() => exports.Meta$outboundSchema),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        licenseKeyId: "license_key_id",
        createdAt: "created_at",
        modifiedAt: "modified_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyActivationBase$;
(function (LicenseKeyActivationBase$) {
    /** @deprecated use `LicenseKeyActivationBase$inboundSchema` instead. */
    LicenseKeyActivationBase$.inboundSchema = exports.LicenseKeyActivationBase$inboundSchema;
    /** @deprecated use `LicenseKeyActivationBase$outboundSchema` instead. */
    LicenseKeyActivationBase$.outboundSchema = exports.LicenseKeyActivationBase$outboundSchema;
})(LicenseKeyActivationBase$ || (exports.LicenseKeyActivationBase$ = LicenseKeyActivationBase$ = {}));
