// @polar-sh/sdk/models/components/licensekeydeactivate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseKeyDeactivate$ = exports.LicenseKeyDeactivate$outboundSchema = exports.LicenseKeyDeactivate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.LicenseKeyDeactivate$inboundSchema = z.object({
    key: z.string(),
    organization_id: z.string(),
    activation_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "activation_id": "activationId",
    });
});
/** @internal */
exports.LicenseKeyDeactivate$outboundSchema = z.object({
    key: z.string(),
    organizationId: z.string(),
    activationId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        activationId: "activation_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyDeactivate$;
(function (LicenseKeyDeactivate$) {
    /** @deprecated use `LicenseKeyDeactivate$inboundSchema` instead. */
    LicenseKeyDeactivate$.inboundSchema = exports.LicenseKeyDeactivate$inboundSchema;
    /** @deprecated use `LicenseKeyDeactivate$outboundSchema` instead. */
    LicenseKeyDeactivate$.outboundSchema = exports.LicenseKeyDeactivate$outboundSchema;
})(LicenseKeyDeactivate$ || (exports.LicenseKeyDeactivate$ = LicenseKeyDeactivate$ = {}));
