// @polar-sh/sdk/models/components/listresourceuserbenefit.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceUserBenefit$ = exports.ListResourceUserBenefit$outboundSchema = exports.ListResourceUserBenefit$inboundSchema = exports.UserBenefit$ = exports.UserBenefit$outboundSchema = exports.UserBenefit$inboundSchema = void 0;
const z = __importStar(require("zod"));
const benefitadssubscriber_js_1 = require("./benefitadssubscriber.js");
const benefitarticlessubscriber_js_1 = require("./benefitarticlessubscriber.js");
const benefitcustomsubscriber_js_1 = require("./benefitcustomsubscriber.js");
const benefitdiscordsubscriber_js_1 = require("./benefitdiscordsubscriber.js");
const benefitdownloadablessubscriber_js_1 = require("./benefitdownloadablessubscriber.js");
const benefitgithubrepositorysubscriber_js_1 = require("./benefitgithubrepositorysubscriber.js");
const benefitlicensekeyssubscriber_js_1 = require("./benefitlicensekeyssubscriber.js");
const pagination_js_1 = require("./pagination.js");
/** @internal */
exports.UserBenefit$inboundSchema = z.union([
    benefitarticlessubscriber_js_1.BenefitArticlesSubscriber$inboundSchema.and(z.object({ type: z.literal("articles") }).transform((v) => ({
        type: v.type,
    }))),
    benefitdiscordsubscriber_js_1.BenefitDiscordSubscriber$inboundSchema.and(z.object({ type: z.literal("discord") }).transform((v) => ({
        type: v.type,
    }))),
    benefitdownloadablessubscriber_js_1.BenefitDownloadablesSubscriber$inboundSchema.and(z.object({ type: z.literal("downloadables") }).transform((v) => ({
        type: v.type,
    }))),
    benefitgithubrepositorysubscriber_js_1.BenefitGitHubRepositorySubscriber$inboundSchema.and(z.object({ type: z.literal("github_repository") }).transform((v) => ({
        type: v.type,
    }))),
    benefitadssubscriber_js_1.BenefitAdsSubscriber$inboundSchema.and(z.object({ type: z.literal("ads") }).transform((v) => ({ type: v.type }))),
    benefitcustomsubscriber_js_1.BenefitCustomSubscriber$inboundSchema.and(z.object({ type: z.literal("custom") }).transform((v) => ({
        type: v.type,
    }))),
    benefitlicensekeyssubscriber_js_1.BenefitLicenseKeysSubscriber$inboundSchema.and(z.object({ type: z.literal("license_keys") }).transform((v) => ({
        type: v.type,
    }))),
]);
/** @internal */
exports.UserBenefit$outboundSchema = z.union([
    benefitarticlessubscriber_js_1.BenefitArticlesSubscriber$outboundSchema.and(z.object({ type: z.literal("articles") }).transform((v) => ({
        type: v.type,
    }))),
    benefitdiscordsubscriber_js_1.BenefitDiscordSubscriber$outboundSchema.and(z.object({ type: z.literal("discord") }).transform((v) => ({
        type: v.type,
    }))),
    benefitdownloadablessubscriber_js_1.BenefitDownloadablesSubscriber$outboundSchema.and(z.object({ type: z.literal("downloadables") }).transform((v) => ({
        type: v.type,
    }))),
    benefitgithubrepositorysubscriber_js_1.BenefitGitHubRepositorySubscriber$outboundSchema.and(z.object({ type: z.literal("github_repository") }).transform((v) => ({
        type: v.type,
    }))),
    benefitadssubscriber_js_1.BenefitAdsSubscriber$outboundSchema.and(z.object({ type: z.literal("ads") }).transform((v) => ({ type: v.type }))),
    benefitcustomsubscriber_js_1.BenefitCustomSubscriber$outboundSchema.and(z.object({ type: z.literal("custom") }).transform((v) => ({
        type: v.type,
    }))),
    benefitlicensekeyssubscriber_js_1.BenefitLicenseKeysSubscriber$outboundSchema.and(z.object({ type: z.literal("license_keys") }).transform((v) => ({
        type: v.type,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserBenefit$;
(function (UserBenefit$) {
    /** @deprecated use `UserBenefit$inboundSchema` instead. */
    UserBenefit$.inboundSchema = exports.UserBenefit$inboundSchema;
    /** @deprecated use `UserBenefit$outboundSchema` instead. */
    UserBenefit$.outboundSchema = exports.UserBenefit$outboundSchema;
})(UserBenefit$ || (exports.UserBenefit$ = UserBenefit$ = {}));
/** @internal */
exports.ListResourceUserBenefit$inboundSchema = z.object({
    items: z.array(z.union([
        benefitarticlessubscriber_js_1.BenefitArticlesSubscriber$inboundSchema.and(z.object({ type: z.literal("articles") }).transform((v) => ({
            type: v.type,
        }))),
        benefitdiscordsubscriber_js_1.BenefitDiscordSubscriber$inboundSchema.and(z.object({ type: z.literal("discord") }).transform((v) => ({
            type: v.type,
        }))),
        benefitdownloadablessubscriber_js_1.BenefitDownloadablesSubscriber$inboundSchema.and(z.object({ type: z.literal("downloadables") }).transform((v) => ({
            type: v.type,
        }))),
        benefitgithubrepositorysubscriber_js_1.BenefitGitHubRepositorySubscriber$inboundSchema.and(z.object({ type: z.literal("github_repository") }).transform((v) => ({
            type: v.type,
        }))),
        benefitadssubscriber_js_1.BenefitAdsSubscriber$inboundSchema.and(z.object({ type: z.literal("ads") }).transform((v) => ({
            type: v.type,
        }))),
        benefitcustomsubscriber_js_1.BenefitCustomSubscriber$inboundSchema.and(z.object({ type: z.literal("custom") }).transform((v) => ({
            type: v.type,
        }))),
        benefitlicensekeyssubscriber_js_1.BenefitLicenseKeysSubscriber$inboundSchema.and(z.object({ type: z.literal("license_keys") }).transform((v) => ({
            type: v.type,
        }))),
    ])),
    pagination: pagination_js_1.Pagination$inboundSchema,
});
/** @internal */
exports.ListResourceUserBenefit$outboundSchema = z.object({
    items: z.array(z.union([
        benefitarticlessubscriber_js_1.BenefitArticlesSubscriber$outboundSchema.and(z.object({ type: z.literal("articles") }).transform((v) => ({
            type: v.type,
        }))),
        benefitdiscordsubscriber_js_1.BenefitDiscordSubscriber$outboundSchema.and(z.object({ type: z.literal("discord") }).transform((v) => ({
            type: v.type,
        }))),
        benefitdownloadablessubscriber_js_1.BenefitDownloadablesSubscriber$outboundSchema.and(z.object({ type: z.literal("downloadables") }).transform((v) => ({
            type: v.type,
        }))),
        benefitgithubrepositorysubscriber_js_1.BenefitGitHubRepositorySubscriber$outboundSchema.and(z.object({ type: z.literal("github_repository") }).transform((v) => ({
            type: v.type,
        }))),
        benefitadssubscriber_js_1.BenefitAdsSubscriber$outboundSchema.and(z.object({ type: z.literal("ads") }).transform((v) => ({
            type: v.type,
        }))),
        benefitcustomsubscriber_js_1.BenefitCustomSubscriber$outboundSchema.and(z.object({ type: z.literal("custom") }).transform((v) => ({
            type: v.type,
        }))),
        benefitlicensekeyssubscriber_js_1.BenefitLicenseKeysSubscriber$outboundSchema.and(z.object({ type: z.literal("license_keys") }).transform((v) => ({
            type: v.type,
        }))),
    ])),
    pagination: pagination_js_1.Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListResourceUserBenefit$;
(function (ListResourceUserBenefit$) {
    /** @deprecated use `ListResourceUserBenefit$inboundSchema` instead. */
    ListResourceUserBenefit$.inboundSchema = exports.ListResourceUserBenefit$inboundSchema;
    /** @deprecated use `ListResourceUserBenefit$outboundSchema` instead. */
    ListResourceUserBenefit$.outboundSchema = exports.ListResourceUserBenefit$outboundSchema;
})(ListResourceUserBenefit$ || (exports.ListResourceUserBenefit$ = ListResourceUserBenefit$ = {}));
