// @polar-sh/sdk/models/components/metricslimits.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricsLimits$ = exports.MetricsLimits$outboundSchema = exports.MetricsLimits$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const rfcdate_js_1 = require("../../types/rfcdate.js");
const metricsintervalslimits_js_1 = require("./metricsintervalslimits.js");
/** @internal */
exports.MetricsLimits$inboundSchema = z.object({
    min_date: z.string().transform(v => new rfcdate_js_1.RFCDate(v)),
    intervals: metricsintervalslimits_js_1.MetricsIntervalsLimits$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "min_date": "minDate",
    });
});
/** @internal */
exports.MetricsLimits$outboundSchema = z.object({
    minDate: z.instanceof(rfcdate_js_1.RFCDate).transform(v => v.toString()),
    intervals: metricsintervalslimits_js_1.MetricsIntervalsLimits$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        minDate: "min_date",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MetricsLimits$;
(function (MetricsLimits$) {
    /** @deprecated use `MetricsLimits$inboundSchema` instead. */
    MetricsLimits$.inboundSchema = exports.MetricsLimits$inboundSchema;
    /** @deprecated use `MetricsLimits$outboundSchema` instead. */
    MetricsLimits$.outboundSchema = exports.MetricsLimits$outboundSchema;
})(MetricsLimits$ || (exports.MetricsLimits$ = MetricsLimits$ = {}));
