// @polar-sh/sdk/models/components/metric.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Metric$ = exports.Metric$outboundSchema = exports.Metric$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const metrictype_js_1 = require("./metrictype.js");
/** @internal */
exports.Metric$inboundSchema = z
    .object({
    slug: z.string(),
    display_name: z.string(),
    type: metrictype_js_1.MetricType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "display_name": "displayName",
    });
});
/** @internal */
exports.Metric$outboundSchema = z.object({
    slug: z.string(),
    displayName: z.string(),
    type: metrictype_js_1.MetricType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        displayName: "display_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Metric$;
(function (Metric$) {
    /** @deprecated use `Metric$inboundSchema` instead. */
    Metric$.inboundSchema = exports.Metric$inboundSchema;
    /** @deprecated use `Metric$outboundSchema` instead. */
    Metric$.outboundSchema = exports.Metric$outboundSchema;
})(Metric$ || (exports.Metric$ = Metric$ = {}));
