// @polar-sh/sdk/models/components/metricsintervalslimits.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricsIntervalsLimits$ = exports.MetricsIntervalsLimits$outboundSchema = exports.MetricsIntervalsLimits$inboundSchema = void 0;
const z = __importStar(require("zod"));
const metricsintervallimit_js_1 = require("./metricsintervallimit.js");
/** @internal */
exports.MetricsIntervalsLimits$inboundSchema = z.object({
    hour: metricsintervallimit_js_1.MetricsIntervalLimit$inboundSchema,
    day: metricsintervallimit_js_1.MetricsIntervalLimit$inboundSchema,
    week: metricsintervallimit_js_1.MetricsIntervalLimit$inboundSchema,
    month: metricsintervallimit_js_1.MetricsIntervalLimit$inboundSchema,
    year: metricsintervallimit_js_1.MetricsIntervalLimit$inboundSchema,
});
/** @internal */
exports.MetricsIntervalsLimits$outboundSchema = z.object({
    hour: metricsintervallimit_js_1.MetricsIntervalLimit$outboundSchema,
    day: metricsintervallimit_js_1.MetricsIntervalLimit$outboundSchema,
    week: metricsintervallimit_js_1.MetricsIntervalLimit$outboundSchema,
    month: metricsintervallimit_js_1.MetricsIntervalLimit$outboundSchema,
    year: metricsintervallimit_js_1.MetricsIntervalLimit$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MetricsIntervalsLimits$;
(function (MetricsIntervalsLimits$) {
    /** @deprecated use `MetricsIntervalsLimits$inboundSchema` instead. */
    MetricsIntervalsLimits$.inboundSchema = exports.MetricsIntervalsLimits$inboundSchema;
    /** @deprecated use `MetricsIntervalsLimits$outboundSchema` instead. */
    MetricsIntervalsLimits$.outboundSchema = exports.MetricsIntervalsLimits$outboundSchema;
})(MetricsIntervalsLimits$ || (exports.MetricsIntervalsLimits$ = MetricsIntervalsLimits$ = {}));
