// @polar-sh/sdk/models/components/onev11oauth21tokenpostxcomponentsrefreshtokenrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$ = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$outboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$inboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$ = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$outboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$inboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType = {
    RefreshToken: "refresh_token",
};
/** @internal */
exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$inboundSchema = z.nativeEnum(exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType);
/** @internal */
exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$outboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$;
(function (Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$) {
    /** @deprecated use `Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$inboundSchema` instead. */
    Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$.inboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$inboundSchema;
    /** @deprecated use `Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$outboundSchema` instead. */
    Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$.outboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$outboundSchema;
})(Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$ || (exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$ = Onev11oauth21tokenPostXComponentsRefreshTokenRequestGrantType$ = {}));
/** @internal */
exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$inboundSchema = z.object({
    grant_type: z.literal("refresh_token").optional(),
    client_id: z.string(),
    client_secret: z.string(),
    refresh_token: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "grant_type": "grantType",
        "client_id": "clientId",
        "client_secret": "clientSecret",
        "refresh_token": "refreshToken",
    });
});
/** @internal */
exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$outboundSchema = z.object({
    grantType: z.literal("refresh_token").default("refresh_token"),
    clientId: z.string(),
    clientSecret: z.string(),
    refreshToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        grantType: "grant_type",
        clientId: "client_id",
        clientSecret: "client_secret",
        refreshToken: "refresh_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Onev11oauth21tokenPostXComponentsRefreshTokenRequest$;
(function (Onev11oauth21tokenPostXComponentsRefreshTokenRequest$) {
    /** @deprecated use `Onev11oauth21tokenPostXComponentsRefreshTokenRequest$inboundSchema` instead. */
    Onev11oauth21tokenPostXComponentsRefreshTokenRequest$.inboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$inboundSchema;
    /** @deprecated use `Onev11oauth21tokenPostXComponentsRefreshTokenRequest$outboundSchema` instead. */
    Onev11oauth21tokenPostXComponentsRefreshTokenRequest$.outboundSchema = exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$outboundSchema;
})(Onev11oauth21tokenPostXComponentsRefreshTokenRequest$ || (exports.Onev11oauth21tokenPostXComponentsRefreshTokenRequest$ = Onev11oauth21tokenPostXComponentsRefreshTokenRequest$ = {}));
