// @polar-sh/sdk/models/components/onev11oauth21tokenpostxcomponentsauthorizationcodetokenrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$ = exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$outboundSchema = exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$inboundSchema = exports.GrantType$ = exports.GrantType$outboundSchema = exports.GrantType$inboundSchema = exports.GrantType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.GrantType = {
    AuthorizationCode: "authorization_code",
};
/** @internal */
exports.GrantType$inboundSchema = z
    .nativeEnum(exports.GrantType);
/** @internal */
exports.GrantType$outboundSchema = exports.GrantType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GrantType$;
(function (GrantType$) {
    /** @deprecated use `GrantType$inboundSchema` instead. */
    GrantType$.inboundSchema = exports.GrantType$inboundSchema;
    /** @deprecated use `GrantType$outboundSchema` instead. */
    GrantType$.outboundSchema = exports.GrantType$outboundSchema;
})(GrantType$ || (exports.GrantType$ = GrantType$ = {}));
/** @internal */
exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$inboundSchema = z.object({
    grant_type: z.literal("authorization_code").optional(),
    client_id: z.string(),
    client_secret: z.string(),
    code: z.string(),
    redirect_uri: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "grant_type": "grantType",
        "client_id": "clientId",
        "client_secret": "clientSecret",
        "redirect_uri": "redirectUri",
    });
});
/** @internal */
exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$outboundSchema = z.object({
    grantType: z.literal("authorization_code").default("authorization_code"),
    clientId: z.string(),
    clientSecret: z.string(),
    code: z.string(),
    redirectUri: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        grantType: "grant_type",
        clientId: "client_id",
        clientSecret: "client_secret",
        redirectUri: "redirect_uri",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$;
(function (Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$) {
    /** @deprecated use `Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$inboundSchema` instead. */
    Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$.inboundSchema = exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$inboundSchema;
    /** @deprecated use `Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$outboundSchema` instead. */
    Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$.outboundSchema = exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$outboundSchema;
})(Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$ || (exports.Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$ = Onev11oauth21tokenPostXComponentsAuthorizationCodeTokenRequest$ = {}));
