// @polar-sh/sdk/models/components/organization.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization$ = exports.Organization$outboundSchema = exports.Organization$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const organizationfeaturesettings_js_1 = require("./organizationfeaturesettings.js");
const organizationprofilesettings_js_1 = require("./organizationprofilesettings.js");
/** @internal */
exports.Organization$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatar_url: z.nullable(z.string()),
    bio: z.nullable(z.string()),
    company: z.nullable(z.string()),
    blog: z.nullable(z.string()),
    location: z.nullable(z.string()),
    email: z.nullable(z.string()),
    twitter_username: z.nullable(z.string()),
    pledge_minimum_amount: z.number().int(),
    pledge_badge_show_amount: z.boolean(),
    default_upfront_split_to_contributors: z.nullable(z.number().int()),
    donations_enabled: z.boolean(),
    profile_settings: z.nullable(organizationprofilesettings_js_1.OrganizationProfileSettings$inboundSchema),
    feature_settings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "avatar_url": "avatarUrl",
        "twitter_username": "twitterUsername",
        "pledge_minimum_amount": "pledgeMinimumAmount",
        "pledge_badge_show_amount": "pledgeBadgeShowAmount",
        "default_upfront_split_to_contributors": "defaultUpfrontSplitToContributors",
        "donations_enabled": "donationsEnabled",
        "profile_settings": "profileSettings",
        "feature_settings": "featureSettings",
    });
});
/** @internal */
exports.Organization$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatarUrl: z.nullable(z.string()),
    bio: z.nullable(z.string()),
    company: z.nullable(z.string()),
    blog: z.nullable(z.string()),
    location: z.nullable(z.string()),
    email: z.nullable(z.string()),
    twitterUsername: z.nullable(z.string()),
    pledgeMinimumAmount: z.number().int(),
    pledgeBadgeShowAmount: z.boolean(),
    defaultUpfrontSplitToContributors: z.nullable(z.number().int()),
    donationsEnabled: z.boolean(),
    profileSettings: z.nullable(organizationprofilesettings_js_1.OrganizationProfileSettings$outboundSchema),
    featureSettings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        avatarUrl: "avatar_url",
        twitterUsername: "twitter_username",
        pledgeMinimumAmount: "pledge_minimum_amount",
        pledgeBadgeShowAmount: "pledge_badge_show_amount",
        defaultUpfrontSplitToContributors: "default_upfront_split_to_contributors",
        donationsEnabled: "donations_enabled",
        profileSettings: "profile_settings",
        featureSettings: "feature_settings",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Organization$;
(function (Organization$) {
    /** @deprecated use `Organization$inboundSchema` instead. */
    Organization$.inboundSchema = exports.Organization$inboundSchema;
    /** @deprecated use `Organization$outboundSchema` instead. */
    Organization$.outboundSchema = exports.Organization$outboundSchema;
})(Organization$ || (exports.Organization$ = Organization$ = {}));
