// @polar-sh/sdk/models/components/organizationupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUpdate$ = exports.OrganizationUpdate$outboundSchema = exports.OrganizationUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const organizationfeaturesettings_js_1 = require("./organizationfeaturesettings.js");
const organizationprofilesettings_js_1 = require("./organizationprofilesettings.js");
/** @internal */
exports.OrganizationUpdate$inboundSchema = z.object({
    name: z.nullable(z.string()).optional(),
    avatar_url: z.nullable(z.string()).optional(),
    default_upfront_split_to_contributors: z.nullable(z.number().int())
        .optional(),
    pledge_badge_show_amount: z.boolean().default(false),
    billing_email: z.nullable(z.string()).optional(),
    default_badge_custom_content: z.nullable(z.string()).optional(),
    pledge_minimum_amount: z.number().int().default(2000),
    total_monthly_spending_limit: z.nullable(z.number().int()).optional(),
    per_user_monthly_spending_limit: z.nullable(z.number().int()).optional(),
    donations_enabled: z.boolean().default(false),
    profile_settings: z.nullable(organizationprofilesettings_js_1.OrganizationProfileSettings$inboundSchema)
        .optional(),
    feature_settings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "avatar_url": "avatarUrl",
        "default_upfront_split_to_contributors": "defaultUpfrontSplitToContributors",
        "pledge_badge_show_amount": "pledgeBadgeShowAmount",
        "billing_email": "billingEmail",
        "default_badge_custom_content": "defaultBadgeCustomContent",
        "pledge_minimum_amount": "pledgeMinimumAmount",
        "total_monthly_spending_limit": "totalMonthlySpendingLimit",
        "per_user_monthly_spending_limit": "perUserMonthlySpendingLimit",
        "donations_enabled": "donationsEnabled",
        "profile_settings": "profileSettings",
        "feature_settings": "featureSettings",
    });
});
/** @internal */
exports.OrganizationUpdate$outboundSchema = z.object({
    name: z.nullable(z.string()).optional(),
    avatarUrl: z.nullable(z.string()).optional(),
    defaultUpfrontSplitToContributors: z.nullable(z.number().int()).optional(),
    pledgeBadgeShowAmount: z.boolean().default(false),
    billingEmail: z.nullable(z.string()).optional(),
    defaultBadgeCustomContent: z.nullable(z.string()).optional(),
    pledgeMinimumAmount: z.number().int().default(2000),
    totalMonthlySpendingLimit: z.nullable(z.number().int()).optional(),
    perUserMonthlySpendingLimit: z.nullable(z.number().int()).optional(),
    donationsEnabled: z.boolean().default(false),
    profileSettings: z.nullable(organizationprofilesettings_js_1.OrganizationProfileSettings$outboundSchema)
        .optional(),
    featureSettings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        avatarUrl: "avatar_url",
        defaultUpfrontSplitToContributors: "default_upfront_split_to_contributors",
        pledgeBadgeShowAmount: "pledge_badge_show_amount",
        billingEmail: "billing_email",
        defaultBadgeCustomContent: "default_badge_custom_content",
        pledgeMinimumAmount: "pledge_minimum_amount",
        totalMonthlySpendingLimit: "total_monthly_spending_limit",
        perUserMonthlySpendingLimit: "per_user_monthly_spending_limit",
        donationsEnabled: "donations_enabled",
        profileSettings: "profile_settings",
        featureSettings: "feature_settings",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationUpdate$;
(function (OrganizationUpdate$) {
    /** @deprecated use `OrganizationUpdate$inboundSchema` instead. */
    OrganizationUpdate$.inboundSchema = exports.OrganizationUpdate$inboundSchema;
    /** @deprecated use `OrganizationUpdate$outboundSchema` instead. */
    OrganizationUpdate$.outboundSchema = exports.OrganizationUpdate$outboundSchema;
})(OrganizationUpdate$ || (exports.OrganizationUpdate$ = OrganizationUpdate$ = {}));
