// @polar-sh/sdk/models/components/organizationprofilesettings.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationProfileSettings$ = exports.OrganizationProfileSettings$outboundSchema = exports.OrganizationProfileSettings$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const organizationsubscribepromotesettings_js_1 = require("./organizationsubscribepromotesettings.js");
/** @internal */
exports.OrganizationProfileSettings$inboundSchema = z.object({
    enabled: z.nullable(z.boolean()).optional(),
    description: z.nullable(z.string()).optional(),
    featured_projects: z.nullable(z.array(z.string())).optional(),
    featured_organizations: z.nullable(z.array(z.string())).optional(),
    links: z.nullable(z.array(z.string())).optional(),
    subscribe: z.nullable(organizationsubscribepromotesettings_js_1.OrganizationSubscribePromoteSettings$inboundSchema)
        .optional(),
    accent_color: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "featured_projects": "featuredProjects",
        "featured_organizations": "featuredOrganizations",
        "accent_color": "accentColor",
    });
});
/** @internal */
exports.OrganizationProfileSettings$outboundSchema = z.object({
    enabled: z.nullable(z.boolean()).optional(),
    description: z.nullable(z.string()).optional(),
    featuredProjects: z.nullable(z.array(z.string())).optional(),
    featuredOrganizations: z.nullable(z.array(z.string())).optional(),
    links: z.nullable(z.array(z.string())).optional(),
    subscribe: z.nullable(organizationsubscribepromotesettings_js_1.OrganizationSubscribePromoteSettings$outboundSchema)
        .optional(),
    accentColor: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        featuredProjects: "featured_projects",
        featuredOrganizations: "featured_organizations",
        accentColor: "accent_color",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationProfileSettings$;
(function (OrganizationProfileSettings$) {
    /** @deprecated use `OrganizationProfileSettings$inboundSchema` instead. */
    OrganizationProfileSettings$.inboundSchema = exports.OrganizationProfileSettings$inboundSchema;
    /** @deprecated use `OrganizationProfileSettings$outboundSchema` instead. */
    OrganizationProfileSettings$.outboundSchema = exports.OrganizationProfileSettings$outboundSchema;
})(OrganizationProfileSettings$ || (exports.OrganizationProfileSettings$ = OrganizationProfileSettings$ = {}));
