// @polar-sh/sdk/models/components/pagination.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pagination$ = exports.Pagination$outboundSchema = exports.Pagination$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.Pagination$inboundSchema = z.object({
    total_count: z.number().int(),
    max_page: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "total_count": "totalCount",
        "max_page": "maxPage",
    });
});
/** @internal */
exports.Pagination$outboundSchema = z.object({
    totalCount: z.number().int(),
    maxPage: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        totalCount: "total_count",
        maxPage: "max_page",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Pagination$;
(function (Pagination$) {
    /** @deprecated use `Pagination$inboundSchema` instead. */
    Pagination$.inboundSchema = exports.Pagination$inboundSchema;
    /** @deprecated use `Pagination$outboundSchema` instead. */
    Pagination$.outboundSchema = exports.Pagination$outboundSchema;
})(Pagination$ || (exports.Pagination$ = Pagination$ = {}));
