// @polar-sh/sdk/models/components/pledgestate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PledgeState$ = exports.PledgeState$outboundSchema = exports.PledgeState$inboundSchema = exports.PledgeState = void 0;
const z = __importStar(require("zod"));
exports.PledgeState = {
    Initiated: "initiated",
    Created: "created",
    Pending: "pending",
    Refunded: "refunded",
    Disputed: "disputed",
    ChargeDisputed: "charge_disputed",
    Cancelled: "cancelled",
};
/** @internal */
exports.PledgeState$inboundSchema = z
    .nativeEnum(exports.PledgeState);
/** @internal */
exports.PledgeState$outboundSchema = exports.PledgeState$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PledgeState$;
(function (PledgeState$) {
    /** @deprecated use `PledgeState$inboundSchema` instead. */
    PledgeState$.inboundSchema = exports.PledgeState$inboundSchema;
    /** @deprecated use `PledgeState$outboundSchema` instead. */
    PledgeState$.outboundSchema = exports.PledgeState$outboundSchema;
})(PledgeState$ || (exports.PledgeState$ = PledgeState$ = {}));
