// @polar-sh/sdk/models/components/polarcheckoutlegacyschemascheckoutcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolarCheckoutLegacySchemasCheckoutCreate$ = exports.PolarCheckoutLegacySchemasCheckoutCreate$outboundSchema = exports.PolarCheckoutLegacySchemasCheckoutCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.PolarCheckoutLegacySchemasCheckoutCreate$inboundSchema = z.object({
    product_price_id: z.string(),
    success_url: z.string(),
    customer_email: z.nullable(z.string()).optional(),
    subscription_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "product_price_id": "productPriceId",
        "success_url": "successUrl",
        "customer_email": "customerEmail",
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
exports.PolarCheckoutLegacySchemasCheckoutCreate$outboundSchema = z.object({
    productPriceId: z.string(),
    successUrl: z.string(),
    customerEmail: z.nullable(z.string()).optional(),
    subscriptionId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        productPriceId: "product_price_id",
        successUrl: "success_url",
        customerEmail: "customer_email",
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PolarCheckoutLegacySchemasCheckoutCreate$;
(function (PolarCheckoutLegacySchemasCheckoutCreate$) {
    /** @deprecated use `PolarCheckoutLegacySchemasCheckoutCreate$inboundSchema` instead. */
    PolarCheckoutLegacySchemasCheckoutCreate$.inboundSchema = exports.PolarCheckoutLegacySchemasCheckoutCreate$inboundSchema;
    /** @deprecated use `PolarCheckoutLegacySchemasCheckoutCreate$outboundSchema` instead. */
    PolarCheckoutLegacySchemasCheckoutCreate$.outboundSchema = exports.PolarCheckoutLegacySchemasCheckoutCreate$outboundSchema;
})(PolarCheckoutLegacySchemasCheckoutCreate$ || (exports.PolarCheckoutLegacySchemasCheckoutCreate$ = PolarCheckoutLegacySchemasCheckoutCreate$ = {}));
