// @polar-sh/sdk/models/components/polarcheckoutlegacyschemascheckout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolarCheckoutLegacySchemasCheckout$ = exports.PolarCheckoutLegacySchemasCheckout$outboundSchema = exports.PolarCheckoutLegacySchemasCheckout$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const product_js_1 = require("./product.js");
const productprice_js_1 = require("./productprice.js");
/** @internal */
exports.PolarCheckoutLegacySchemasCheckout$inboundSchema = z.object({
    id: z.string(),
    url: z.nullable(z.string()).optional(),
    customer_email: z.nullable(z.string()),
    customer_name: z.nullable(z.string()),
    product: product_js_1.Product$inboundSchema,
    product_price: productprice_js_1.ProductPrice$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_email": "customerEmail",
        "customer_name": "customerName",
        "product_price": "productPrice",
    });
});
/** @internal */
exports.PolarCheckoutLegacySchemasCheckout$outboundSchema = z.object({
    id: z.string(),
    url: z.nullable(z.string()).optional(),
    customerEmail: z.nullable(z.string()),
    customerName: z.nullable(z.string()),
    product: product_js_1.Product$outboundSchema,
    productPrice: productprice_js_1.ProductPrice$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerEmail: "customer_email",
        customerName: "customer_name",
        productPrice: "product_price",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PolarCheckoutLegacySchemasCheckout$;
(function (PolarCheckoutLegacySchemasCheckout$) {
    /** @deprecated use `PolarCheckoutLegacySchemasCheckout$inboundSchema` instead. */
    PolarCheckoutLegacySchemasCheckout$.inboundSchema = exports.PolarCheckoutLegacySchemasCheckout$inboundSchema;
    /** @deprecated use `PolarCheckoutLegacySchemasCheckout$outboundSchema` instead. */
    PolarCheckoutLegacySchemasCheckout$.outboundSchema = exports.PolarCheckoutLegacySchemasCheckout$outboundSchema;
})(PolarCheckoutLegacySchemasCheckout$ || (exports.PolarCheckoutLegacySchemasCheckout$ = PolarCheckoutLegacySchemasCheckout$ = {}));
