// @polar-sh/sdk/models/components/productpriceonetimefree.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceOneTimeFree$ = exports.ProductPriceOneTimeFree$outboundSchema = exports.ProductPriceOneTimeFree$inboundSchema = exports.ProductPriceOneTimeFreeType$ = exports.ProductPriceOneTimeFreeType$outboundSchema = exports.ProductPriceOneTimeFreeType$inboundSchema = exports.ProductPriceOneTimeFreeAmountType$ = exports.ProductPriceOneTimeFreeAmountType$outboundSchema = exports.ProductPriceOneTimeFreeAmountType$inboundSchema = exports.ProductPriceOneTimeFreeType = exports.ProductPriceOneTimeFreeAmountType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.ProductPriceOneTimeFreeAmountType = {
    Free: "free",
};
/**
 * The type of the price.
 */
exports.ProductPriceOneTimeFreeType = {
    OneTime: "one_time",
};
/** @internal */
exports.ProductPriceOneTimeFreeAmountType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeFreeAmountType);
/** @internal */
exports.ProductPriceOneTimeFreeAmountType$outboundSchema = exports.ProductPriceOneTimeFreeAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFreeAmountType$;
(function (ProductPriceOneTimeFreeAmountType$) {
    /** @deprecated use `ProductPriceOneTimeFreeAmountType$inboundSchema` instead. */
    ProductPriceOneTimeFreeAmountType$.inboundSchema = exports.ProductPriceOneTimeFreeAmountType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFreeAmountType$outboundSchema` instead. */
    ProductPriceOneTimeFreeAmountType$.outboundSchema = exports.ProductPriceOneTimeFreeAmountType$outboundSchema;
})(ProductPriceOneTimeFreeAmountType$ || (exports.ProductPriceOneTimeFreeAmountType$ = ProductPriceOneTimeFreeAmountType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFreeType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeFreeType);
/** @internal */
exports.ProductPriceOneTimeFreeType$outboundSchema = exports.ProductPriceOneTimeFreeType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFreeType$;
(function (ProductPriceOneTimeFreeType$) {
    /** @deprecated use `ProductPriceOneTimeFreeType$inboundSchema` instead. */
    ProductPriceOneTimeFreeType$.inboundSchema = exports.ProductPriceOneTimeFreeType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFreeType$outboundSchema` instead. */
    ProductPriceOneTimeFreeType$.outboundSchema = exports.ProductPriceOneTimeFreeType$outboundSchema;
})(ProductPriceOneTimeFreeType$ || (exports.ProductPriceOneTimeFreeType$ = ProductPriceOneTimeFreeType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFree$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("free").optional(),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: z.literal("one_time").optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
    });
});
/** @internal */
exports.ProductPriceOneTimeFree$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("free").default("free"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: z.literal("one_time").default("one_time"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFree$;
(function (ProductPriceOneTimeFree$) {
    /** @deprecated use `ProductPriceOneTimeFree$inboundSchema` instead. */
    ProductPriceOneTimeFree$.inboundSchema = exports.ProductPriceOneTimeFree$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFree$outboundSchema` instead. */
    ProductPriceOneTimeFree$.outboundSchema = exports.ProductPriceOneTimeFree$outboundSchema;
})(ProductPriceOneTimeFree$ || (exports.ProductPriceOneTimeFree$ = ProductPriceOneTimeFree$ = {}));
