// @polar-sh/sdk/models/components/productpriceonetimefreecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceOneTimeFreeCreate$ = exports.ProductPriceOneTimeFreeCreate$outboundSchema = exports.ProductPriceOneTimeFreeCreate$inboundSchema = exports.ProductPriceOneTimeFreeCreateAmountType$ = exports.ProductPriceOneTimeFreeCreateAmountType$outboundSchema = exports.ProductPriceOneTimeFreeCreateAmountType$inboundSchema = exports.ProductPriceOneTimeFreeCreateType$ = exports.ProductPriceOneTimeFreeCreateType$outboundSchema = exports.ProductPriceOneTimeFreeCreateType$inboundSchema = exports.ProductPriceOneTimeFreeCreateAmountType = exports.ProductPriceOneTimeFreeCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.ProductPriceOneTimeFreeCreateType = {
    OneTime: "one_time",
};
exports.ProductPriceOneTimeFreeCreateAmountType = {
    Free: "free",
};
/** @internal */
exports.ProductPriceOneTimeFreeCreateType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeFreeCreateType);
/** @internal */
exports.ProductPriceOneTimeFreeCreateType$outboundSchema = exports.ProductPriceOneTimeFreeCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFreeCreateType$;
(function (ProductPriceOneTimeFreeCreateType$) {
    /** @deprecated use `ProductPriceOneTimeFreeCreateType$inboundSchema` instead. */
    ProductPriceOneTimeFreeCreateType$.inboundSchema = exports.ProductPriceOneTimeFreeCreateType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFreeCreateType$outboundSchema` instead. */
    ProductPriceOneTimeFreeCreateType$.outboundSchema = exports.ProductPriceOneTimeFreeCreateType$outboundSchema;
})(ProductPriceOneTimeFreeCreateType$ || (exports.ProductPriceOneTimeFreeCreateType$ = ProductPriceOneTimeFreeCreateType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFreeCreateAmountType$inboundSchema = z
    .nativeEnum(exports.ProductPriceOneTimeFreeCreateAmountType);
/** @internal */
exports.ProductPriceOneTimeFreeCreateAmountType$outboundSchema = exports.ProductPriceOneTimeFreeCreateAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFreeCreateAmountType$;
(function (ProductPriceOneTimeFreeCreateAmountType$) {
    /** @deprecated use `ProductPriceOneTimeFreeCreateAmountType$inboundSchema` instead. */
    ProductPriceOneTimeFreeCreateAmountType$.inboundSchema = exports.ProductPriceOneTimeFreeCreateAmountType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFreeCreateAmountType$outboundSchema` instead. */
    ProductPriceOneTimeFreeCreateAmountType$.outboundSchema = exports.ProductPriceOneTimeFreeCreateAmountType$outboundSchema;
})(ProductPriceOneTimeFreeCreateAmountType$ || (exports.ProductPriceOneTimeFreeCreateAmountType$ = ProductPriceOneTimeFreeCreateAmountType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFreeCreate$inboundSchema = z.object({
    type: z.literal("one_time").optional(),
    amount_type: z.literal("free").optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
    });
});
/** @internal */
exports.ProductPriceOneTimeFreeCreate$outboundSchema = z.object({
    type: z.literal("one_time").default("one_time"),
    amountType: z.literal("free").default("free"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFreeCreate$;
(function (ProductPriceOneTimeFreeCreate$) {
    /** @deprecated use `ProductPriceOneTimeFreeCreate$inboundSchema` instead. */
    ProductPriceOneTimeFreeCreate$.inboundSchema = exports.ProductPriceOneTimeFreeCreate$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFreeCreate$outboundSchema` instead. */
    ProductPriceOneTimeFreeCreate$.outboundSchema = exports.ProductPriceOneTimeFreeCreate$outboundSchema;
})(ProductPriceOneTimeFreeCreate$ || (exports.ProductPriceOneTimeFreeCreate$ = ProductPriceOneTimeFreeCreate$ = {}));
