// @polar-sh/sdk/models/components/productpricerecurringfreecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceRecurringFreeCreate$ = exports.ProductPriceRecurringFreeCreate$outboundSchema = exports.ProductPriceRecurringFreeCreate$inboundSchema = exports.ProductPriceRecurringFreeCreateAmountType$ = exports.ProductPriceRecurringFreeCreateAmountType$outboundSchema = exports.ProductPriceRecurringFreeCreateAmountType$inboundSchema = exports.ProductPriceRecurringFreeCreateType$ = exports.ProductPriceRecurringFreeCreateType$outboundSchema = exports.ProductPriceRecurringFreeCreateType$inboundSchema = exports.ProductPriceRecurringFreeCreateAmountType = exports.ProductPriceRecurringFreeCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
exports.ProductPriceRecurringFreeCreateType = {
    Recurring: "recurring",
};
exports.ProductPriceRecurringFreeCreateAmountType = {
    Free: "free",
};
/** @internal */
exports.ProductPriceRecurringFreeCreateType$inboundSchema = z.nativeEnum(exports.ProductPriceRecurringFreeCreateType);
/** @internal */
exports.ProductPriceRecurringFreeCreateType$outboundSchema = exports.ProductPriceRecurringFreeCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFreeCreateType$;
(function (ProductPriceRecurringFreeCreateType$) {
    /** @deprecated use `ProductPriceRecurringFreeCreateType$inboundSchema` instead. */
    ProductPriceRecurringFreeCreateType$.inboundSchema = exports.ProductPriceRecurringFreeCreateType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFreeCreateType$outboundSchema` instead. */
    ProductPriceRecurringFreeCreateType$.outboundSchema = exports.ProductPriceRecurringFreeCreateType$outboundSchema;
})(ProductPriceRecurringFreeCreateType$ || (exports.ProductPriceRecurringFreeCreateType$ = ProductPriceRecurringFreeCreateType$ = {}));
/** @internal */
exports.ProductPriceRecurringFreeCreateAmountType$inboundSchema = z
    .nativeEnum(exports.ProductPriceRecurringFreeCreateAmountType);
/** @internal */
exports.ProductPriceRecurringFreeCreateAmountType$outboundSchema = exports.ProductPriceRecurringFreeCreateAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFreeCreateAmountType$;
(function (ProductPriceRecurringFreeCreateAmountType$) {
    /** @deprecated use `ProductPriceRecurringFreeCreateAmountType$inboundSchema` instead. */
    ProductPriceRecurringFreeCreateAmountType$.inboundSchema = exports.ProductPriceRecurringFreeCreateAmountType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFreeCreateAmountType$outboundSchema` instead. */
    ProductPriceRecurringFreeCreateAmountType$.outboundSchema = exports.ProductPriceRecurringFreeCreateAmountType$outboundSchema;
})(ProductPriceRecurringFreeCreateAmountType$ || (exports.ProductPriceRecurringFreeCreateAmountType$ = ProductPriceRecurringFreeCreateAmountType$ = {}));
/** @internal */
exports.ProductPriceRecurringFreeCreate$inboundSchema = z.object({
    type: z.literal("recurring").optional(),
    amount_type: z.literal("free").optional(),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
exports.ProductPriceRecurringFreeCreate$outboundSchema = z.object({
    type: z.literal("recurring").default("recurring"),
    amountType: z.literal("free").default("free"),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFreeCreate$;
(function (ProductPriceRecurringFreeCreate$) {
    /** @deprecated use `ProductPriceRecurringFreeCreate$inboundSchema` instead. */
    ProductPriceRecurringFreeCreate$.inboundSchema = exports.ProductPriceRecurringFreeCreate$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFreeCreate$outboundSchema` instead. */
    ProductPriceRecurringFreeCreate$.outboundSchema = exports.ProductPriceRecurringFreeCreate$outboundSchema;
})(ProductPriceRecurringFreeCreate$ || (exports.ProductPriceRecurringFreeCreate$ = ProductPriceRecurringFreeCreate$ = {}));
