// @polar-sh/sdk/models/components/product.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductInput$ = exports.ProductInput$outboundSchema = exports.ProductInput$inboundSchema = exports.Product$ = exports.Product$outboundSchema = exports.Product$inboundSchema = exports.ProductBenefits$ = exports.ProductBenefits$outboundSchema = exports.ProductBenefits$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitarticles_js_1 = require("./benefitarticles.js");
const benefitbase_js_1 = require("./benefitbase.js");
const productmediafileread_js_1 = require("./productmediafileread.js");
const productprice_js_1 = require("./productprice.js");
/** @internal */
exports.ProductBenefits$inboundSchema = z.union([benefitbase_js_1.BenefitBase$inboundSchema, benefitarticles_js_1.BenefitArticles$inboundSchema]);
/** @internal */
exports.ProductBenefits$outboundSchema = z.union([benefitbase_js_1.BenefitBase$outboundSchema, benefitarticles_js_1.BenefitArticles$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductBenefits$;
(function (ProductBenefits$) {
    /** @deprecated use `ProductBenefits$inboundSchema` instead. */
    ProductBenefits$.inboundSchema = exports.ProductBenefits$inboundSchema;
    /** @deprecated use `ProductBenefits$outboundSchema` instead. */
    ProductBenefits$.outboundSchema = exports.ProductBenefits$outboundSchema;
})(ProductBenefits$ || (exports.ProductBenefits$ = ProductBenefits$ = {}));
/** @internal */
exports.Product$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$inboundSchema),
    benefits: z.array(z.union([benefitbase_js_1.BenefitBase$inboundSchema, benefitarticles_js_1.BenefitArticles$inboundSchema])),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.Product$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$outboundSchema),
    benefits: z.array(z.union([benefitbase_js_1.BenefitBase$outboundSchema, benefitarticles_js_1.BenefitArticles$outboundSchema])),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Product$;
(function (Product$) {
    /** @deprecated use `Product$inboundSchema` instead. */
    Product$.inboundSchema = exports.Product$inboundSchema;
    /** @deprecated use `Product$outboundSchema` instead. */
    Product$.outboundSchema = exports.Product$outboundSchema;
})(Product$ || (exports.Product$ = Product$ = {}));
/** @internal */
exports.ProductInput$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$inboundSchema),
    benefits: z.array(z.union([benefitbase_js_1.BenefitBase$inboundSchema, benefitarticles_js_1.BenefitArticles$inboundSchema])),
    medias: z.array(productmediafileread_js_1.ProductMediaFileReadInput$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ProductInput$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$outboundSchema),
    benefits: z.array(z.union([benefitbase_js_1.BenefitBase$outboundSchema, benefitarticles_js_1.BenefitArticles$outboundSchema])),
    medias: z.array(productmediafileread_js_1.ProductMediaFileReadInput$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductInput$;
(function (ProductInput$) {
    /** @deprecated use `ProductInput$inboundSchema` instead. */
    ProductInput$.inboundSchema = exports.ProductInput$inboundSchema;
    /** @deprecated use `ProductInput$outboundSchema` instead. */
    ProductInput$.outboundSchema = exports.ProductInput$outboundSchema;
})(ProductInput$ || (exports.ProductInput$ = ProductInput$ = {}));
