// @polar-sh/sdk/models/components/productpriceonetimefixed.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceOneTimeFixed$ = exports.ProductPriceOneTimeFixed$outboundSchema = exports.ProductPriceOneTimeFixed$inboundSchema = exports.ProductPriceOneTimeFixedType$ = exports.ProductPriceOneTimeFixedType$outboundSchema = exports.ProductPriceOneTimeFixedType$inboundSchema = exports.ProductPriceOneTimeFixedAmountType$ = exports.ProductPriceOneTimeFixedAmountType$outboundSchema = exports.ProductPriceOneTimeFixedAmountType$inboundSchema = exports.ProductPriceOneTimeFixedType = exports.ProductPriceOneTimeFixedAmountType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.ProductPriceOneTimeFixedAmountType = {
    Fixed: "fixed",
};
/**
 * The type of the price.
 */
exports.ProductPriceOneTimeFixedType = {
    OneTime: "one_time",
};
/** @internal */
exports.ProductPriceOneTimeFixedAmountType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeFixedAmountType);
/** @internal */
exports.ProductPriceOneTimeFixedAmountType$outboundSchema = exports.ProductPriceOneTimeFixedAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFixedAmountType$;
(function (ProductPriceOneTimeFixedAmountType$) {
    /** @deprecated use `ProductPriceOneTimeFixedAmountType$inboundSchema` instead. */
    ProductPriceOneTimeFixedAmountType$.inboundSchema = exports.ProductPriceOneTimeFixedAmountType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFixedAmountType$outboundSchema` instead. */
    ProductPriceOneTimeFixedAmountType$.outboundSchema = exports.ProductPriceOneTimeFixedAmountType$outboundSchema;
})(ProductPriceOneTimeFixedAmountType$ || (exports.ProductPriceOneTimeFixedAmountType$ = ProductPriceOneTimeFixedAmountType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFixedType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeFixedType);
/** @internal */
exports.ProductPriceOneTimeFixedType$outboundSchema = exports.ProductPriceOneTimeFixedType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFixedType$;
(function (ProductPriceOneTimeFixedType$) {
    /** @deprecated use `ProductPriceOneTimeFixedType$inboundSchema` instead. */
    ProductPriceOneTimeFixedType$.inboundSchema = exports.ProductPriceOneTimeFixedType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFixedType$outboundSchema` instead. */
    ProductPriceOneTimeFixedType$.outboundSchema = exports.ProductPriceOneTimeFixedType$outboundSchema;
})(ProductPriceOneTimeFixedType$ || (exports.ProductPriceOneTimeFixedType$ = ProductPriceOneTimeFixedType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFixed$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("fixed").optional(),
    is_archived: z.boolean(),
    product_id: z.string(),
    price_currency: z.string(),
    price_amount: z.number().int(),
    type: z.literal("one_time").optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "price_currency": "priceCurrency",
        "price_amount": "priceAmount",
    });
});
/** @internal */
exports.ProductPriceOneTimeFixed$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("fixed").default("fixed"),
    isArchived: z.boolean(),
    productId: z.string(),
    priceCurrency: z.string(),
    priceAmount: z.number().int(),
    type: z.literal("one_time").default("one_time"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        priceCurrency: "price_currency",
        priceAmount: "price_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFixed$;
(function (ProductPriceOneTimeFixed$) {
    /** @deprecated use `ProductPriceOneTimeFixed$inboundSchema` instead. */
    ProductPriceOneTimeFixed$.inboundSchema = exports.ProductPriceOneTimeFixed$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFixed$outboundSchema` instead. */
    ProductPriceOneTimeFixed$.outboundSchema = exports.ProductPriceOneTimeFixed$outboundSchema;
})(ProductPriceOneTimeFixed$ || (exports.ProductPriceOneTimeFixed$ = ProductPriceOneTimeFixed$ = {}));
