// @polar-sh/sdk/models/components/productpricerecurringcustom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceRecurringCustom$ = exports.ProductPriceRecurringCustom$outboundSchema = exports.ProductPriceRecurringCustom$inboundSchema = exports.ProductPriceRecurringCustomType$ = exports.ProductPriceRecurringCustomType$outboundSchema = exports.ProductPriceRecurringCustomType$inboundSchema = exports.ProductPriceRecurringCustomAmountType$ = exports.ProductPriceRecurringCustomAmountType$outboundSchema = exports.ProductPriceRecurringCustomAmountType$inboundSchema = exports.ProductPriceRecurringCustomType = exports.ProductPriceRecurringCustomAmountType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
exports.ProductPriceRecurringCustomAmountType = {
    Custom: "custom",
};
/**
 * The type of the price.
 */
exports.ProductPriceRecurringCustomType = {
    Recurring: "recurring",
};
/** @internal */
exports.ProductPriceRecurringCustomAmountType$inboundSchema = z.nativeEnum(exports.ProductPriceRecurringCustomAmountType);
/** @internal */
exports.ProductPriceRecurringCustomAmountType$outboundSchema = exports.ProductPriceRecurringCustomAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringCustomAmountType$;
(function (ProductPriceRecurringCustomAmountType$) {
    /** @deprecated use `ProductPriceRecurringCustomAmountType$inboundSchema` instead. */
    ProductPriceRecurringCustomAmountType$.inboundSchema = exports.ProductPriceRecurringCustomAmountType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringCustomAmountType$outboundSchema` instead. */
    ProductPriceRecurringCustomAmountType$.outboundSchema = exports.ProductPriceRecurringCustomAmountType$outboundSchema;
})(ProductPriceRecurringCustomAmountType$ || (exports.ProductPriceRecurringCustomAmountType$ = ProductPriceRecurringCustomAmountType$ = {}));
/** @internal */
exports.ProductPriceRecurringCustomType$inboundSchema = z.nativeEnum(exports.ProductPriceRecurringCustomType);
/** @internal */
exports.ProductPriceRecurringCustomType$outboundSchema = exports.ProductPriceRecurringCustomType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringCustomType$;
(function (ProductPriceRecurringCustomType$) {
    /** @deprecated use `ProductPriceRecurringCustomType$inboundSchema` instead. */
    ProductPriceRecurringCustomType$.inboundSchema = exports.ProductPriceRecurringCustomType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringCustomType$outboundSchema` instead. */
    ProductPriceRecurringCustomType$.outboundSchema = exports.ProductPriceRecurringCustomType$outboundSchema;
})(ProductPriceRecurringCustomType$ || (exports.ProductPriceRecurringCustomType$ = ProductPriceRecurringCustomType$ = {}));
/** @internal */
exports.ProductPriceRecurringCustom$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("custom").optional(),
    is_archived: z.boolean(),
    product_id: z.string(),
    price_currency: z.string(),
    minimum_amount: z.nullable(z.number().int()),
    maximum_amount: z.nullable(z.number().int()),
    preset_amount: z.nullable(z.number().int()),
    type: z.literal("recurring").optional(),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
exports.ProductPriceRecurringCustom$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("custom").default("custom"),
    isArchived: z.boolean(),
    productId: z.string(),
    priceCurrency: z.string(),
    minimumAmount: z.nullable(z.number().int()),
    maximumAmount: z.nullable(z.number().int()),
    presetAmount: z.nullable(z.number().int()),
    type: z.literal("recurring").default("recurring"),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringCustom$;
(function (ProductPriceRecurringCustom$) {
    /** @deprecated use `ProductPriceRecurringCustom$inboundSchema` instead. */
    ProductPriceRecurringCustom$.inboundSchema = exports.ProductPriceRecurringCustom$inboundSchema;
    /** @deprecated use `ProductPriceRecurringCustom$outboundSchema` instead. */
    ProductPriceRecurringCustom$.outboundSchema = exports.ProductPriceRecurringCustom$outboundSchema;
})(ProductPriceRecurringCustom$ || (exports.ProductPriceRecurringCustom$ = ProductPriceRecurringCustom$ = {}));
