// @polar-sh/sdk/models/components/productpricerecurringfree.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceRecurringFree$ = exports.ProductPriceRecurringFree$outboundSchema = exports.ProductPriceRecurringFree$inboundSchema = exports.ProductPriceRecurringFreeType$ = exports.ProductPriceRecurringFreeType$outboundSchema = exports.ProductPriceRecurringFreeType$inboundSchema = exports.ProductPriceRecurringFreeAmountType$ = exports.ProductPriceRecurringFreeAmountType$outboundSchema = exports.ProductPriceRecurringFreeAmountType$inboundSchema = exports.ProductPriceRecurringFreeType = exports.ProductPriceRecurringFreeAmountType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
exports.ProductPriceRecurringFreeAmountType = {
    Free: "free",
};
/**
 * The type of the price.
 */
exports.ProductPriceRecurringFreeType = {
    Recurring: "recurring",
};
/** @internal */
exports.ProductPriceRecurringFreeAmountType$inboundSchema = z.nativeEnum(exports.ProductPriceRecurringFreeAmountType);
/** @internal */
exports.ProductPriceRecurringFreeAmountType$outboundSchema = exports.ProductPriceRecurringFreeAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFreeAmountType$;
(function (ProductPriceRecurringFreeAmountType$) {
    /** @deprecated use `ProductPriceRecurringFreeAmountType$inboundSchema` instead. */
    ProductPriceRecurringFreeAmountType$.inboundSchema = exports.ProductPriceRecurringFreeAmountType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFreeAmountType$outboundSchema` instead. */
    ProductPriceRecurringFreeAmountType$.outboundSchema = exports.ProductPriceRecurringFreeAmountType$outboundSchema;
})(ProductPriceRecurringFreeAmountType$ || (exports.ProductPriceRecurringFreeAmountType$ = ProductPriceRecurringFreeAmountType$ = {}));
/** @internal */
exports.ProductPriceRecurringFreeType$inboundSchema = z.nativeEnum(exports.ProductPriceRecurringFreeType);
/** @internal */
exports.ProductPriceRecurringFreeType$outboundSchema = exports.ProductPriceRecurringFreeType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFreeType$;
(function (ProductPriceRecurringFreeType$) {
    /** @deprecated use `ProductPriceRecurringFreeType$inboundSchema` instead. */
    ProductPriceRecurringFreeType$.inboundSchema = exports.ProductPriceRecurringFreeType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFreeType$outboundSchema` instead. */
    ProductPriceRecurringFreeType$.outboundSchema = exports.ProductPriceRecurringFreeType$outboundSchema;
})(ProductPriceRecurringFreeType$ || (exports.ProductPriceRecurringFreeType$ = ProductPriceRecurringFreeType$ = {}));
/** @internal */
exports.ProductPriceRecurringFree$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("free").optional(),
    is_archived: z.boolean(),
    product_id: z.string(),
    type: z.literal("recurring").optional(),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
exports.ProductPriceRecurringFree$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("free").default("free"),
    isArchived: z.boolean(),
    productId: z.string(),
    type: z.literal("recurring").default("recurring"),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFree$;
(function (ProductPriceRecurringFree$) {
    /** @deprecated use `ProductPriceRecurringFree$inboundSchema` instead. */
    ProductPriceRecurringFree$.inboundSchema = exports.ProductPriceRecurringFree$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFree$outboundSchema` instead. */
    ProductPriceRecurringFree$.outboundSchema = exports.ProductPriceRecurringFree$outboundSchema;
})(ProductPriceRecurringFree$ || (exports.ProductPriceRecurringFree$ = ProductPriceRecurringFree$ = {}));
