// @polar-sh/sdk/models/components/productpriceonetimefixedcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceOneTimeFixedCreate$ = exports.ProductPriceOneTimeFixedCreate$outboundSchema = exports.ProductPriceOneTimeFixedCreate$inboundSchema = exports.ProductPriceOneTimeFixedCreateAmountType$ = exports.ProductPriceOneTimeFixedCreateAmountType$outboundSchema = exports.ProductPriceOneTimeFixedCreateAmountType$inboundSchema = exports.ProductPriceOneTimeFixedCreateType$ = exports.ProductPriceOneTimeFixedCreateType$outboundSchema = exports.ProductPriceOneTimeFixedCreateType$inboundSchema = exports.ProductPriceOneTimeFixedCreateAmountType = exports.ProductPriceOneTimeFixedCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.ProductPriceOneTimeFixedCreateType = {
    OneTime: "one_time",
};
exports.ProductPriceOneTimeFixedCreateAmountType = {
    Fixed: "fixed",
};
/** @internal */
exports.ProductPriceOneTimeFixedCreateType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeFixedCreateType);
/** @internal */
exports.ProductPriceOneTimeFixedCreateType$outboundSchema = exports.ProductPriceOneTimeFixedCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFixedCreateType$;
(function (ProductPriceOneTimeFixedCreateType$) {
    /** @deprecated use `ProductPriceOneTimeFixedCreateType$inboundSchema` instead. */
    ProductPriceOneTimeFixedCreateType$.inboundSchema = exports.ProductPriceOneTimeFixedCreateType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFixedCreateType$outboundSchema` instead. */
    ProductPriceOneTimeFixedCreateType$.outboundSchema = exports.ProductPriceOneTimeFixedCreateType$outboundSchema;
})(ProductPriceOneTimeFixedCreateType$ || (exports.ProductPriceOneTimeFixedCreateType$ = ProductPriceOneTimeFixedCreateType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFixedCreateAmountType$inboundSchema = z
    .nativeEnum(exports.ProductPriceOneTimeFixedCreateAmountType);
/** @internal */
exports.ProductPriceOneTimeFixedCreateAmountType$outboundSchema = exports.ProductPriceOneTimeFixedCreateAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFixedCreateAmountType$;
(function (ProductPriceOneTimeFixedCreateAmountType$) {
    /** @deprecated use `ProductPriceOneTimeFixedCreateAmountType$inboundSchema` instead. */
    ProductPriceOneTimeFixedCreateAmountType$.inboundSchema = exports.ProductPriceOneTimeFixedCreateAmountType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFixedCreateAmountType$outboundSchema` instead. */
    ProductPriceOneTimeFixedCreateAmountType$.outboundSchema = exports.ProductPriceOneTimeFixedCreateAmountType$outboundSchema;
})(ProductPriceOneTimeFixedCreateAmountType$ || (exports.ProductPriceOneTimeFixedCreateAmountType$ = ProductPriceOneTimeFixedCreateAmountType$ = {}));
/** @internal */
exports.ProductPriceOneTimeFixedCreate$inboundSchema = z.object({
    type: z.literal("one_time").optional(),
    amount_type: z.literal("fixed").optional(),
    price_amount: z.number().int(),
    price_currency: z.string().default("usd"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
        "price_amount": "priceAmount",
        "price_currency": "priceCurrency",
    });
});
/** @internal */
exports.ProductPriceOneTimeFixedCreate$outboundSchema = z.object({
    type: z.literal("one_time").default("one_time"),
    amountType: z.literal("fixed").default("fixed"),
    priceAmount: z.number().int(),
    priceCurrency: z.string().default("usd"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
        priceAmount: "price_amount",
        priceCurrency: "price_currency",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeFixedCreate$;
(function (ProductPriceOneTimeFixedCreate$) {
    /** @deprecated use `ProductPriceOneTimeFixedCreate$inboundSchema` instead. */
    ProductPriceOneTimeFixedCreate$.inboundSchema = exports.ProductPriceOneTimeFixedCreate$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeFixedCreate$outboundSchema` instead. */
    ProductPriceOneTimeFixedCreate$.outboundSchema = exports.ProductPriceOneTimeFixedCreate$outboundSchema;
})(ProductPriceOneTimeFixedCreate$ || (exports.ProductPriceOneTimeFixedCreate$ = ProductPriceOneTimeFixedCreate$ = {}));
