// @polar-sh/sdk/models/components/productpriceonetimecustomcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceOneTimeCustomCreate$ = exports.ProductPriceOneTimeCustomCreate$outboundSchema = exports.ProductPriceOneTimeCustomCreate$inboundSchema = exports.ProductPriceOneTimeCustomCreateAmountType$ = exports.ProductPriceOneTimeCustomCreateAmountType$outboundSchema = exports.ProductPriceOneTimeCustomCreateAmountType$inboundSchema = exports.ProductPriceOneTimeCustomCreateType$ = exports.ProductPriceOneTimeCustomCreateType$outboundSchema = exports.ProductPriceOneTimeCustomCreateType$inboundSchema = exports.ProductPriceOneTimeCustomCreateAmountType = exports.ProductPriceOneTimeCustomCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.ProductPriceOneTimeCustomCreateType = {
    OneTime: "one_time",
};
exports.ProductPriceOneTimeCustomCreateAmountType = {
    Custom: "custom",
};
/** @internal */
exports.ProductPriceOneTimeCustomCreateType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeCustomCreateType);
/** @internal */
exports.ProductPriceOneTimeCustomCreateType$outboundSchema = exports.ProductPriceOneTimeCustomCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeCustomCreateType$;
(function (ProductPriceOneTimeCustomCreateType$) {
    /** @deprecated use `ProductPriceOneTimeCustomCreateType$inboundSchema` instead. */
    ProductPriceOneTimeCustomCreateType$.inboundSchema = exports.ProductPriceOneTimeCustomCreateType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeCustomCreateType$outboundSchema` instead. */
    ProductPriceOneTimeCustomCreateType$.outboundSchema = exports.ProductPriceOneTimeCustomCreateType$outboundSchema;
})(ProductPriceOneTimeCustomCreateType$ || (exports.ProductPriceOneTimeCustomCreateType$ = ProductPriceOneTimeCustomCreateType$ = {}));
/** @internal */
exports.ProductPriceOneTimeCustomCreateAmountType$inboundSchema = z
    .nativeEnum(exports.ProductPriceOneTimeCustomCreateAmountType);
/** @internal */
exports.ProductPriceOneTimeCustomCreateAmountType$outboundSchema = exports.ProductPriceOneTimeCustomCreateAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeCustomCreateAmountType$;
(function (ProductPriceOneTimeCustomCreateAmountType$) {
    /** @deprecated use `ProductPriceOneTimeCustomCreateAmountType$inboundSchema` instead. */
    ProductPriceOneTimeCustomCreateAmountType$.inboundSchema = exports.ProductPriceOneTimeCustomCreateAmountType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeCustomCreateAmountType$outboundSchema` instead. */
    ProductPriceOneTimeCustomCreateAmountType$.outboundSchema = exports.ProductPriceOneTimeCustomCreateAmountType$outboundSchema;
})(ProductPriceOneTimeCustomCreateAmountType$ || (exports.ProductPriceOneTimeCustomCreateAmountType$ = ProductPriceOneTimeCustomCreateAmountType$ = {}));
/** @internal */
exports.ProductPriceOneTimeCustomCreate$inboundSchema = z.object({
    type: z.literal("one_time").optional(),
    amount_type: z.literal("custom").optional(),
    price_currency: z.string().default("usd"),
    minimum_amount: z.nullable(z.number().int()).optional(),
    maximum_amount: z.nullable(z.number().int()).optional(),
    preset_amount: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
});
/** @internal */
exports.ProductPriceOneTimeCustomCreate$outboundSchema = z.object({
    type: z.literal("one_time").default("one_time"),
    amountType: z.literal("custom").default("custom"),
    priceCurrency: z.string().default("usd"),
    minimumAmount: z.nullable(z.number().int()).optional(),
    maximumAmount: z.nullable(z.number().int()).optional(),
    presetAmount: z.nullable(z.number().int()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeCustomCreate$;
(function (ProductPriceOneTimeCustomCreate$) {
    /** @deprecated use `ProductPriceOneTimeCustomCreate$inboundSchema` instead. */
    ProductPriceOneTimeCustomCreate$.inboundSchema = exports.ProductPriceOneTimeCustomCreate$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeCustomCreate$outboundSchema` instead. */
    ProductPriceOneTimeCustomCreate$.outboundSchema = exports.ProductPriceOneTimeCustomCreate$outboundSchema;
})(ProductPriceOneTimeCustomCreate$ || (exports.ProductPriceOneTimeCustomCreate$ = ProductPriceOneTimeCustomCreate$ = {}));
