// @polar-sh/sdk/models/components/productonetimecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductOneTimeCreate$ = exports.ProductOneTimeCreate$outboundSchema = exports.ProductOneTimeCreate$inboundSchema = exports.Prices$ = exports.Prices$outboundSchema = exports.Prices$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const productpriceonetimecustomcreate_js_1 = require("./productpriceonetimecustomcreate.js");
const productpriceonetimefixedcreate_js_1 = require("./productpriceonetimefixedcreate.js");
const productpriceonetimefreecreate_js_1 = require("./productpriceonetimefreecreate.js");
/** @internal */
exports.Prices$inboundSchema = z
    .union([
    productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$inboundSchema,
    productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$inboundSchema,
    productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$inboundSchema,
]);
/** @internal */
exports.Prices$outboundSchema = z.union([
    productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$outboundSchema,
    productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$outboundSchema,
    productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Prices$;
(function (Prices$) {
    /** @deprecated use `Prices$inboundSchema` instead. */
    Prices$.inboundSchema = exports.Prices$inboundSchema;
    /** @deprecated use `Prices$outboundSchema` instead. */
    Prices$.outboundSchema = exports.Prices$outboundSchema;
})(Prices$ || (exports.Prices$ = Prices$ = {}));
/** @internal */
exports.ProductOneTimeCreate$inboundSchema = z.object({
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$inboundSchema,
        productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$inboundSchema,
        productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$inboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ProductOneTimeCreate$outboundSchema = z.object({
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$outboundSchema,
        productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$outboundSchema,
        productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$outboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductOneTimeCreate$;
(function (ProductOneTimeCreate$) {
    /** @deprecated use `ProductOneTimeCreate$inboundSchema` instead. */
    ProductOneTimeCreate$.inboundSchema = exports.ProductOneTimeCreate$inboundSchema;
    /** @deprecated use `ProductOneTimeCreate$outboundSchema` instead. */
    ProductOneTimeCreate$.outboundSchema = exports.ProductOneTimeCreate$outboundSchema;
})(ProductOneTimeCreate$ || (exports.ProductOneTimeCreate$ = ProductOneTimeCreate$ = {}));
