// @polar-sh/sdk/models/components/productpriceonetimecustom.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceOneTimeCustom$ = exports.ProductPriceOneTimeCustom$outboundSchema = exports.ProductPriceOneTimeCustom$inboundSchema = exports.ProductPriceOneTimeCustomType$ = exports.ProductPriceOneTimeCustomType$outboundSchema = exports.ProductPriceOneTimeCustomType$inboundSchema = exports.ProductPriceOneTimeCustomAmountType$ = exports.ProductPriceOneTimeCustomAmountType$outboundSchema = exports.ProductPriceOneTimeCustomAmountType$inboundSchema = exports.ProductPriceOneTimeCustomType = exports.ProductPriceOneTimeCustomAmountType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.ProductPriceOneTimeCustomAmountType = {
    Custom: "custom",
};
/**
 * The type of the price.
 */
exports.ProductPriceOneTimeCustomType = {
    OneTime: "one_time",
};
/** @internal */
exports.ProductPriceOneTimeCustomAmountType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeCustomAmountType);
/** @internal */
exports.ProductPriceOneTimeCustomAmountType$outboundSchema = exports.ProductPriceOneTimeCustomAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeCustomAmountType$;
(function (ProductPriceOneTimeCustomAmountType$) {
    /** @deprecated use `ProductPriceOneTimeCustomAmountType$inboundSchema` instead. */
    ProductPriceOneTimeCustomAmountType$.inboundSchema = exports.ProductPriceOneTimeCustomAmountType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeCustomAmountType$outboundSchema` instead. */
    ProductPriceOneTimeCustomAmountType$.outboundSchema = exports.ProductPriceOneTimeCustomAmountType$outboundSchema;
})(ProductPriceOneTimeCustomAmountType$ || (exports.ProductPriceOneTimeCustomAmountType$ = ProductPriceOneTimeCustomAmountType$ = {}));
/** @internal */
exports.ProductPriceOneTimeCustomType$inboundSchema = z.nativeEnum(exports.ProductPriceOneTimeCustomType);
/** @internal */
exports.ProductPriceOneTimeCustomType$outboundSchema = exports.ProductPriceOneTimeCustomType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeCustomType$;
(function (ProductPriceOneTimeCustomType$) {
    /** @deprecated use `ProductPriceOneTimeCustomType$inboundSchema` instead. */
    ProductPriceOneTimeCustomType$.inboundSchema = exports.ProductPriceOneTimeCustomType$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeCustomType$outboundSchema` instead. */
    ProductPriceOneTimeCustomType$.outboundSchema = exports.ProductPriceOneTimeCustomType$outboundSchema;
})(ProductPriceOneTimeCustomType$ || (exports.ProductPriceOneTimeCustomType$ = ProductPriceOneTimeCustomType$ = {}));
/** @internal */
exports.ProductPriceOneTimeCustom$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("custom").optional(),
    is_archived: z.boolean(),
    product_id: z.string(),
    price_currency: z.string(),
    minimum_amount: z.nullable(z.number().int()),
    maximum_amount: z.nullable(z.number().int()),
    preset_amount: z.nullable(z.number().int()),
    type: z.literal("one_time").optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "price_currency": "priceCurrency",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
});
/** @internal */
exports.ProductPriceOneTimeCustom$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("custom").default("custom"),
    isArchived: z.boolean(),
    productId: z.string(),
    priceCurrency: z.string(),
    minimumAmount: z.nullable(z.number().int()),
    maximumAmount: z.nullable(z.number().int()),
    presetAmount: z.nullable(z.number().int()),
    type: z.literal("one_time").default("one_time"),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        priceCurrency: "price_currency",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceOneTimeCustom$;
(function (ProductPriceOneTimeCustom$) {
    /** @deprecated use `ProductPriceOneTimeCustom$inboundSchema` instead. */
    ProductPriceOneTimeCustom$.inboundSchema = exports.ProductPriceOneTimeCustom$inboundSchema;
    /** @deprecated use `ProductPriceOneTimeCustom$outboundSchema` instead. */
    ProductPriceOneTimeCustom$.outboundSchema = exports.ProductPriceOneTimeCustom$outboundSchema;
})(ProductPriceOneTimeCustom$ || (exports.ProductPriceOneTimeCustom$ = ProductPriceOneTimeCustom$ = {}));
