// @polar-sh/sdk/models/components/productpricerecurringfixed.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceRecurringFixed$ = exports.ProductPriceRecurringFixed$outboundSchema = exports.ProductPriceRecurringFixed$inboundSchema = exports.ProductPriceRecurringFixedType$ = exports.ProductPriceRecurringFixedType$outboundSchema = exports.ProductPriceRecurringFixedType$inboundSchema = exports.AmountType$ = exports.AmountType$outboundSchema = exports.AmountType$inboundSchema = exports.ProductPriceRecurringFixedType = exports.AmountType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
exports.AmountType = {
    Fixed: "fixed",
};
/**
 * The type of the price.
 */
exports.ProductPriceRecurringFixedType = {
    Recurring: "recurring",
};
/** @internal */
exports.AmountType$inboundSchema = z
    .nativeEnum(exports.AmountType);
/** @internal */
exports.AmountType$outboundSchema = exports.AmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AmountType$;
(function (AmountType$) {
    /** @deprecated use `AmountType$inboundSchema` instead. */
    AmountType$.inboundSchema = exports.AmountType$inboundSchema;
    /** @deprecated use `AmountType$outboundSchema` instead. */
    AmountType$.outboundSchema = exports.AmountType$outboundSchema;
})(AmountType$ || (exports.AmountType$ = AmountType$ = {}));
/** @internal */
exports.ProductPriceRecurringFixedType$inboundSchema = z.nativeEnum(exports.ProductPriceRecurringFixedType);
/** @internal */
exports.ProductPriceRecurringFixedType$outboundSchema = exports.ProductPriceRecurringFixedType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFixedType$;
(function (ProductPriceRecurringFixedType$) {
    /** @deprecated use `ProductPriceRecurringFixedType$inboundSchema` instead. */
    ProductPriceRecurringFixedType$.inboundSchema = exports.ProductPriceRecurringFixedType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFixedType$outboundSchema` instead. */
    ProductPriceRecurringFixedType$.outboundSchema = exports.ProductPriceRecurringFixedType$outboundSchema;
})(ProductPriceRecurringFixedType$ || (exports.ProductPriceRecurringFixedType$ = ProductPriceRecurringFixedType$ = {}));
/** @internal */
exports.ProductPriceRecurringFixed$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount_type: z.literal("fixed").optional(),
    is_archived: z.boolean(),
    product_id: z.string(),
    price_currency: z.string(),
    price_amount: z.number().int(),
    type: z.literal("recurring").optional(),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "is_archived": "isArchived",
        "product_id": "productId",
        "price_currency": "priceCurrency",
        "price_amount": "priceAmount",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
exports.ProductPriceRecurringFixed$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amountType: z.literal("fixed").default("fixed"),
    isArchived: z.boolean(),
    productId: z.string(),
    priceCurrency: z.string(),
    priceAmount: z.number().int(),
    type: z.literal("recurring").default("recurring"),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        isArchived: "is_archived",
        productId: "product_id",
        priceCurrency: "price_currency",
        priceAmount: "price_amount",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFixed$;
(function (ProductPriceRecurringFixed$) {
    /** @deprecated use `ProductPriceRecurringFixed$inboundSchema` instead. */
    ProductPriceRecurringFixed$.inboundSchema = exports.ProductPriceRecurringFixed$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFixed$outboundSchema` instead. */
    ProductPriceRecurringFixed$.outboundSchema = exports.ProductPriceRecurringFixed$outboundSchema;
})(ProductPriceRecurringFixed$ || (exports.ProductPriceRecurringFixed$ = ProductPriceRecurringFixed$ = {}));
