// @polar-sh/sdk/models/components/productpricerecurringfixedcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceRecurringFixedCreate$ = exports.ProductPriceRecurringFixedCreate$outboundSchema = exports.ProductPriceRecurringFixedCreate$inboundSchema = exports.ProductPriceRecurringFixedCreateAmountType$ = exports.ProductPriceRecurringFixedCreateAmountType$outboundSchema = exports.ProductPriceRecurringFixedCreateAmountType$inboundSchema = exports.ProductPriceRecurringFixedCreateType$ = exports.ProductPriceRecurringFixedCreateType$outboundSchema = exports.ProductPriceRecurringFixedCreateType$inboundSchema = exports.ProductPriceRecurringFixedCreateAmountType = exports.ProductPriceRecurringFixedCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
exports.ProductPriceRecurringFixedCreateType = {
    Recurring: "recurring",
};
exports.ProductPriceRecurringFixedCreateAmountType = {
    Fixed: "fixed",
};
/** @internal */
exports.ProductPriceRecurringFixedCreateType$inboundSchema = z.nativeEnum(exports.ProductPriceRecurringFixedCreateType);
/** @internal */
exports.ProductPriceRecurringFixedCreateType$outboundSchema = exports.ProductPriceRecurringFixedCreateType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFixedCreateType$;
(function (ProductPriceRecurringFixedCreateType$) {
    /** @deprecated use `ProductPriceRecurringFixedCreateType$inboundSchema` instead. */
    ProductPriceRecurringFixedCreateType$.inboundSchema = exports.ProductPriceRecurringFixedCreateType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFixedCreateType$outboundSchema` instead. */
    ProductPriceRecurringFixedCreateType$.outboundSchema = exports.ProductPriceRecurringFixedCreateType$outboundSchema;
})(ProductPriceRecurringFixedCreateType$ || (exports.ProductPriceRecurringFixedCreateType$ = ProductPriceRecurringFixedCreateType$ = {}));
/** @internal */
exports.ProductPriceRecurringFixedCreateAmountType$inboundSchema = z
    .nativeEnum(exports.ProductPriceRecurringFixedCreateAmountType);
/** @internal */
exports.ProductPriceRecurringFixedCreateAmountType$outboundSchema = exports.ProductPriceRecurringFixedCreateAmountType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFixedCreateAmountType$;
(function (ProductPriceRecurringFixedCreateAmountType$) {
    /** @deprecated use `ProductPriceRecurringFixedCreateAmountType$inboundSchema` instead. */
    ProductPriceRecurringFixedCreateAmountType$.inboundSchema = exports.ProductPriceRecurringFixedCreateAmountType$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFixedCreateAmountType$outboundSchema` instead. */
    ProductPriceRecurringFixedCreateAmountType$.outboundSchema = exports.ProductPriceRecurringFixedCreateAmountType$outboundSchema;
})(ProductPriceRecurringFixedCreateAmountType$ || (exports.ProductPriceRecurringFixedCreateAmountType$ = ProductPriceRecurringFixedCreateAmountType$ = {}));
/** @internal */
exports.ProductPriceRecurringFixedCreate$inboundSchema = z.object({
    type: z.literal("recurring").optional(),
    amount_type: z.literal("fixed").optional(),
    price_amount: z.number().int(),
    price_currency: z.string().default("usd"),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_type": "amountType",
        "price_amount": "priceAmount",
        "price_currency": "priceCurrency",
        "recurring_interval": "recurringInterval",
    });
});
/** @internal */
exports.ProductPriceRecurringFixedCreate$outboundSchema = z.object({
    type: z.literal("recurring").default("recurring"),
    amountType: z.literal("fixed").default("fixed"),
    priceAmount: z.number().int(),
    priceCurrency: z.string().default("usd"),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountType: "amount_type",
        priceAmount: "price_amount",
        priceCurrency: "price_currency",
        recurringInterval: "recurring_interval",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceRecurringFixedCreate$;
(function (ProductPriceRecurringFixedCreate$) {
    /** @deprecated use `ProductPriceRecurringFixedCreate$inboundSchema` instead. */
    ProductPriceRecurringFixedCreate$.inboundSchema = exports.ProductPriceRecurringFixedCreate$inboundSchema;
    /** @deprecated use `ProductPriceRecurringFixedCreate$outboundSchema` instead. */
    ProductPriceRecurringFixedCreate$.outboundSchema = exports.ProductPriceRecurringFixedCreate$outboundSchema;
})(ProductPriceRecurringFixedCreate$ || (exports.ProductPriceRecurringFixedCreate$ = ProductPriceRecurringFixedCreate$ = {}));
