// @polar-sh/sdk/models/components/reactions.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Reactions$ = exports.Reactions$outboundSchema = exports.Reactions$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.Reactions$inboundSchema = z.object({
    total_count: z.number().int(),
    plus_one: z.number().int(),
    minus_one: z.number().int(),
    laugh: z.number().int(),
    hooray: z.number().int(),
    confused: z.number().int(),
    heart: z.number().int(),
    rocket: z.number().int(),
    eyes: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "total_count": "totalCount",
        "plus_one": "plusOne",
        "minus_one": "minusOne",
    });
});
/** @internal */
exports.Reactions$outboundSchema = z.object({
    totalCount: z.number().int(),
    plusOne: z.number().int(),
    minusOne: z.number().int(),
    laugh: z.number().int(),
    hooray: z.number().int(),
    confused: z.number().int(),
    heart: z.number().int(),
    rocket: z.number().int(),
    eyes: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        totalCount: "total_count",
        plusOne: "plus_one",
        minusOne: "minus_one",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Reactions$;
(function (Reactions$) {
    /** @deprecated use `Reactions$inboundSchema` instead. */
    Reactions$.inboundSchema = exports.Reactions$inboundSchema;
    /** @deprecated use `Reactions$outboundSchema` instead. */
    Reactions$.outboundSchema = exports.Reactions$outboundSchema;
})(Reactions$ || (exports.Reactions$ = Reactions$ = {}));
