// @polar-sh/sdk/models/components/repository.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository$ = exports.Repository$outboundSchema = exports.Repository$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const externalorganization_js_1 = require("./externalorganization.js");
const repositoryprofilesettings_js_1 = require("./repositoryprofilesettings.js");
/** @internal */
exports.Repository$inboundSchema = z.object({
    id: z.string(),
    platform: z.literal("github").optional(),
    is_private: z.boolean(),
    name: z.string(),
    description: z.nullable(z.string()),
    stars: z.nullable(z.number().int()),
    license: z.nullable(z.string()),
    homepage: z.nullable(z.string()),
    profile_settings: z.nullable(repositoryprofilesettings_js_1.RepositoryProfileSettings$inboundSchema),
    organization: externalorganization_js_1.ExternalOrganization$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "is_private": "isPrivate",
        "profile_settings": "profileSettings",
    });
});
/** @internal */
exports.Repository$outboundSchema = z.object({
    id: z.string(),
    platform: z.literal("github").default("github"),
    isPrivate: z.boolean(),
    name: z.string(),
    description: z.nullable(z.string()),
    stars: z.nullable(z.number().int()),
    license: z.nullable(z.string()),
    homepage: z.nullable(z.string()),
    profileSettings: z.nullable(repositoryprofilesettings_js_1.RepositoryProfileSettings$outboundSchema),
    organization: externalorganization_js_1.ExternalOrganization$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        isPrivate: "is_private",
        profileSettings: "profile_settings",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Repository$;
(function (Repository$) {
    /** @deprecated use `Repository$inboundSchema` instead. */
    Repository$.inboundSchema = exports.Repository$inboundSchema;
    /** @deprecated use `Repository$outboundSchema` instead. */
    Repository$.outboundSchema = exports.Repository$outboundSchema;
})(Repository$ || (exports.Repository$ = Repository$ = {}));
