// @polar-sh/sdk/models/components/repositoryprofilesettingsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryProfileSettingsUpdate$ = exports.RepositoryProfileSettingsUpdate$outboundSchema = exports.RepositoryProfileSettingsUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.RepositoryProfileSettingsUpdate$inboundSchema = z.object({
    set_description: z.nullable(z.boolean()).optional(),
    description: z.nullable(z.string()).optional(),
    set_cover_image_url: z.nullable(z.boolean()).optional(),
    cover_image_url: z.nullable(z.string()).optional(),
    featured_organizations: z.nullable(z.array(z.string())).optional(),
    highlighted_subscription_tiers: z.nullable(z.array(z.string())).optional(),
    links: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "set_description": "setDescription",
        "set_cover_image_url": "setCoverImageUrl",
        "cover_image_url": "coverImageUrl",
        "featured_organizations": "featuredOrganizations",
        "highlighted_subscription_tiers": "highlightedSubscriptionTiers",
    });
});
/** @internal */
exports.RepositoryProfileSettingsUpdate$outboundSchema = z.object({
    setDescription: z.nullable(z.boolean()).optional(),
    description: z.nullable(z.string()).optional(),
    setCoverImageUrl: z.nullable(z.boolean()).optional(),
    coverImageUrl: z.nullable(z.string()).optional(),
    featuredOrganizations: z.nullable(z.array(z.string())).optional(),
    highlightedSubscriptionTiers: z.nullable(z.array(z.string())).optional(),
    links: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        setDescription: "set_description",
        setCoverImageUrl: "set_cover_image_url",
        coverImageUrl: "cover_image_url",
        featuredOrganizations: "featured_organizations",
        highlightedSubscriptionTiers: "highlighted_subscription_tiers",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RepositoryProfileSettingsUpdate$;
(function (RepositoryProfileSettingsUpdate$) {
    /** @deprecated use `RepositoryProfileSettingsUpdate$inboundSchema` instead. */
    RepositoryProfileSettingsUpdate$.inboundSchema = exports.RepositoryProfileSettingsUpdate$inboundSchema;
    /** @deprecated use `RepositoryProfileSettingsUpdate$outboundSchema` instead. */
    RepositoryProfileSettingsUpdate$.outboundSchema = exports.RepositoryProfileSettingsUpdate$outboundSchema;
})(RepositoryProfileSettingsUpdate$ || (exports.RepositoryProfileSettingsUpdate$ = RepositoryProfileSettingsUpdate$ = {}));
