// @polar-sh/sdk/models/components/tokenresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenResponse$ = exports.TokenResponse$outboundSchema = exports.TokenResponse$inboundSchema = exports.TokenType$ = exports.TokenType$outboundSchema = exports.TokenType$inboundSchema = exports.TokenType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.TokenType = {
    Bearer: "Bearer",
};
/** @internal */
exports.TokenType$inboundSchema = z
    .nativeEnum(exports.TokenType);
/** @internal */
exports.TokenType$outboundSchema = exports.TokenType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TokenType$;
(function (TokenType$) {
    /** @deprecated use `TokenType$inboundSchema` instead. */
    TokenType$.inboundSchema = exports.TokenType$inboundSchema;
    /** @deprecated use `TokenType$outboundSchema` instead. */
    TokenType$.outboundSchema = exports.TokenType$outboundSchema;
})(TokenType$ || (exports.TokenType$ = TokenType$ = {}));
/** @internal */
exports.TokenResponse$inboundSchema = z.object({
    access_token: z.string(),
    token_type: z.literal("Bearer").optional(),
    expires_in: z.number().int(),
    refresh_token: z.nullable(z.string()),
    scope: z.string(),
    id_token: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "access_token": "accessToken",
        "token_type": "tokenType",
        "expires_in": "expiresIn",
        "refresh_token": "refreshToken",
        "id_token": "idToken",
    });
});
/** @internal */
exports.TokenResponse$outboundSchema = z.object({
    accessToken: z.string(),
    tokenType: z.literal("Bearer").default("Bearer"),
    expiresIn: z.number().int(),
    refreshToken: z.nullable(z.string()),
    scope: z.string(),
    idToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accessToken: "access_token",
        tokenType: "token_type",
        expiresIn: "expires_in",
        refreshToken: "refresh_token",
        idToken: "id_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TokenResponse$;
(function (TokenResponse$) {
    /** @deprecated use `TokenResponse$inboundSchema` instead. */
    TokenResponse$.inboundSchema = exports.TokenResponse$inboundSchema;
    /** @deprecated use `TokenResponse$outboundSchema` instead. */
    TokenResponse$.outboundSchema = exports.TokenResponse$outboundSchema;
})(TokenResponse$ || (exports.TokenResponse$ = TokenResponse$ = {}));
