// @polar-sh/sdk/models/components/useradvertisementcampaignupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAdvertisementCampaignUpdate$ = exports.UserAdvertisementCampaignUpdate$outboundSchema = exports.UserAdvertisementCampaignUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.UserAdvertisementCampaignUpdate$inboundSchema = z.object({
    image_url: z.nullable(z.string()).optional(),
    image_url_dark: z.nullable(z.string()).optional(),
    text: z.nullable(z.string()).optional(),
    link_url: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "image_url": "imageUrl",
        "image_url_dark": "imageUrlDark",
        "link_url": "linkUrl",
    });
});
/** @internal */
exports.UserAdvertisementCampaignUpdate$outboundSchema = z.object({
    imageUrl: z.nullable(z.string()).optional(),
    imageUrlDark: z.nullable(z.string()).optional(),
    text: z.nullable(z.string()).optional(),
    linkUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        imageUrl: "image_url",
        imageUrlDark: "image_url_dark",
        linkUrl: "link_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserAdvertisementCampaignUpdate$;
(function (UserAdvertisementCampaignUpdate$) {
    /** @deprecated use `UserAdvertisementCampaignUpdate$inboundSchema` instead. */
    UserAdvertisementCampaignUpdate$.inboundSchema = exports.UserAdvertisementCampaignUpdate$inboundSchema;
    /** @deprecated use `UserAdvertisementCampaignUpdate$outboundSchema` instead. */
    UserAdvertisementCampaignUpdate$.outboundSchema = exports.UserAdvertisementCampaignUpdate$outboundSchema;
})(UserAdvertisementCampaignUpdate$ || (exports.UserAdvertisementCampaignUpdate$ = UserAdvertisementCampaignUpdate$ = {}));
