// @polar-sh/sdk/models/components/userorderproduct.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserOrderProduct$ = exports.UserOrderProduct$outboundSchema = exports.UserOrderProduct$inboundSchema = exports.Benefits$ = exports.Benefits$outboundSchema = exports.Benefits$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitarticles_js_1 = require("./benefitarticles.js");
const benefitbase_js_1 = require("./benefitbase.js");
const productmediafileread_js_1 = require("./productmediafileread.js");
const productprice_js_1 = require("./productprice.js");
/** @internal */
exports.Benefits$inboundSchema = z.union([benefitbase_js_1.BenefitBase$inboundSchema, benefitarticles_js_1.BenefitArticles$inboundSchema]);
/** @internal */
exports.Benefits$outboundSchema = z.union([benefitbase_js_1.BenefitBase$outboundSchema, benefitarticles_js_1.BenefitArticles$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Benefits$;
(function (Benefits$) {
    /** @deprecated use `Benefits$inboundSchema` instead. */
    Benefits$.inboundSchema = exports.Benefits$inboundSchema;
    /** @deprecated use `Benefits$outboundSchema` instead. */
    Benefits$.outboundSchema = exports.Benefits$outboundSchema;
})(Benefits$ || (exports.Benefits$ = Benefits$ = {}));
/** @internal */
exports.UserOrderProduct$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$inboundSchema),
    benefits: z.array(z.union([benefitbase_js_1.BenefitBase$inboundSchema, benefitarticles_js_1.BenefitArticles$inboundSchema])),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.UserOrderProduct$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$outboundSchema),
    benefits: z.array(z.union([benefitbase_js_1.BenefitBase$outboundSchema, benefitarticles_js_1.BenefitArticles$outboundSchema])),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserOrderProduct$;
(function (UserOrderProduct$) {
    /** @deprecated use `UserOrderProduct$inboundSchema` instead. */
    UserOrderProduct$.inboundSchema = exports.UserOrderProduct$inboundSchema;
    /** @deprecated use `UserOrderProduct$outboundSchema` instead. */
    UserOrderProduct$.outboundSchema = exports.UserOrderProduct$outboundSchema;
})(UserOrderProduct$ || (exports.UserOrderProduct$ = UserOrderProduct$ = {}));
