// @polar-sh/sdk/models/components/usersubscriptionsortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSubscriptionSortProperty$ = exports.UserSubscriptionSortProperty$outboundSchema = exports.UserSubscriptionSortProperty$inboundSchema = exports.UserSubscriptionSortProperty = void 0;
const z = __importStar(require("zod"));
exports.UserSubscriptionSortProperty = {
    StartedAt: "started_at",
    MinusStartedAt: "-started_at",
    Amount: "amount",
    MinusAmount: "-amount",
    Status: "status",
    MinusStatus: "-status",
    Organization: "organization",
    MinusOrganization: "-organization",
    Product: "product",
    MinusProduct: "-product",
};
/** @internal */
exports.UserSubscriptionSortProperty$inboundSchema = z.nativeEnum(exports.UserSubscriptionSortProperty);
/** @internal */
exports.UserSubscriptionSortProperty$outboundSchema = exports.UserSubscriptionSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserSubscriptionSortProperty$;
(function (UserSubscriptionSortProperty$) {
    /** @deprecated use `UserSubscriptionSortProperty$inboundSchema` instead. */
    UserSubscriptionSortProperty$.inboundSchema = exports.UserSubscriptionSortProperty$inboundSchema;
    /** @deprecated use `UserSubscriptionSortProperty$outboundSchema` instead. */
    UserSubscriptionSortProperty$.outboundSchema = exports.UserSubscriptionSortProperty$outboundSchema;
})(UserSubscriptionSortProperty$ || (exports.UserSubscriptionSortProperty$ = UserSubscriptionSortProperty$ = {}));
