// @polar-sh/sdk/models/components/userinfouser.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserInfoUser$ = exports.UserInfoUser$outboundSchema = exports.UserInfoUser$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.UserInfoUser$inboundSchema = z.object({
    sub: z.string(),
    name: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    email_verified: z.nullable(z.boolean()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "email_verified": "emailVerified",
    });
});
/** @internal */
exports.UserInfoUser$outboundSchema = z.object({
    sub: z.string(),
    name: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()).optional(),
    emailVerified: z.nullable(z.boolean()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        emailVerified: "email_verified",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserInfoUser$;
(function (UserInfoUser$) {
    /** @deprecated use `UserInfoUser$inboundSchema` instead. */
    UserInfoUser$.inboundSchema = exports.UserInfoUser$inboundSchema;
    /** @deprecated use `UserInfoUser$outboundSchema` instead. */
    UserInfoUser$.outboundSchema = exports.UserInfoUser$outboundSchema;
})(UserInfoUser$ || (exports.UserInfoUser$ = UserInfoUser$ = {}));
