// @polar-sh/sdk/models/components/useradvertisementcampaign.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAdvertisementCampaign$ = exports.UserAdvertisementCampaign$outboundSchema = exports.UserAdvertisementCampaign$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.UserAdvertisementCampaign$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    user_id: z.string(),
    views: z.number().int(),
    clicks: z.number().int(),
    image_url: z.string(),
    image_url_dark: z.nullable(z.string()),
    text: z.string(),
    link_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "user_id": "userId",
        "image_url": "imageUrl",
        "image_url_dark": "imageUrlDark",
        "link_url": "linkUrl",
    });
});
/** @internal */
exports.UserAdvertisementCampaign$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    userId: z.string(),
    views: z.number().int(),
    clicks: z.number().int(),
    imageUrl: z.string(),
    imageUrlDark: z.nullable(z.string()),
    text: z.string(),
    linkUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        userId: "user_id",
        imageUrl: "image_url",
        imageUrlDark: "image_url_dark",
        linkUrl: "link_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserAdvertisementCampaign$;
(function (UserAdvertisementCampaign$) {
    /** @deprecated use `UserAdvertisementCampaign$inboundSchema` instead. */
    UserAdvertisementCampaign$.inboundSchema = exports.UserAdvertisementCampaign$inboundSchema;
    /** @deprecated use `UserAdvertisementCampaign$outboundSchema` instead. */
    UserAdvertisementCampaign$.outboundSchema = exports.UserAdvertisementCampaign$outboundSchema;
})(UserAdvertisementCampaign$ || (exports.UserAdvertisementCampaign$ = UserAdvertisementCampaign$ = {}));
