// @polar-sh/sdk/models/components/webhookproductupdatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookProductUpdatedPayload$ = exports.WebhookProductUpdatedPayload$outboundSchema = exports.WebhookProductUpdatedPayload$inboundSchema = exports.WebhookProductUpdatedPayloadType$ = exports.WebhookProductUpdatedPayloadType$outboundSchema = exports.WebhookProductUpdatedPayloadType$inboundSchema = exports.WebhookProductUpdatedPayloadType = void 0;
const z = __importStar(require("zod"));
const product_js_1 = require("./product.js");
exports.WebhookProductUpdatedPayloadType = {
    ProductUpdated: "product.updated",
};
/** @internal */
exports.WebhookProductUpdatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookProductUpdatedPayloadType);
/** @internal */
exports.WebhookProductUpdatedPayloadType$outboundSchema = exports.WebhookProductUpdatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookProductUpdatedPayloadType$;
(function (WebhookProductUpdatedPayloadType$) {
    /** @deprecated use `WebhookProductUpdatedPayloadType$inboundSchema` instead. */
    WebhookProductUpdatedPayloadType$.inboundSchema = exports.WebhookProductUpdatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookProductUpdatedPayloadType$outboundSchema` instead. */
    WebhookProductUpdatedPayloadType$.outboundSchema = exports.WebhookProductUpdatedPayloadType$outboundSchema;
})(WebhookProductUpdatedPayloadType$ || (exports.WebhookProductUpdatedPayloadType$ = WebhookProductUpdatedPayloadType$ = {}));
/** @internal */
exports.WebhookProductUpdatedPayload$inboundSchema = z.object({
    type: z.literal("product.updated").optional(),
    data: product_js_1.ProductInput$inboundSchema,
});
/** @internal */
exports.WebhookProductUpdatedPayload$outboundSchema = z.object({
    type: z.literal("product.updated").default("product.updated"),
    data: product_js_1.ProductInput$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookProductUpdatedPayload$;
(function (WebhookProductUpdatedPayload$) {
    /** @deprecated use `WebhookProductUpdatedPayload$inboundSchema` instead. */
    WebhookProductUpdatedPayload$.inboundSchema = exports.WebhookProductUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookProductUpdatedPayload$outboundSchema` instead. */
    WebhookProductUpdatedPayload$.outboundSchema = exports.WebhookProductUpdatedPayload$outboundSchema;
})(WebhookProductUpdatedPayload$ || (exports.WebhookProductUpdatedPayload$ = WebhookProductUpdatedPayload$ = {}));
