// @polar-sh/sdk/models/components/webhookordercreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookOrderCreatedPayload$ = exports.WebhookOrderCreatedPayload$outboundSchema = exports.WebhookOrderCreatedPayload$inboundSchema = exports.WebhookOrderCreatedPayloadType$ = exports.WebhookOrderCreatedPayloadType$outboundSchema = exports.WebhookOrderCreatedPayloadType$inboundSchema = exports.WebhookOrderCreatedPayloadType = void 0;
const z = __importStar(require("zod"));
const order_js_1 = require("./order.js");
exports.WebhookOrderCreatedPayloadType = {
    OrderCreated: "order.created",
};
/** @internal */
exports.WebhookOrderCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookOrderCreatedPayloadType);
/** @internal */
exports.WebhookOrderCreatedPayloadType$outboundSchema = exports.WebhookOrderCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookOrderCreatedPayloadType$;
(function (WebhookOrderCreatedPayloadType$) {
    /** @deprecated use `WebhookOrderCreatedPayloadType$inboundSchema` instead. */
    WebhookOrderCreatedPayloadType$.inboundSchema = exports.WebhookOrderCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookOrderCreatedPayloadType$outboundSchema` instead. */
    WebhookOrderCreatedPayloadType$.outboundSchema = exports.WebhookOrderCreatedPayloadType$outboundSchema;
})(WebhookOrderCreatedPayloadType$ || (exports.WebhookOrderCreatedPayloadType$ = WebhookOrderCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookOrderCreatedPayload$inboundSchema = z.object({
    type: z.literal("order.created").optional(),
    data: order_js_1.Order$inboundSchema,
});
/** @internal */
exports.WebhookOrderCreatedPayload$outboundSchema = z.object({
    type: z.literal("order.created").default("order.created"),
    data: order_js_1.Order$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookOrderCreatedPayload$;
(function (WebhookOrderCreatedPayload$) {
    /** @deprecated use `WebhookOrderCreatedPayload$inboundSchema` instead. */
    WebhookOrderCreatedPayload$.inboundSchema = exports.WebhookOrderCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookOrderCreatedPayload$outboundSchema` instead. */
    WebhookOrderCreatedPayload$.outboundSchema = exports.WebhookOrderCreatedPayload$outboundSchema;
})(WebhookOrderCreatedPayload$ || (exports.WebhookOrderCreatedPayload$ = WebhookOrderCreatedPayload$ = {}));
