// @polar-sh/sdk/models/components/webhookbenefitgrantcreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookBenefitGrantCreatedPayload$ = exports.WebhookBenefitGrantCreatedPayload$outboundSchema = exports.WebhookBenefitGrantCreatedPayload$inboundSchema = exports.WebhookBenefitGrantCreatedPayloadType$ = exports.WebhookBenefitGrantCreatedPayloadType$outboundSchema = exports.WebhookBenefitGrantCreatedPayloadType$inboundSchema = exports.WebhookBenefitGrantCreatedPayloadType = void 0;
const z = __importStar(require("zod"));
const benefitgrantwebhook_js_1 = require("./benefitgrantwebhook.js");
exports.WebhookBenefitGrantCreatedPayloadType = {
    BenefitGrantCreated: "benefit_grant.created",
};
/** @internal */
exports.WebhookBenefitGrantCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookBenefitGrantCreatedPayloadType);
/** @internal */
exports.WebhookBenefitGrantCreatedPayloadType$outboundSchema = exports.WebhookBenefitGrantCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitGrantCreatedPayloadType$;
(function (WebhookBenefitGrantCreatedPayloadType$) {
    /** @deprecated use `WebhookBenefitGrantCreatedPayloadType$inboundSchema` instead. */
    WebhookBenefitGrantCreatedPayloadType$.inboundSchema = exports.WebhookBenefitGrantCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantCreatedPayloadType$outboundSchema` instead. */
    WebhookBenefitGrantCreatedPayloadType$.outboundSchema = exports.WebhookBenefitGrantCreatedPayloadType$outboundSchema;
})(WebhookBenefitGrantCreatedPayloadType$ || (exports.WebhookBenefitGrantCreatedPayloadType$ = WebhookBenefitGrantCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookBenefitGrantCreatedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.created").optional(),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$inboundSchema,
});
/** @internal */
exports.WebhookBenefitGrantCreatedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.created").default("benefit_grant.created"),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitGrantCreatedPayload$;
(function (WebhookBenefitGrantCreatedPayload$) {
    /** @deprecated use `WebhookBenefitGrantCreatedPayload$inboundSchema` instead. */
    WebhookBenefitGrantCreatedPayload$.inboundSchema = exports.WebhookBenefitGrantCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantCreatedPayload$outboundSchema` instead. */
    WebhookBenefitGrantCreatedPayload$.outboundSchema = exports.WebhookBenefitGrantCreatedPayload$outboundSchema;
})(WebhookBenefitGrantCreatedPayload$ || (exports.WebhookBenefitGrantCreatedPayload$ = WebhookBenefitGrantCreatedPayload$ = {}));
