// @polar-sh/sdk/models/components/webhooksubscriptionactivepayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookSubscriptionActivePayload$ = exports.WebhookSubscriptionActivePayload$outboundSchema = exports.WebhookSubscriptionActivePayload$inboundSchema = exports.WebhookSubscriptionActivePayloadType$ = exports.WebhookSubscriptionActivePayloadType$outboundSchema = exports.WebhookSubscriptionActivePayloadType$inboundSchema = exports.WebhookSubscriptionActivePayloadType = void 0;
const z = __importStar(require("zod"));
const subscriptioninput_js_1 = require("./subscriptioninput.js");
exports.WebhookSubscriptionActivePayloadType = {
    SubscriptionActive: "subscription.active",
};
/** @internal */
exports.WebhookSubscriptionActivePayloadType$inboundSchema = z.nativeEnum(exports.WebhookSubscriptionActivePayloadType);
/** @internal */
exports.WebhookSubscriptionActivePayloadType$outboundSchema = exports.WebhookSubscriptionActivePayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionActivePayloadType$;
(function (WebhookSubscriptionActivePayloadType$) {
    /** @deprecated use `WebhookSubscriptionActivePayloadType$inboundSchema` instead. */
    WebhookSubscriptionActivePayloadType$.inboundSchema = exports.WebhookSubscriptionActivePayloadType$inboundSchema;
    /** @deprecated use `WebhookSubscriptionActivePayloadType$outboundSchema` instead. */
    WebhookSubscriptionActivePayloadType$.outboundSchema = exports.WebhookSubscriptionActivePayloadType$outboundSchema;
})(WebhookSubscriptionActivePayloadType$ || (exports.WebhookSubscriptionActivePayloadType$ = WebhookSubscriptionActivePayloadType$ = {}));
/** @internal */
exports.WebhookSubscriptionActivePayload$inboundSchema = z.object({
    type: z.literal("subscription.active").optional(),
    data: subscriptioninput_js_1.SubscriptionInput$inboundSchema,
});
/** @internal */
exports.WebhookSubscriptionActivePayload$outboundSchema = z.object({
    type: z.literal("subscription.active").default("subscription.active"),
    data: subscriptioninput_js_1.SubscriptionInput$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionActivePayload$;
(function (WebhookSubscriptionActivePayload$) {
    /** @deprecated use `WebhookSubscriptionActivePayload$inboundSchema` instead. */
    WebhookSubscriptionActivePayload$.inboundSchema = exports.WebhookSubscriptionActivePayload$inboundSchema;
    /** @deprecated use `WebhookSubscriptionActivePayload$outboundSchema` instead. */
    WebhookSubscriptionActivePayload$.outboundSchema = exports.WebhookSubscriptionActivePayload$outboundSchema;
})(WebhookSubscriptionActivePayload$ || (exports.WebhookSubscriptionActivePayload$ = WebhookSubscriptionActivePayload$ = {}));
