// @polar-sh/sdk/models/components/webhookpledgecreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookPledgeCreatedPayload$ = exports.WebhookPledgeCreatedPayload$outboundSchema = exports.WebhookPledgeCreatedPayload$inboundSchema = exports.WebhookPledgeCreatedPayloadType$ = exports.WebhookPledgeCreatedPayloadType$outboundSchema = exports.WebhookPledgeCreatedPayloadType$inboundSchema = exports.WebhookPledgeCreatedPayloadType = void 0;
const z = __importStar(require("zod"));
const pledge_js_1 = require("./pledge.js");
exports.WebhookPledgeCreatedPayloadType = {
    PledgeCreated: "pledge.created",
};
/** @internal */
exports.WebhookPledgeCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookPledgeCreatedPayloadType);
/** @internal */
exports.WebhookPledgeCreatedPayloadType$outboundSchema = exports.WebhookPledgeCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPledgeCreatedPayloadType$;
(function (WebhookPledgeCreatedPayloadType$) {
    /** @deprecated use `WebhookPledgeCreatedPayloadType$inboundSchema` instead. */
    WebhookPledgeCreatedPayloadType$.inboundSchema = exports.WebhookPledgeCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookPledgeCreatedPayloadType$outboundSchema` instead. */
    WebhookPledgeCreatedPayloadType$.outboundSchema = exports.WebhookPledgeCreatedPayloadType$outboundSchema;
})(WebhookPledgeCreatedPayloadType$ || (exports.WebhookPledgeCreatedPayloadType$ = WebhookPledgeCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookPledgeCreatedPayload$inboundSchema = z.object({
    type: z.literal("pledge.created").optional(),
    data: pledge_js_1.Pledge$inboundSchema,
});
/** @internal */
exports.WebhookPledgeCreatedPayload$outboundSchema = z.object({
    type: z.literal("pledge.created").default("pledge.created"),
    data: pledge_js_1.Pledge$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPledgeCreatedPayload$;
(function (WebhookPledgeCreatedPayload$) {
    /** @deprecated use `WebhookPledgeCreatedPayload$inboundSchema` instead. */
    WebhookPledgeCreatedPayload$.inboundSchema = exports.WebhookPledgeCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookPledgeCreatedPayload$outboundSchema` instead. */
    WebhookPledgeCreatedPayload$.outboundSchema = exports.WebhookPledgeCreatedPayload$outboundSchema;
})(WebhookPledgeCreatedPayload$ || (exports.WebhookPledgeCreatedPayload$ = WebhookPledgeCreatedPayload$ = {}));
