// @polar-sh/sdk/models/components/webhookdonationcreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookDonationCreatedPayload$ = exports.WebhookDonationCreatedPayload$outboundSchema = exports.WebhookDonationCreatedPayload$inboundSchema = exports.WebhookDonationCreatedPayloadType$ = exports.WebhookDonationCreatedPayloadType$outboundSchema = exports.WebhookDonationCreatedPayloadType$inboundSchema = exports.WebhookDonationCreatedPayloadType = void 0;
const z = __importStar(require("zod"));
const donation_js_1 = require("./donation.js");
exports.WebhookDonationCreatedPayloadType = {
    DonationCreated: "donation.created",
};
/** @internal */
exports.WebhookDonationCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookDonationCreatedPayloadType);
/** @internal */
exports.WebhookDonationCreatedPayloadType$outboundSchema = exports.WebhookDonationCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookDonationCreatedPayloadType$;
(function (WebhookDonationCreatedPayloadType$) {
    /** @deprecated use `WebhookDonationCreatedPayloadType$inboundSchema` instead. */
    WebhookDonationCreatedPayloadType$.inboundSchema = exports.WebhookDonationCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookDonationCreatedPayloadType$outboundSchema` instead. */
    WebhookDonationCreatedPayloadType$.outboundSchema = exports.WebhookDonationCreatedPayloadType$outboundSchema;
})(WebhookDonationCreatedPayloadType$ || (exports.WebhookDonationCreatedPayloadType$ = WebhookDonationCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookDonationCreatedPayload$inboundSchema = z.object({
    type: z.literal("donation.created").optional(),
    data: donation_js_1.Donation$inboundSchema,
});
/** @internal */
exports.WebhookDonationCreatedPayload$outboundSchema = z.object({
    type: z.literal("donation.created").default("donation.created"),
    data: donation_js_1.Donation$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookDonationCreatedPayload$;
(function (WebhookDonationCreatedPayload$) {
    /** @deprecated use `WebhookDonationCreatedPayload$inboundSchema` instead. */
    WebhookDonationCreatedPayload$.inboundSchema = exports.WebhookDonationCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookDonationCreatedPayload$outboundSchema` instead. */
    WebhookDonationCreatedPayload$.outboundSchema = exports.WebhookDonationCreatedPayload$outboundSchema;
})(WebhookDonationCreatedPayload$ || (exports.WebhookDonationCreatedPayload$ = WebhookDonationCreatedPayload$ = {}));
