// @polar-sh/sdk/models/components/webhookorganizationupdatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookOrganizationUpdatedPayload$ = exports.WebhookOrganizationUpdatedPayload$outboundSchema = exports.WebhookOrganizationUpdatedPayload$inboundSchema = exports.WebhookOrganizationUpdatedPayloadType$ = exports.WebhookOrganizationUpdatedPayloadType$outboundSchema = exports.WebhookOrganizationUpdatedPayloadType$inboundSchema = exports.WebhookOrganizationUpdatedPayloadType = void 0;
const z = __importStar(require("zod"));
const organization_js_1 = require("./organization.js");
exports.WebhookOrganizationUpdatedPayloadType = {
    OrganizationUpdated: "organization.updated",
};
/** @internal */
exports.WebhookOrganizationUpdatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookOrganizationUpdatedPayloadType);
/** @internal */
exports.WebhookOrganizationUpdatedPayloadType$outboundSchema = exports.WebhookOrganizationUpdatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookOrganizationUpdatedPayloadType$;
(function (WebhookOrganizationUpdatedPayloadType$) {
    /** @deprecated use `WebhookOrganizationUpdatedPayloadType$inboundSchema` instead. */
    WebhookOrganizationUpdatedPayloadType$.inboundSchema = exports.WebhookOrganizationUpdatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookOrganizationUpdatedPayloadType$outboundSchema` instead. */
    WebhookOrganizationUpdatedPayloadType$.outboundSchema = exports.WebhookOrganizationUpdatedPayloadType$outboundSchema;
})(WebhookOrganizationUpdatedPayloadType$ || (exports.WebhookOrganizationUpdatedPayloadType$ = WebhookOrganizationUpdatedPayloadType$ = {}));
/** @internal */
exports.WebhookOrganizationUpdatedPayload$inboundSchema = z.object({
    type: z.literal("organization.updated").optional(),
    data: organization_js_1.Organization$inboundSchema,
});
/** @internal */
exports.WebhookOrganizationUpdatedPayload$outboundSchema = z.object({
    type: z.literal("organization.updated").default("organization.updated"),
    data: organization_js_1.Organization$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookOrganizationUpdatedPayload$;
(function (WebhookOrganizationUpdatedPayload$) {
    /** @deprecated use `WebhookOrganizationUpdatedPayload$inboundSchema` instead. */
    WebhookOrganizationUpdatedPayload$.inboundSchema = exports.WebhookOrganizationUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookOrganizationUpdatedPayload$outboundSchema` instead. */
    WebhookOrganizationUpdatedPayload$.outboundSchema = exports.WebhookOrganizationUpdatedPayload$outboundSchema;
})(WebhookOrganizationUpdatedPayload$ || (exports.WebhookOrganizationUpdatedPayload$ = WebhookOrganizationUpdatedPayload$ = {}));
