// @polar-sh/sdk/models/components/webhookbenefitgrantupdatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookBenefitGrantUpdatedPayload$ = exports.WebhookBenefitGrantUpdatedPayload$outboundSchema = exports.WebhookBenefitGrantUpdatedPayload$inboundSchema = exports.WebhookBenefitGrantUpdatedPayloadType$ = exports.WebhookBenefitGrantUpdatedPayloadType$outboundSchema = exports.WebhookBenefitGrantUpdatedPayloadType$inboundSchema = exports.WebhookBenefitGrantUpdatedPayloadType = void 0;
const z = __importStar(require("zod"));
const benefitgrantwebhook_js_1 = require("./benefitgrantwebhook.js");
exports.WebhookBenefitGrantUpdatedPayloadType = {
    BenefitGrantUpdated: "benefit_grant.updated",
};
/** @internal */
exports.WebhookBenefitGrantUpdatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookBenefitGrantUpdatedPayloadType);
/** @internal */
exports.WebhookBenefitGrantUpdatedPayloadType$outboundSchema = exports.WebhookBenefitGrantUpdatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitGrantUpdatedPayloadType$;
(function (WebhookBenefitGrantUpdatedPayloadType$) {
    /** @deprecated use `WebhookBenefitGrantUpdatedPayloadType$inboundSchema` instead. */
    WebhookBenefitGrantUpdatedPayloadType$.inboundSchema = exports.WebhookBenefitGrantUpdatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantUpdatedPayloadType$outboundSchema` instead. */
    WebhookBenefitGrantUpdatedPayloadType$.outboundSchema = exports.WebhookBenefitGrantUpdatedPayloadType$outboundSchema;
})(WebhookBenefitGrantUpdatedPayloadType$ || (exports.WebhookBenefitGrantUpdatedPayloadType$ = WebhookBenefitGrantUpdatedPayloadType$ = {}));
/** @internal */
exports.WebhookBenefitGrantUpdatedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.updated").optional(),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$inboundSchema,
});
/** @internal */
exports.WebhookBenefitGrantUpdatedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.updated").default("benefit_grant.updated"),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitGrantUpdatedPayload$;
(function (WebhookBenefitGrantUpdatedPayload$) {
    /** @deprecated use `WebhookBenefitGrantUpdatedPayload$inboundSchema` instead. */
    WebhookBenefitGrantUpdatedPayload$.inboundSchema = exports.WebhookBenefitGrantUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantUpdatedPayload$outboundSchema` instead. */
    WebhookBenefitGrantUpdatedPayload$.outboundSchema = exports.WebhookBenefitGrantUpdatedPayload$outboundSchema;
})(WebhookBenefitGrantUpdatedPayload$ || (exports.WebhookBenefitGrantUpdatedPayload$ = WebhookBenefitGrantUpdatedPayload$ = {}));
