// @polar-sh/sdk/models/components/webhookpledgeupdatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookPledgeUpdatedPayload$ = exports.WebhookPledgeUpdatedPayload$outboundSchema = exports.WebhookPledgeUpdatedPayload$inboundSchema = exports.WebhookPledgeUpdatedPayloadType$ = exports.WebhookPledgeUpdatedPayloadType$outboundSchema = exports.WebhookPledgeUpdatedPayloadType$inboundSchema = exports.WebhookPledgeUpdatedPayloadType = void 0;
const z = __importStar(require("zod"));
const pledge_js_1 = require("./pledge.js");
exports.WebhookPledgeUpdatedPayloadType = {
    PledgeUpdated: "pledge.updated",
};
/** @internal */
exports.WebhookPledgeUpdatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookPledgeUpdatedPayloadType);
/** @internal */
exports.WebhookPledgeUpdatedPayloadType$outboundSchema = exports.WebhookPledgeUpdatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPledgeUpdatedPayloadType$;
(function (WebhookPledgeUpdatedPayloadType$) {
    /** @deprecated use `WebhookPledgeUpdatedPayloadType$inboundSchema` instead. */
    WebhookPledgeUpdatedPayloadType$.inboundSchema = exports.WebhookPledgeUpdatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookPledgeUpdatedPayloadType$outboundSchema` instead. */
    WebhookPledgeUpdatedPayloadType$.outboundSchema = exports.WebhookPledgeUpdatedPayloadType$outboundSchema;
})(WebhookPledgeUpdatedPayloadType$ || (exports.WebhookPledgeUpdatedPayloadType$ = WebhookPledgeUpdatedPayloadType$ = {}));
/** @internal */
exports.WebhookPledgeUpdatedPayload$inboundSchema = z.object({
    type: z.literal("pledge.updated").optional(),
    data: pledge_js_1.Pledge$inboundSchema,
});
/** @internal */
exports.WebhookPledgeUpdatedPayload$outboundSchema = z.object({
    type: z.literal("pledge.updated").default("pledge.updated"),
    data: pledge_js_1.Pledge$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPledgeUpdatedPayload$;
(function (WebhookPledgeUpdatedPayload$) {
    /** @deprecated use `WebhookPledgeUpdatedPayload$inboundSchema` instead. */
    WebhookPledgeUpdatedPayload$.inboundSchema = exports.WebhookPledgeUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookPledgeUpdatedPayload$outboundSchema` instead. */
    WebhookPledgeUpdatedPayload$.outboundSchema = exports.WebhookPledgeUpdatedPayload$outboundSchema;
})(WebhookPledgeUpdatedPayload$ || (exports.WebhookPledgeUpdatedPayload$ = WebhookPledgeUpdatedPayload$ = {}));
