// @polar-sh/sdk/models/components/webhooksubscriptionrevokedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookSubscriptionRevokedPayload$ = exports.WebhookSubscriptionRevokedPayload$outboundSchema = exports.WebhookSubscriptionRevokedPayload$inboundSchema = exports.WebhookSubscriptionRevokedPayloadType$ = exports.WebhookSubscriptionRevokedPayloadType$outboundSchema = exports.WebhookSubscriptionRevokedPayloadType$inboundSchema = exports.WebhookSubscriptionRevokedPayloadType = void 0;
const z = __importStar(require("zod"));
const subscriptioninput_js_1 = require("./subscriptioninput.js");
exports.WebhookSubscriptionRevokedPayloadType = {
    SubscriptionRevoked: "subscription.revoked",
};
/** @internal */
exports.WebhookSubscriptionRevokedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookSubscriptionRevokedPayloadType);
/** @internal */
exports.WebhookSubscriptionRevokedPayloadType$outboundSchema = exports.WebhookSubscriptionRevokedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionRevokedPayloadType$;
(function (WebhookSubscriptionRevokedPayloadType$) {
    /** @deprecated use `WebhookSubscriptionRevokedPayloadType$inboundSchema` instead. */
    WebhookSubscriptionRevokedPayloadType$.inboundSchema = exports.WebhookSubscriptionRevokedPayloadType$inboundSchema;
    /** @deprecated use `WebhookSubscriptionRevokedPayloadType$outboundSchema` instead. */
    WebhookSubscriptionRevokedPayloadType$.outboundSchema = exports.WebhookSubscriptionRevokedPayloadType$outboundSchema;
})(WebhookSubscriptionRevokedPayloadType$ || (exports.WebhookSubscriptionRevokedPayloadType$ = WebhookSubscriptionRevokedPayloadType$ = {}));
/** @internal */
exports.WebhookSubscriptionRevokedPayload$inboundSchema = z.object({
    type: z.literal("subscription.revoked").optional(),
    data: subscriptioninput_js_1.SubscriptionInput$inboundSchema,
});
/** @internal */
exports.WebhookSubscriptionRevokedPayload$outboundSchema = z.object({
    type: z.literal("subscription.revoked").default("subscription.revoked"),
    data: subscriptioninput_js_1.SubscriptionInput$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionRevokedPayload$;
(function (WebhookSubscriptionRevokedPayload$) {
    /** @deprecated use `WebhookSubscriptionRevokedPayload$inboundSchema` instead. */
    WebhookSubscriptionRevokedPayload$.inboundSchema = exports.WebhookSubscriptionRevokedPayload$inboundSchema;
    /** @deprecated use `WebhookSubscriptionRevokedPayload$outboundSchema` instead. */
    WebhookSubscriptionRevokedPayload$.outboundSchema = exports.WebhookSubscriptionRevokedPayload$outboundSchema;
})(WebhookSubscriptionRevokedPayload$ || (exports.WebhookSubscriptionRevokedPayload$ = WebhookSubscriptionRevokedPayload$ = {}));
