// @polar-sh/sdk/models/components/webhookbenefitcreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookBenefitCreatedPayload$ = exports.WebhookBenefitCreatedPayload$outboundSchema = exports.WebhookBenefitCreatedPayload$inboundSchema = exports.WebhookBenefitCreatedPayloadType$ = exports.WebhookBenefitCreatedPayloadType$outboundSchema = exports.WebhookBenefitCreatedPayloadType$inboundSchema = exports.WebhookBenefitCreatedPayloadType = void 0;
const z = __importStar(require("zod"));
const benefitinput_js_1 = require("./benefitinput.js");
exports.WebhookBenefitCreatedPayloadType = {
    BenefitCreated: "benefit.created",
};
/** @internal */
exports.WebhookBenefitCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookBenefitCreatedPayloadType);
/** @internal */
exports.WebhookBenefitCreatedPayloadType$outboundSchema = exports.WebhookBenefitCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitCreatedPayloadType$;
(function (WebhookBenefitCreatedPayloadType$) {
    /** @deprecated use `WebhookBenefitCreatedPayloadType$inboundSchema` instead. */
    WebhookBenefitCreatedPayloadType$.inboundSchema = exports.WebhookBenefitCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookBenefitCreatedPayloadType$outboundSchema` instead. */
    WebhookBenefitCreatedPayloadType$.outboundSchema = exports.WebhookBenefitCreatedPayloadType$outboundSchema;
})(WebhookBenefitCreatedPayloadType$ || (exports.WebhookBenefitCreatedPayloadType$ = WebhookBenefitCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookBenefitCreatedPayload$inboundSchema = z.object({
    type: z.literal("benefit.created").optional(),
    data: benefitinput_js_1.BenefitInput$inboundSchema,
});
/** @internal */
exports.WebhookBenefitCreatedPayload$outboundSchema = z.object({
    type: z.literal("benefit.created").default("benefit.created"),
    data: benefitinput_js_1.BenefitInput$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitCreatedPayload$;
(function (WebhookBenefitCreatedPayload$) {
    /** @deprecated use `WebhookBenefitCreatedPayload$inboundSchema` instead. */
    WebhookBenefitCreatedPayload$.inboundSchema = exports.WebhookBenefitCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitCreatedPayload$outboundSchema` instead. */
    WebhookBenefitCreatedPayload$.outboundSchema = exports.WebhookBenefitCreatedPayload$outboundSchema;
})(WebhookBenefitCreatedPayload$ || (exports.WebhookBenefitCreatedPayload$ = WebhookBenefitCreatedPayload$ = {}));
