// @polar-sh/sdk/models/components/webhookbenefitgrantrevokedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookBenefitGrantRevokedPayload$ = exports.WebhookBenefitGrantRevokedPayload$outboundSchema = exports.WebhookBenefitGrantRevokedPayload$inboundSchema = exports.WebhookBenefitGrantRevokedPayloadType$ = exports.WebhookBenefitGrantRevokedPayloadType$outboundSchema = exports.WebhookBenefitGrantRevokedPayloadType$inboundSchema = exports.WebhookBenefitGrantRevokedPayloadType = void 0;
const z = __importStar(require("zod"));
const benefitgrantwebhook_js_1 = require("./benefitgrantwebhook.js");
exports.WebhookBenefitGrantRevokedPayloadType = {
    BenefitGrantRevoked: "benefit_grant.revoked",
};
/** @internal */
exports.WebhookBenefitGrantRevokedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookBenefitGrantRevokedPayloadType);
/** @internal */
exports.WebhookBenefitGrantRevokedPayloadType$outboundSchema = exports.WebhookBenefitGrantRevokedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitGrantRevokedPayloadType$;
(function (WebhookBenefitGrantRevokedPayloadType$) {
    /** @deprecated use `WebhookBenefitGrantRevokedPayloadType$inboundSchema` instead. */
    WebhookBenefitGrantRevokedPayloadType$.inboundSchema = exports.WebhookBenefitGrantRevokedPayloadType$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantRevokedPayloadType$outboundSchema` instead. */
    WebhookBenefitGrantRevokedPayloadType$.outboundSchema = exports.WebhookBenefitGrantRevokedPayloadType$outboundSchema;
})(WebhookBenefitGrantRevokedPayloadType$ || (exports.WebhookBenefitGrantRevokedPayloadType$ = WebhookBenefitGrantRevokedPayloadType$ = {}));
/** @internal */
exports.WebhookBenefitGrantRevokedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.revoked").optional(),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$inboundSchema,
});
/** @internal */
exports.WebhookBenefitGrantRevokedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.revoked").default("benefit_grant.revoked"),
    data: benefitgrantwebhook_js_1.BenefitGrantWebhook$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookBenefitGrantRevokedPayload$;
(function (WebhookBenefitGrantRevokedPayload$) {
    /** @deprecated use `WebhookBenefitGrantRevokedPayload$inboundSchema` instead. */
    WebhookBenefitGrantRevokedPayload$.inboundSchema = exports.WebhookBenefitGrantRevokedPayload$inboundSchema;
    /** @deprecated use `WebhookBenefitGrantRevokedPayload$outboundSchema` instead. */
    WebhookBenefitGrantRevokedPayload$.outboundSchema = exports.WebhookBenefitGrantRevokedPayload$outboundSchema;
})(WebhookBenefitGrantRevokedPayload$ || (exports.WebhookBenefitGrantRevokedPayload$ = WebhookBenefitGrantRevokedPayload$ = {}));
