// @polar-sh/sdk/models/operations/advertisementslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdvertisementsListResponse$ = exports.AdvertisementsListResponse$outboundSchema = exports.AdvertisementsListResponse$inboundSchema = exports.AdvertisementsListRequest$ = exports.AdvertisementsListRequest$outboundSchema = exports.AdvertisementsListRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AdvertisementsListRequest$inboundSchema = z.object({
    benefit_id: z.string(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.AdvertisementSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "benefit_id": "benefitId",
    });
});
/** @internal */
exports.AdvertisementsListRequest$outboundSchema = z.object({
    benefitId: z.string(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.AdvertisementSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        benefitId: "benefit_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AdvertisementsListRequest$;
(function (AdvertisementsListRequest$) {
    /** @deprecated use `AdvertisementsListRequest$inboundSchema` instead. */
    AdvertisementsListRequest$.inboundSchema = exports.AdvertisementsListRequest$inboundSchema;
    /** @deprecated use `AdvertisementsListRequest$outboundSchema` instead. */
    AdvertisementsListRequest$.outboundSchema = exports.AdvertisementsListRequest$outboundSchema;
})(AdvertisementsListRequest$ || (exports.AdvertisementsListRequest$ = AdvertisementsListRequest$ = {}));
/** @internal */
exports.AdvertisementsListResponse$inboundSchema = z.object({
    Result: components.AdvertisementCampaignListResource$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.AdvertisementsListResponse$outboundSchema = z.object({
    result: components.AdvertisementCampaignListResource$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AdvertisementsListResponse$;
(function (AdvertisementsListResponse$) {
    /** @deprecated use `AdvertisementsListResponse$inboundSchema` instead. */
    AdvertisementsListResponse$.inboundSchema = exports.AdvertisementsListResponse$inboundSchema;
    /** @deprecated use `AdvertisementsListResponse$outboundSchema` instead. */
    AdvertisementsListResponse$.outboundSchema = exports.AdvertisementsListResponse$outboundSchema;
})(AdvertisementsListResponse$ || (exports.AdvertisementsListResponse$ = AdvertisementsListResponse$ = {}));
