// @polar-sh/sdk/models/operations/articleslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArticlesListResponse$ = exports.ArticlesListResponse$outboundSchema = exports.ArticlesListResponse$inboundSchema = exports.ArticlesListRequest$ = exports.ArticlesListRequest$outboundSchema = exports.ArticlesListRequest$inboundSchema = exports.ArticleVisibilityFilter$ = exports.ArticleVisibilityFilter$outboundSchema = exports.ArticleVisibilityFilter$inboundSchema = exports.ArticlesListQueryParamOrganizationIDFilter$ = exports.ArticlesListQueryParamOrganizationIDFilter$outboundSchema = exports.ArticlesListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.ArticlesListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.ArticlesListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArticlesListQueryParamOrganizationIDFilter$;
(function (ArticlesListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `ArticlesListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    ArticlesListQueryParamOrganizationIDFilter$.inboundSchema = exports.ArticlesListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `ArticlesListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    ArticlesListQueryParamOrganizationIDFilter$.outboundSchema = exports.ArticlesListQueryParamOrganizationIDFilter$outboundSchema;
})(ArticlesListQueryParamOrganizationIDFilter$ || (exports.ArticlesListQueryParamOrganizationIDFilter$ = ArticlesListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.ArticleVisibilityFilter$inboundSchema = z.union([
    components.ArticleVisibility$inboundSchema,
    z.array(components.ArticleVisibility$inboundSchema),
]);
/** @internal */
exports.ArticleVisibilityFilter$outboundSchema = z.union([
    components.ArticleVisibility$outboundSchema,
    z.array(components.ArticleVisibility$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArticleVisibilityFilter$;
(function (ArticleVisibilityFilter$) {
    /** @deprecated use `ArticleVisibilityFilter$inboundSchema` instead. */
    ArticleVisibilityFilter$.inboundSchema = exports.ArticleVisibilityFilter$inboundSchema;
    /** @deprecated use `ArticleVisibilityFilter$outboundSchema` instead. */
    ArticleVisibilityFilter$.outboundSchema = exports.ArticleVisibilityFilter$outboundSchema;
})(ArticleVisibilityFilter$ || (exports.ArticleVisibilityFilter$ = ArticleVisibilityFilter$ = {}));
/** @internal */
exports.ArticlesListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    slug: z.nullable(z.string()).optional(),
    visibility: z.nullable(z.union([
        components.ArticleVisibility$inboundSchema,
        z.array(components.ArticleVisibility$inboundSchema),
    ])).optional(),
    is_subscribed: z.nullable(z.boolean()).optional(),
    is_published: z.nullable(z.boolean()).optional(),
    is_pinned: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "is_subscribed": "isSubscribed",
        "is_published": "isPublished",
        "is_pinned": "isPinned",
    });
});
/** @internal */
exports.ArticlesListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    slug: z.nullable(z.string()).optional(),
    visibility: z.nullable(z.union([
        components.ArticleVisibility$outboundSchema,
        z.array(components.ArticleVisibility$outboundSchema),
    ])).optional(),
    isSubscribed: z.nullable(z.boolean()).optional(),
    isPublished: z.nullable(z.boolean()).optional(),
    isPinned: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        isSubscribed: "is_subscribed",
        isPublished: "is_published",
        isPinned: "is_pinned",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArticlesListRequest$;
(function (ArticlesListRequest$) {
    /** @deprecated use `ArticlesListRequest$inboundSchema` instead. */
    ArticlesListRequest$.inboundSchema = exports.ArticlesListRequest$inboundSchema;
    /** @deprecated use `ArticlesListRequest$outboundSchema` instead. */
    ArticlesListRequest$.outboundSchema = exports.ArticlesListRequest$outboundSchema;
})(ArticlesListRequest$ || (exports.ArticlesListRequest$ = ArticlesListRequest$ = {}));
/** @internal */
exports.ArticlesListResponse$inboundSchema = z.object({
    Result: components.ListResourceArticle$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.ArticlesListResponse$outboundSchema = z.object({
    result: components.ListResourceArticle$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ArticlesListResponse$;
(function (ArticlesListResponse$) {
    /** @deprecated use `ArticlesListResponse$inboundSchema` instead. */
    ArticlesListResponse$.inboundSchema = exports.ArticlesListResponse$inboundSchema;
    /** @deprecated use `ArticlesListResponse$outboundSchema` instead. */
    ArticlesListResponse$.outboundSchema = exports.ArticlesListResponse$outboundSchema;
})(ArticlesListResponse$ || (exports.ArticlesListResponse$ = ArticlesListResponse$ = {}));
