// @polar-sh/sdk/models/operations/benefitsgrants.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitsGrantsResponse$ = exports.BenefitsGrantsResponse$outboundSchema = exports.BenefitsGrantsResponse$inboundSchema = exports.BenefitsGrantsRequest$ = exports.BenefitsGrantsRequest$outboundSchema = exports.BenefitsGrantsRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BenefitsGrantsRequest$inboundSchema = z.object({
    id: z.string(),
    is_granted: z.nullable(z.boolean()).optional(),
    user_id: z.nullable(z.string()).optional(),
    github_user_id: z.nullable(z.number().int()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "is_granted": "isGranted",
        "user_id": "userId",
        "github_user_id": "githubUserId",
    });
});
/** @internal */
exports.BenefitsGrantsRequest$outboundSchema = z.object({
    id: z.string(),
    isGranted: z.nullable(z.boolean()).optional(),
    userId: z.nullable(z.string()).optional(),
    githubUserId: z.nullable(z.number().int()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        isGranted: "is_granted",
        userId: "user_id",
        githubUserId: "github_user_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitsGrantsRequest$;
(function (BenefitsGrantsRequest$) {
    /** @deprecated use `BenefitsGrantsRequest$inboundSchema` instead. */
    BenefitsGrantsRequest$.inboundSchema = exports.BenefitsGrantsRequest$inboundSchema;
    /** @deprecated use `BenefitsGrantsRequest$outboundSchema` instead. */
    BenefitsGrantsRequest$.outboundSchema = exports.BenefitsGrantsRequest$outboundSchema;
})(BenefitsGrantsRequest$ || (exports.BenefitsGrantsRequest$ = BenefitsGrantsRequest$ = {}));
/** @internal */
exports.BenefitsGrantsResponse$inboundSchema = z.object({
    Result: components.ListResourceBenefitGrant$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.BenefitsGrantsResponse$outboundSchema = z.object({
    result: components.ListResourceBenefitGrant$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitsGrantsResponse$;
(function (BenefitsGrantsResponse$) {
    /** @deprecated use `BenefitsGrantsResponse$inboundSchema` instead. */
    BenefitsGrantsResponse$.inboundSchema = exports.BenefitsGrantsResponse$inboundSchema;
    /** @deprecated use `BenefitsGrantsResponse$outboundSchema` instead. */
    BenefitsGrantsResponse$.outboundSchema = exports.BenefitsGrantsResponse$outboundSchema;
})(BenefitsGrantsResponse$ || (exports.BenefitsGrantsResponse$ = BenefitsGrantsResponse$ = {}));
