// @polar-sh/sdk/models/operations/benefitsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitsUpdateRequest$ = exports.BenefitsUpdateRequest$outboundSchema = exports.BenefitsUpdateRequest$inboundSchema = exports.BenefitsUpdateBenefitUpdate$ = exports.BenefitsUpdateBenefitUpdate$outboundSchema = exports.BenefitsUpdateBenefitUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BenefitsUpdateBenefitUpdate$inboundSchema = z.union([
    components.BenefitArticlesUpdate$inboundSchema,
    components.BenefitAdsUpdate$inboundSchema,
    components.BenefitCustomUpdate$inboundSchema,
    components.BenefitDiscordUpdate$inboundSchema,
    components.BenefitGitHubRepositoryUpdate$inboundSchema,
    components.BenefitDownloadablesUpdate$inboundSchema,
    components.BenefitLicenseKeysUpdate$inboundSchema,
]);
/** @internal */
exports.BenefitsUpdateBenefitUpdate$outboundSchema = z.union([
    components.BenefitArticlesUpdate$outboundSchema,
    components.BenefitAdsUpdate$outboundSchema,
    components.BenefitCustomUpdate$outboundSchema,
    components.BenefitDiscordUpdate$outboundSchema,
    components.BenefitGitHubRepositoryUpdate$outboundSchema,
    components.BenefitDownloadablesUpdate$outboundSchema,
    components.BenefitLicenseKeysUpdate$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitsUpdateBenefitUpdate$;
(function (BenefitsUpdateBenefitUpdate$) {
    /** @deprecated use `BenefitsUpdateBenefitUpdate$inboundSchema` instead. */
    BenefitsUpdateBenefitUpdate$.inboundSchema = exports.BenefitsUpdateBenefitUpdate$inboundSchema;
    /** @deprecated use `BenefitsUpdateBenefitUpdate$outboundSchema` instead. */
    BenefitsUpdateBenefitUpdate$.outboundSchema = exports.BenefitsUpdateBenefitUpdate$outboundSchema;
})(BenefitsUpdateBenefitUpdate$ || (exports.BenefitsUpdateBenefitUpdate$ = BenefitsUpdateBenefitUpdate$ = {}));
/** @internal */
exports.BenefitsUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    RequestBody: z.union([
        components.BenefitArticlesUpdate$inboundSchema,
        components.BenefitAdsUpdate$inboundSchema,
        components.BenefitCustomUpdate$inboundSchema,
        components.BenefitDiscordUpdate$inboundSchema,
        components.BenefitGitHubRepositoryUpdate$inboundSchema,
        components.BenefitDownloadablesUpdate$inboundSchema,
        components.BenefitLicenseKeysUpdate$inboundSchema,
    ]),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
exports.BenefitsUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    requestBody: z.union([
        components.BenefitArticlesUpdate$outboundSchema,
        components.BenefitAdsUpdate$outboundSchema,
        components.BenefitCustomUpdate$outboundSchema,
        components.BenefitDiscordUpdate$outboundSchema,
        components.BenefitGitHubRepositoryUpdate$outboundSchema,
        components.BenefitDownloadablesUpdate$outboundSchema,
        components.BenefitLicenseKeysUpdate$outboundSchema,
    ]),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitsUpdateRequest$;
(function (BenefitsUpdateRequest$) {
    /** @deprecated use `BenefitsUpdateRequest$inboundSchema` instead. */
    BenefitsUpdateRequest$.inboundSchema = exports.BenefitsUpdateRequest$inboundSchema;
    /** @deprecated use `BenefitsUpdateRequest$outboundSchema` instead. */
    BenefitsUpdateRequest$.outboundSchema = exports.BenefitsUpdateRequest$outboundSchema;
})(BenefitsUpdateRequest$ || (exports.BenefitsUpdateRequest$ = BenefitsUpdateRequest$ = {}));
