// @polar-sh/sdk/models/operations/benefitslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitsListResponse$ = exports.BenefitsListResponse$outboundSchema = exports.BenefitsListResponse$inboundSchema = exports.BenefitsListRequest$ = exports.BenefitsListRequest$outboundSchema = exports.BenefitsListRequest$inboundSchema = exports.QueryParamBenefitTypeFilter$ = exports.QueryParamBenefitTypeFilter$outboundSchema = exports.QueryParamBenefitTypeFilter$inboundSchema = exports.BenefitsListQueryParamOrganizationIDFilter$ = exports.BenefitsListQueryParamOrganizationIDFilter$outboundSchema = exports.BenefitsListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BenefitsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.BenefitsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitsListQueryParamOrganizationIDFilter$;
(function (BenefitsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `BenefitsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    BenefitsListQueryParamOrganizationIDFilter$.inboundSchema = exports.BenefitsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `BenefitsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    BenefitsListQueryParamOrganizationIDFilter$.outboundSchema = exports.BenefitsListQueryParamOrganizationIDFilter$outboundSchema;
})(BenefitsListQueryParamOrganizationIDFilter$ || (exports.BenefitsListQueryParamOrganizationIDFilter$ = BenefitsListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.QueryParamBenefitTypeFilter$inboundSchema = z.union([
    components.BenefitType$inboundSchema,
    z.array(components.BenefitType$inboundSchema),
]);
/** @internal */
exports.QueryParamBenefitTypeFilter$outboundSchema = z.union([
    components.BenefitType$outboundSchema,
    z.array(components.BenefitType$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamBenefitTypeFilter$;
(function (QueryParamBenefitTypeFilter$) {
    /** @deprecated use `QueryParamBenefitTypeFilter$inboundSchema` instead. */
    QueryParamBenefitTypeFilter$.inboundSchema = exports.QueryParamBenefitTypeFilter$inboundSchema;
    /** @deprecated use `QueryParamBenefitTypeFilter$outboundSchema` instead. */
    QueryParamBenefitTypeFilter$.outboundSchema = exports.QueryParamBenefitTypeFilter$outboundSchema;
})(QueryParamBenefitTypeFilter$ || (exports.QueryParamBenefitTypeFilter$ = QueryParamBenefitTypeFilter$ = {}));
/** @internal */
exports.BenefitsListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    type_filter: z.nullable(z.union([
        components.BenefitType$inboundSchema,
        z.array(components.BenefitType$inboundSchema),
    ])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "type_filter": "typeFilter",
    });
});
/** @internal */
exports.BenefitsListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    typeFilter: z.nullable(z.union([
        components.BenefitType$outboundSchema,
        z.array(components.BenefitType$outboundSchema),
    ])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        typeFilter: "type_filter",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitsListRequest$;
(function (BenefitsListRequest$) {
    /** @deprecated use `BenefitsListRequest$inboundSchema` instead. */
    BenefitsListRequest$.inboundSchema = exports.BenefitsListRequest$inboundSchema;
    /** @deprecated use `BenefitsListRequest$outboundSchema` instead. */
    BenefitsListRequest$.outboundSchema = exports.BenefitsListRequest$outboundSchema;
})(BenefitsListRequest$ || (exports.BenefitsListRequest$ = BenefitsListRequest$ = {}));
/** @internal */
exports.BenefitsListResponse$inboundSchema = z.object({
    Result: components.ListResourceBenefit$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.BenefitsListResponse$outboundSchema = z.object({
    result: components.ListResourceBenefit$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitsListResponse$;
(function (BenefitsListResponse$) {
    /** @deprecated use `BenefitsListResponse$inboundSchema` instead. */
    BenefitsListResponse$.inboundSchema = exports.BenefitsListResponse$inboundSchema;
    /** @deprecated use `BenefitsListResponse$outboundSchema` instead. */
    BenefitsListResponse$.outboundSchema = exports.BenefitsListResponse$outboundSchema;
})(BenefitsListResponse$ || (exports.BenefitsListResponse$ = BenefitsListResponse$ = {}));
