// @polar-sh/sdk/models/operations/checkoutscustomlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutsCustomListResponse$ = exports.CheckoutsCustomListResponse$outboundSchema = exports.CheckoutsCustomListResponse$inboundSchema = exports.CheckoutsCustomListRequest$ = exports.CheckoutsCustomListRequest$outboundSchema = exports.CheckoutsCustomListRequest$inboundSchema = exports.CheckoutsCustomListQueryParamProductIDFilter$ = exports.CheckoutsCustomListQueryParamProductIDFilter$outboundSchema = exports.CheckoutsCustomListQueryParamProductIDFilter$inboundSchema = exports.CheckoutsCustomListQueryParamOrganizationIDFilter$ = exports.CheckoutsCustomListQueryParamOrganizationIDFilter$outboundSchema = exports.CheckoutsCustomListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CheckoutsCustomListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutsCustomListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsCustomListQueryParamOrganizationIDFilter$;
(function (CheckoutsCustomListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `CheckoutsCustomListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    CheckoutsCustomListQueryParamOrganizationIDFilter$.inboundSchema = exports.CheckoutsCustomListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `CheckoutsCustomListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    CheckoutsCustomListQueryParamOrganizationIDFilter$.outboundSchema = exports.CheckoutsCustomListQueryParamOrganizationIDFilter$outboundSchema;
})(CheckoutsCustomListQueryParamOrganizationIDFilter$ || (exports.CheckoutsCustomListQueryParamOrganizationIDFilter$ = CheckoutsCustomListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.CheckoutsCustomListQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutsCustomListQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsCustomListQueryParamProductIDFilter$;
(function (CheckoutsCustomListQueryParamProductIDFilter$) {
    /** @deprecated use `CheckoutsCustomListQueryParamProductIDFilter$inboundSchema` instead. */
    CheckoutsCustomListQueryParamProductIDFilter$.inboundSchema = exports.CheckoutsCustomListQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `CheckoutsCustomListQueryParamProductIDFilter$outboundSchema` instead. */
    CheckoutsCustomListQueryParamProductIDFilter$.outboundSchema = exports.CheckoutsCustomListQueryParamProductIDFilter$outboundSchema;
})(CheckoutsCustomListQueryParamProductIDFilter$ || (exports.CheckoutsCustomListQueryParamProductIDFilter$ = CheckoutsCustomListQueryParamProductIDFilter$ = {}));
/** @internal */
exports.CheckoutsCustomListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.CheckoutSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
    });
});
/** @internal */
exports.CheckoutsCustomListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.CheckoutSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsCustomListRequest$;
(function (CheckoutsCustomListRequest$) {
    /** @deprecated use `CheckoutsCustomListRequest$inboundSchema` instead. */
    CheckoutsCustomListRequest$.inboundSchema = exports.CheckoutsCustomListRequest$inboundSchema;
    /** @deprecated use `CheckoutsCustomListRequest$outboundSchema` instead. */
    CheckoutsCustomListRequest$.outboundSchema = exports.CheckoutsCustomListRequest$outboundSchema;
})(CheckoutsCustomListRequest$ || (exports.CheckoutsCustomListRequest$ = CheckoutsCustomListRequest$ = {}));
/** @internal */
exports.CheckoutsCustomListResponse$inboundSchema = z.object({
    Result: components.ListResourceCheckout$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.CheckoutsCustomListResponse$outboundSchema = z.object({
    result: components.ListResourceCheckout$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutsCustomListResponse$;
(function (CheckoutsCustomListResponse$) {
    /** @deprecated use `CheckoutsCustomListResponse$inboundSchema` instead. */
    CheckoutsCustomListResponse$.inboundSchema = exports.CheckoutsCustomListResponse$inboundSchema;
    /** @deprecated use `CheckoutsCustomListResponse$outboundSchema` instead. */
    CheckoutsCustomListResponse$.outboundSchema = exports.CheckoutsCustomListResponse$outboundSchema;
})(CheckoutsCustomListResponse$ || (exports.CheckoutsCustomListResponse$ = CheckoutsCustomListResponse$ = {}));
