// @polar-sh/sdk/models/operations/checkoutlinkslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinksListResponse$ = exports.CheckoutLinksListResponse$outboundSchema = exports.CheckoutLinksListResponse$inboundSchema = exports.CheckoutLinksListRequest$ = exports.CheckoutLinksListRequest$outboundSchema = exports.CheckoutLinksListRequest$inboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$ = exports.CheckoutLinksListQueryParamProductIDFilter$outboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$inboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$ = exports.CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListQueryParamOrganizationIDFilter$;
(function (CheckoutLinksListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    CheckoutLinksListQueryParamOrganizationIDFilter$.inboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    CheckoutLinksListQueryParamOrganizationIDFilter$.outboundSchema = exports.CheckoutLinksListQueryParamOrganizationIDFilter$outboundSchema;
})(CheckoutLinksListQueryParamOrganizationIDFilter$ || (exports.CheckoutLinksListQueryParamOrganizationIDFilter$ = CheckoutLinksListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.CheckoutLinksListQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.CheckoutLinksListQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListQueryParamProductIDFilter$;
(function (CheckoutLinksListQueryParamProductIDFilter$) {
    /** @deprecated use `CheckoutLinksListQueryParamProductIDFilter$inboundSchema` instead. */
    CheckoutLinksListQueryParamProductIDFilter$.inboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `CheckoutLinksListQueryParamProductIDFilter$outboundSchema` instead. */
    CheckoutLinksListQueryParamProductIDFilter$.outboundSchema = exports.CheckoutLinksListQueryParamProductIDFilter$outboundSchema;
})(CheckoutLinksListQueryParamProductIDFilter$ || (exports.CheckoutLinksListQueryParamProductIDFilter$ = CheckoutLinksListQueryParamProductIDFilter$ = {}));
/** @internal */
exports.CheckoutLinksListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.CheckoutLinkSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
    });
});
/** @internal */
exports.CheckoutLinksListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.CheckoutLinkSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListRequest$;
(function (CheckoutLinksListRequest$) {
    /** @deprecated use `CheckoutLinksListRequest$inboundSchema` instead. */
    CheckoutLinksListRequest$.inboundSchema = exports.CheckoutLinksListRequest$inboundSchema;
    /** @deprecated use `CheckoutLinksListRequest$outboundSchema` instead. */
    CheckoutLinksListRequest$.outboundSchema = exports.CheckoutLinksListRequest$outboundSchema;
})(CheckoutLinksListRequest$ || (exports.CheckoutLinksListRequest$ = CheckoutLinksListRequest$ = {}));
/** @internal */
exports.CheckoutLinksListResponse$inboundSchema = z.object({
    Result: components.ListResourceCheckoutLink$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.CheckoutLinksListResponse$outboundSchema = z.object({
    result: components.ListResourceCheckoutLink$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinksListResponse$;
(function (CheckoutLinksListResponse$) {
    /** @deprecated use `CheckoutLinksListResponse$inboundSchema` instead. */
    CheckoutLinksListResponse$.inboundSchema = exports.CheckoutLinksListResponse$inboundSchema;
    /** @deprecated use `CheckoutLinksListResponse$outboundSchema` instead. */
    CheckoutLinksListResponse$.outboundSchema = exports.CheckoutLinksListResponse$outboundSchema;
})(CheckoutLinksListResponse$ || (exports.CheckoutLinksListResponse$ = CheckoutLinksListResponse$ = {}));
