// @polar-sh/sdk/models/operations/externalorganizationslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalOrganizationsListResponse$ = exports.ExternalOrganizationsListResponse$outboundSchema = exports.ExternalOrganizationsListResponse$inboundSchema = exports.ExternalOrganizationsListRequest$ = exports.ExternalOrganizationsListRequest$outboundSchema = exports.ExternalOrganizationsListRequest$inboundSchema = exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$ = exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$outboundSchema = exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$inboundSchema = exports.RepositoryNameFilter$ = exports.RepositoryNameFilter$outboundSchema = exports.RepositoryNameFilter$inboundSchema = exports.PlatformFilter$ = exports.PlatformFilter$outboundSchema = exports.PlatformFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.PlatformFilter$inboundSchema = z.union([
    components.Platforms$inboundSchema,
    z.array(components.Platforms$inboundSchema),
]);
/** @internal */
exports.PlatformFilter$outboundSchema = z.union([
    components.Platforms$outboundSchema,
    z.array(components.Platforms$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PlatformFilter$;
(function (PlatformFilter$) {
    /** @deprecated use `PlatformFilter$inboundSchema` instead. */
    PlatformFilter$.inboundSchema = exports.PlatformFilter$inboundSchema;
    /** @deprecated use `PlatformFilter$outboundSchema` instead. */
    PlatformFilter$.outboundSchema = exports.PlatformFilter$outboundSchema;
})(PlatformFilter$ || (exports.PlatformFilter$ = PlatformFilter$ = {}));
/** @internal */
exports.RepositoryNameFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.RepositoryNameFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RepositoryNameFilter$;
(function (RepositoryNameFilter$) {
    /** @deprecated use `RepositoryNameFilter$inboundSchema` instead. */
    RepositoryNameFilter$.inboundSchema = exports.RepositoryNameFilter$inboundSchema;
    /** @deprecated use `RepositoryNameFilter$outboundSchema` instead. */
    RepositoryNameFilter$.outboundSchema = exports.RepositoryNameFilter$outboundSchema;
})(RepositoryNameFilter$ || (exports.RepositoryNameFilter$ = RepositoryNameFilter$ = {}));
/** @internal */
exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExternalOrganizationsListQueryParamOrganizationIDFilter$;
(function (ExternalOrganizationsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `ExternalOrganizationsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    ExternalOrganizationsListQueryParamOrganizationIDFilter$.inboundSchema = exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `ExternalOrganizationsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    ExternalOrganizationsListQueryParamOrganizationIDFilter$.outboundSchema = exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$outboundSchema;
})(ExternalOrganizationsListQueryParamOrganizationIDFilter$ || (exports.ExternalOrganizationsListQueryParamOrganizationIDFilter$ = ExternalOrganizationsListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.ExternalOrganizationsListRequest$inboundSchema = z.object({
    platform: z.nullable(z.union([
        components.Platforms$inboundSchema,
        z.array(components.Platforms$inboundSchema),
    ])).optional(),
    name: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.ExternalOrganizationSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ExternalOrganizationsListRequest$outboundSchema = z.object({
    platform: z.nullable(z.union([
        components.Platforms$outboundSchema,
        z.array(components.Platforms$outboundSchema),
    ])).optional(),
    name: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.ExternalOrganizationSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExternalOrganizationsListRequest$;
(function (ExternalOrganizationsListRequest$) {
    /** @deprecated use `ExternalOrganizationsListRequest$inboundSchema` instead. */
    ExternalOrganizationsListRequest$.inboundSchema = exports.ExternalOrganizationsListRequest$inboundSchema;
    /** @deprecated use `ExternalOrganizationsListRequest$outboundSchema` instead. */
    ExternalOrganizationsListRequest$.outboundSchema = exports.ExternalOrganizationsListRequest$outboundSchema;
})(ExternalOrganizationsListRequest$ || (exports.ExternalOrganizationsListRequest$ = ExternalOrganizationsListRequest$ = {}));
/** @internal */
exports.ExternalOrganizationsListResponse$inboundSchema = z.object({
    Result: components.ListResourceExternalOrganization$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.ExternalOrganizationsListResponse$outboundSchema = z.object({
    result: components.ListResourceExternalOrganization$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExternalOrganizationsListResponse$;
(function (ExternalOrganizationsListResponse$) {
    /** @deprecated use `ExternalOrganizationsListResponse$inboundSchema` instead. */
    ExternalOrganizationsListResponse$.inboundSchema = exports.ExternalOrganizationsListResponse$inboundSchema;
    /** @deprecated use `ExternalOrganizationsListResponse$outboundSchema` instead. */
    ExternalOrganizationsListResponse$.outboundSchema = exports.ExternalOrganizationsListResponse$outboundSchema;
})(ExternalOrganizationsListResponse$ || (exports.ExternalOrganizationsListResponse$ = ExternalOrganizationsListResponse$ = {}));
