// @polar-sh/sdk/models/operations/fileslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesListResponse$ = exports.FilesListResponse$outboundSchema = exports.FilesListResponse$inboundSchema = exports.FilesListRequest$ = exports.FilesListRequest$outboundSchema = exports.FilesListRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.FilesListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.string()).optional(),
    ids: z.nullable(z.array(z.string())).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.FilesListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.string()).optional(),
    ids: z.nullable(z.array(z.string())).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesListRequest$;
(function (FilesListRequest$) {
    /** @deprecated use `FilesListRequest$inboundSchema` instead. */
    FilesListRequest$.inboundSchema = exports.FilesListRequest$inboundSchema;
    /** @deprecated use `FilesListRequest$outboundSchema` instead. */
    FilesListRequest$.outboundSchema = exports.FilesListRequest$outboundSchema;
})(FilesListRequest$ || (exports.FilesListRequest$ = FilesListRequest$ = {}));
/** @internal */
exports.FilesListResponse$inboundSchema = z.object({
    Result: components.ListResourceFileRead$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.FilesListResponse$outboundSchema = z.object({
    result: components.ListResourceFileRead$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesListResponse$;
(function (FilesListResponse$) {
    /** @deprecated use `FilesListResponse$inboundSchema` instead. */
    FilesListResponse$.inboundSchema = exports.FilesListResponse$inboundSchema;
    /** @deprecated use `FilesListResponse$outboundSchema` instead. */
    FilesListResponse$.outboundSchema = exports.FilesListResponse$outboundSchema;
})(FilesListResponse$ || (exports.FilesListResponse$ = FilesListResponse$ = {}));
