// @polar-sh/sdk/models/operations/filesupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesUpdateResponseFilesUpdate$ = exports.FilesUpdateResponseFilesUpdate$outboundSchema = exports.FilesUpdateResponseFilesUpdate$inboundSchema = exports.FilesUpdateRequest$ = exports.FilesUpdateRequest$outboundSchema = exports.FilesUpdateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.FilesUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    FilePatch: components.FilePatch$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "FilePatch": "filePatch",
    });
});
/** @internal */
exports.FilesUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    filePatch: components.FilePatch$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        filePatch: "FilePatch",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUpdateRequest$;
(function (FilesUpdateRequest$) {
    /** @deprecated use `FilesUpdateRequest$inboundSchema` instead. */
    FilesUpdateRequest$.inboundSchema = exports.FilesUpdateRequest$inboundSchema;
    /** @deprecated use `FilesUpdateRequest$outboundSchema` instead. */
    FilesUpdateRequest$.outboundSchema = exports.FilesUpdateRequest$outboundSchema;
})(FilesUpdateRequest$ || (exports.FilesUpdateRequest$ = FilesUpdateRequest$ = {}));
/** @internal */
exports.FilesUpdateResponseFilesUpdate$inboundSchema = z.union([
    components.DownloadableFileRead$inboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
    components.ProductMediaFileRead$inboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    components.OrganizationAvatarFileRead$inboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
]);
/** @internal */
exports.FilesUpdateResponseFilesUpdate$outboundSchema = z.union([
    components.DownloadableFileRead$outboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
    components.ProductMediaFileRead$outboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    components.OrganizationAvatarFileRead$outboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUpdateResponseFilesUpdate$;
(function (FilesUpdateResponseFilesUpdate$) {
    /** @deprecated use `FilesUpdateResponseFilesUpdate$inboundSchema` instead. */
    FilesUpdateResponseFilesUpdate$.inboundSchema = exports.FilesUpdateResponseFilesUpdate$inboundSchema;
    /** @deprecated use `FilesUpdateResponseFilesUpdate$outboundSchema` instead. */
    FilesUpdateResponseFilesUpdate$.outboundSchema = exports.FilesUpdateResponseFilesUpdate$outboundSchema;
})(FilesUpdateResponseFilesUpdate$ || (exports.FilesUpdateResponseFilesUpdate$ = FilesUpdateResponseFilesUpdate$ = {}));
