// @polar-sh/sdk/models/operations/filesuploaded.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesUploadedResponseFilesUploaded$ = exports.FilesUploadedResponseFilesUploaded$outboundSchema = exports.FilesUploadedResponseFilesUploaded$inboundSchema = exports.FilesUploadedRequest$ = exports.FilesUploadedRequest$outboundSchema = exports.FilesUploadedRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.FilesUploadedRequest$inboundSchema = z.object({
    id: z.string(),
    FileUploadCompleted: components.FileUploadCompleted$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "FileUploadCompleted": "fileUploadCompleted",
    });
});
/** @internal */
exports.FilesUploadedRequest$outboundSchema = z.object({
    id: z.string(),
    fileUploadCompleted: components.FileUploadCompleted$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fileUploadCompleted: "FileUploadCompleted",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUploadedRequest$;
(function (FilesUploadedRequest$) {
    /** @deprecated use `FilesUploadedRequest$inboundSchema` instead. */
    FilesUploadedRequest$.inboundSchema = exports.FilesUploadedRequest$inboundSchema;
    /** @deprecated use `FilesUploadedRequest$outboundSchema` instead. */
    FilesUploadedRequest$.outboundSchema = exports.FilesUploadedRequest$outboundSchema;
})(FilesUploadedRequest$ || (exports.FilesUploadedRequest$ = FilesUploadedRequest$ = {}));
/** @internal */
exports.FilesUploadedResponseFilesUploaded$inboundSchema = z.union([
    components.DownloadableFileRead$inboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
    components.ProductMediaFileRead$inboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    components.OrganizationAvatarFileRead$inboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
]);
/** @internal */
exports.FilesUploadedResponseFilesUploaded$outboundSchema = z.union([
    components.DownloadableFileRead$outboundSchema.and(z.object({ service: z.literal("downloadable") }).transform((v) => ({
        service: v.service,
    }))),
    components.ProductMediaFileRead$outboundSchema.and(z.object({ service: z.literal("product_media") }).transform((v) => ({
        service: v.service,
    }))),
    components.OrganizationAvatarFileRead$outboundSchema.and(z.object({ service: z.literal("organization_avatar") }).transform((v) => ({
        service: v.service,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesUploadedResponseFilesUploaded$;
(function (FilesUploadedResponseFilesUploaded$) {
    /** @deprecated use `FilesUploadedResponseFilesUploaded$inboundSchema` instead. */
    FilesUploadedResponseFilesUploaded$.inboundSchema = exports.FilesUploadedResponseFilesUploaded$inboundSchema;
    /** @deprecated use `FilesUploadedResponseFilesUploaded$outboundSchema` instead. */
    FilesUploadedResponseFilesUploaded$.outboundSchema = exports.FilesUploadedResponseFilesUploaded$outboundSchema;
})(FilesUploadedResponseFilesUploaded$ || (exports.FilesUploadedResponseFilesUploaded$ = FilesUploadedResponseFilesUploaded$ = {}));
