// @polar-sh/sdk/models/operations/metricsget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricsGetRequest$ = exports.MetricsGetRequest$outboundSchema = exports.MetricsGetRequest$inboundSchema = exports.MetricsGetQueryParamProductPriceTypeFilter$ = exports.MetricsGetQueryParamProductPriceTypeFilter$outboundSchema = exports.MetricsGetQueryParamProductPriceTypeFilter$inboundSchema = exports.MetricsGetQueryParamProductIDFilter$ = exports.MetricsGetQueryParamProductIDFilter$outboundSchema = exports.MetricsGetQueryParamProductIDFilter$inboundSchema = exports.MetricsGetQueryParamOrganizationIDFilter$ = exports.MetricsGetQueryParamOrganizationIDFilter$outboundSchema = exports.MetricsGetQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const rfcdate_js_1 = require("../../types/rfcdate.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.MetricsGetQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.MetricsGetQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MetricsGetQueryParamOrganizationIDFilter$;
(function (MetricsGetQueryParamOrganizationIDFilter$) {
    /** @deprecated use `MetricsGetQueryParamOrganizationIDFilter$inboundSchema` instead. */
    MetricsGetQueryParamOrganizationIDFilter$.inboundSchema = exports.MetricsGetQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `MetricsGetQueryParamOrganizationIDFilter$outboundSchema` instead. */
    MetricsGetQueryParamOrganizationIDFilter$.outboundSchema = exports.MetricsGetQueryParamOrganizationIDFilter$outboundSchema;
})(MetricsGetQueryParamOrganizationIDFilter$ || (exports.MetricsGetQueryParamOrganizationIDFilter$ = MetricsGetQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.MetricsGetQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.MetricsGetQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MetricsGetQueryParamProductIDFilter$;
(function (MetricsGetQueryParamProductIDFilter$) {
    /** @deprecated use `MetricsGetQueryParamProductIDFilter$inboundSchema` instead. */
    MetricsGetQueryParamProductIDFilter$.inboundSchema = exports.MetricsGetQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `MetricsGetQueryParamProductIDFilter$outboundSchema` instead. */
    MetricsGetQueryParamProductIDFilter$.outboundSchema = exports.MetricsGetQueryParamProductIDFilter$outboundSchema;
})(MetricsGetQueryParamProductIDFilter$ || (exports.MetricsGetQueryParamProductIDFilter$ = MetricsGetQueryParamProductIDFilter$ = {}));
/** @internal */
exports.MetricsGetQueryParamProductPriceTypeFilter$inboundSchema = z.union([
    components.ProductPriceType$inboundSchema,
    z.array(components.ProductPriceType$inboundSchema),
]);
/** @internal */
exports.MetricsGetQueryParamProductPriceTypeFilter$outboundSchema = z.union([
    components.ProductPriceType$outboundSchema,
    z.array(components.ProductPriceType$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MetricsGetQueryParamProductPriceTypeFilter$;
(function (MetricsGetQueryParamProductPriceTypeFilter$) {
    /** @deprecated use `MetricsGetQueryParamProductPriceTypeFilter$inboundSchema` instead. */
    MetricsGetQueryParamProductPriceTypeFilter$.inboundSchema = exports.MetricsGetQueryParamProductPriceTypeFilter$inboundSchema;
    /** @deprecated use `MetricsGetQueryParamProductPriceTypeFilter$outboundSchema` instead. */
    MetricsGetQueryParamProductPriceTypeFilter$.outboundSchema = exports.MetricsGetQueryParamProductPriceTypeFilter$outboundSchema;
})(MetricsGetQueryParamProductPriceTypeFilter$ || (exports.MetricsGetQueryParamProductPriceTypeFilter$ = MetricsGetQueryParamProductPriceTypeFilter$ = {}));
/** @internal */
exports.MetricsGetRequest$inboundSchema = z.object({
    start_date: z.string().transform(v => new rfcdate_js_1.RFCDate(v)),
    end_date: z.string().transform(v => new rfcdate_js_1.RFCDate(v)),
    interval: components.Interval$inboundSchema,
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    product_price_type: z.nullable(z.union([
        components.ProductPriceType$inboundSchema,
        z.array(components.ProductPriceType$inboundSchema),
    ])).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "start_date": "startDate",
        "end_date": "endDate",
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_price_type": "productPriceType",
    });
});
/** @internal */
exports.MetricsGetRequest$outboundSchema = z.object({
    startDate: z.instanceof(rfcdate_js_1.RFCDate).transform(v => v.toString()),
    endDate: z.instanceof(rfcdate_js_1.RFCDate).transform(v => v.toString()),
    interval: components.Interval$outboundSchema,
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    productPriceType: z.nullable(z.union([
        components.ProductPriceType$outboundSchema,
        z.array(components.ProductPriceType$outboundSchema),
    ])).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        startDate: "start_date",
        endDate: "end_date",
        organizationId: "organization_id",
        productId: "product_id",
        productPriceType: "product_price_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MetricsGetRequest$;
(function (MetricsGetRequest$) {
    /** @deprecated use `MetricsGetRequest$inboundSchema` instead. */
    MetricsGetRequest$.inboundSchema = exports.MetricsGetRequest$inboundSchema;
    /** @deprecated use `MetricsGetRequest$outboundSchema` instead. */
    MetricsGetRequest$.outboundSchema = exports.MetricsGetRequest$outboundSchema;
})(MetricsGetRequest$ || (exports.MetricsGetRequest$ = MetricsGetRequest$ = {}));
