// @polar-sh/sdk/models/operations/oauth2clientslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Oauth2ClientsListResponse$ = exports.Oauth2ClientsListResponse$outboundSchema = exports.Oauth2ClientsListResponse$inboundSchema = exports.Oauth2ClientsListRequest$ = exports.Oauth2ClientsListRequest$outboundSchema = exports.Oauth2ClientsListRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.Oauth2ClientsListRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
});
/** @internal */
exports.Oauth2ClientsListRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Oauth2ClientsListRequest$;
(function (Oauth2ClientsListRequest$) {
    /** @deprecated use `Oauth2ClientsListRequest$inboundSchema` instead. */
    Oauth2ClientsListRequest$.inboundSchema = exports.Oauth2ClientsListRequest$inboundSchema;
    /** @deprecated use `Oauth2ClientsListRequest$outboundSchema` instead. */
    Oauth2ClientsListRequest$.outboundSchema = exports.Oauth2ClientsListRequest$outboundSchema;
})(Oauth2ClientsListRequest$ || (exports.Oauth2ClientsListRequest$ = Oauth2ClientsListRequest$ = {}));
/** @internal */
exports.Oauth2ClientsListResponse$inboundSchema = z.object({
    Result: components.ListResourceOAuth2Client$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.Oauth2ClientsListResponse$outboundSchema = z.object({
    result: components.ListResourceOAuth2Client$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Oauth2ClientsListResponse$;
(function (Oauth2ClientsListResponse$) {
    /** @deprecated use `Oauth2ClientsListResponse$inboundSchema` instead. */
    Oauth2ClientsListResponse$.inboundSchema = exports.Oauth2ClientsListResponse$inboundSchema;
    /** @deprecated use `Oauth2ClientsListResponse$outboundSchema` instead. */
    Oauth2ClientsListResponse$.outboundSchema = exports.Oauth2ClientsListResponse$outboundSchema;
})(Oauth2ClientsListResponse$ || (exports.Oauth2ClientsListResponse$ = Oauth2ClientsListResponse$ = {}));
