// @polar-sh/sdk/models/operations/oauth2clientsoauth2updateclient.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Oauth2ClientsOauth2UpdateClientRequest$ = exports.Oauth2ClientsOauth2UpdateClientRequest$outboundSchema = exports.Oauth2ClientsOauth2UpdateClientRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.Oauth2ClientsOauth2UpdateClientRequest$inboundSchema = z.object({
    client_id: z.string(),
    OAuth2ClientConfigurationUpdate: components.OAuth2ClientConfigurationUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_id": "clientId",
        "OAuth2ClientConfigurationUpdate": "oAuth2ClientConfigurationUpdate",
    });
});
/** @internal */
exports.Oauth2ClientsOauth2UpdateClientRequest$outboundSchema = z.object({
    clientId: z.string(),
    oAuth2ClientConfigurationUpdate: components.OAuth2ClientConfigurationUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientId: "client_id",
        oAuth2ClientConfigurationUpdate: "OAuth2ClientConfigurationUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Oauth2ClientsOauth2UpdateClientRequest$;
(function (Oauth2ClientsOauth2UpdateClientRequest$) {
    /** @deprecated use `Oauth2ClientsOauth2UpdateClientRequest$inboundSchema` instead. */
    Oauth2ClientsOauth2UpdateClientRequest$.inboundSchema = exports.Oauth2ClientsOauth2UpdateClientRequest$inboundSchema;
    /** @deprecated use `Oauth2ClientsOauth2UpdateClientRequest$outboundSchema` instead. */
    Oauth2ClientsOauth2UpdateClientRequest$.outboundSchema = exports.Oauth2ClientsOauth2UpdateClientRequest$outboundSchema;
})(Oauth2ClientsOauth2UpdateClientRequest$ || (exports.Oauth2ClientsOauth2UpdateClientRequest$ = Oauth2ClientsOauth2UpdateClientRequest$ = {}));
