// @polar-sh/sdk/models/operations/orderslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrdersListResponse$ = exports.OrdersListResponse$outboundSchema = exports.OrdersListResponse$inboundSchema = exports.OrdersListRequest$ = exports.OrdersListRequest$outboundSchema = exports.OrdersListRequest$inboundSchema = exports.UserIDFilter$ = exports.UserIDFilter$outboundSchema = exports.UserIDFilter$inboundSchema = exports.QueryParamProductPriceTypeFilter$ = exports.QueryParamProductPriceTypeFilter$outboundSchema = exports.QueryParamProductPriceTypeFilter$inboundSchema = exports.OrdersListQueryParamProductIDFilter$ = exports.OrdersListQueryParamProductIDFilter$outboundSchema = exports.OrdersListQueryParamProductIDFilter$inboundSchema = exports.OrdersListQueryParamOrganizationIDFilter$ = exports.OrdersListQueryParamOrganizationIDFilter$outboundSchema = exports.OrdersListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.OrdersListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrdersListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersListQueryParamOrganizationIDFilter$;
(function (OrdersListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `OrdersListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    OrdersListQueryParamOrganizationIDFilter$.inboundSchema = exports.OrdersListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `OrdersListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    OrdersListQueryParamOrganizationIDFilter$.outboundSchema = exports.OrdersListQueryParamOrganizationIDFilter$outboundSchema;
})(OrdersListQueryParamOrganizationIDFilter$ || (exports.OrdersListQueryParamOrganizationIDFilter$ = OrdersListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.OrdersListQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrdersListQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersListQueryParamProductIDFilter$;
(function (OrdersListQueryParamProductIDFilter$) {
    /** @deprecated use `OrdersListQueryParamProductIDFilter$inboundSchema` instead. */
    OrdersListQueryParamProductIDFilter$.inboundSchema = exports.OrdersListQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `OrdersListQueryParamProductIDFilter$outboundSchema` instead. */
    OrdersListQueryParamProductIDFilter$.outboundSchema = exports.OrdersListQueryParamProductIDFilter$outboundSchema;
})(OrdersListQueryParamProductIDFilter$ || (exports.OrdersListQueryParamProductIDFilter$ = OrdersListQueryParamProductIDFilter$ = {}));
/** @internal */
exports.QueryParamProductPriceTypeFilter$inboundSchema = z.union([
    components.ProductPriceType$inboundSchema,
    z.array(components.ProductPriceType$inboundSchema),
]);
/** @internal */
exports.QueryParamProductPriceTypeFilter$outboundSchema = z.union([
    components.ProductPriceType$outboundSchema,
    z.array(components.ProductPriceType$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamProductPriceTypeFilter$;
(function (QueryParamProductPriceTypeFilter$) {
    /** @deprecated use `QueryParamProductPriceTypeFilter$inboundSchema` instead. */
    QueryParamProductPriceTypeFilter$.inboundSchema = exports.QueryParamProductPriceTypeFilter$inboundSchema;
    /** @deprecated use `QueryParamProductPriceTypeFilter$outboundSchema` instead. */
    QueryParamProductPriceTypeFilter$.outboundSchema = exports.QueryParamProductPriceTypeFilter$outboundSchema;
})(QueryParamProductPriceTypeFilter$ || (exports.QueryParamProductPriceTypeFilter$ = QueryParamProductPriceTypeFilter$ = {}));
/** @internal */
exports.UserIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.UserIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserIDFilter$;
(function (UserIDFilter$) {
    /** @deprecated use `UserIDFilter$inboundSchema` instead. */
    UserIDFilter$.inboundSchema = exports.UserIDFilter$inboundSchema;
    /** @deprecated use `UserIDFilter$outboundSchema` instead. */
    UserIDFilter$.outboundSchema = exports.UserIDFilter$outboundSchema;
})(UserIDFilter$ || (exports.UserIDFilter$ = UserIDFilter$ = {}));
/** @internal */
exports.OrdersListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    product_price_type: z.nullable(z.union([
        components.ProductPriceType$inboundSchema,
        z.array(components.ProductPriceType$inboundSchema),
    ])).optional(),
    user_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.OrderSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_price_type": "productPriceType",
        "user_id": "userId",
    });
});
/** @internal */
exports.OrdersListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    productPriceType: z.nullable(z.union([
        components.ProductPriceType$outboundSchema,
        z.array(components.ProductPriceType$outboundSchema),
    ])).optional(),
    userId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.OrderSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
        productPriceType: "product_price_type",
        userId: "user_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersListRequest$;
(function (OrdersListRequest$) {
    /** @deprecated use `OrdersListRequest$inboundSchema` instead. */
    OrdersListRequest$.inboundSchema = exports.OrdersListRequest$inboundSchema;
    /** @deprecated use `OrdersListRequest$outboundSchema` instead. */
    OrdersListRequest$.outboundSchema = exports.OrdersListRequest$outboundSchema;
})(OrdersListRequest$ || (exports.OrdersListRequest$ = OrdersListRequest$ = {}));
/** @internal */
exports.OrdersListResponse$inboundSchema = z.object({
    Result: components.ListResourceOrder$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.OrdersListResponse$outboundSchema = z.object({
    result: components.ListResourceOrder$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersListResponse$;
(function (OrdersListResponse$) {
    /** @deprecated use `OrdersListResponse$inboundSchema` instead. */
    OrdersListResponse$.inboundSchema = exports.OrdersListResponse$inboundSchema;
    /** @deprecated use `OrdersListResponse$outboundSchema` instead. */
    OrdersListResponse$.outboundSchema = exports.OrdersListResponse$outboundSchema;
})(OrdersListResponse$ || (exports.OrdersListResponse$ = OrdersListResponse$ = {}));
