// @polar-sh/sdk/models/operations/organizationslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationsListResponse$ = exports.OrganizationsListResponse$outboundSchema = exports.OrganizationsListResponse$inboundSchema = exports.OrganizationsListRequest$ = exports.OrganizationsListRequest$outboundSchema = exports.OrganizationsListRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.OrganizationsListRequest$inboundSchema = z.object({
    slug: z.nullable(z.string()).optional(),
    is_member: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.OrganizationSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "is_member": "isMember",
    });
});
/** @internal */
exports.OrganizationsListRequest$outboundSchema = z.object({
    slug: z.nullable(z.string()).optional(),
    isMember: z.nullable(z.boolean()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.OrganizationSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        isMember: "is_member",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationsListRequest$;
(function (OrganizationsListRequest$) {
    /** @deprecated use `OrganizationsListRequest$inboundSchema` instead. */
    OrganizationsListRequest$.inboundSchema = exports.OrganizationsListRequest$inboundSchema;
    /** @deprecated use `OrganizationsListRequest$outboundSchema` instead. */
    OrganizationsListRequest$.outboundSchema = exports.OrganizationsListRequest$outboundSchema;
})(OrganizationsListRequest$ || (exports.OrganizationsListRequest$ = OrganizationsListRequest$ = {}));
/** @internal */
exports.OrganizationsListResponse$inboundSchema = z.object({
    Result: components.ListResourceOrganization$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.OrganizationsListResponse$outboundSchema = z.object({
    result: components.ListResourceOrganization$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationsListResponse$;
(function (OrganizationsListResponse$) {
    /** @deprecated use `OrganizationsListResponse$inboundSchema` instead. */
    OrganizationsListResponse$.inboundSchema = exports.OrganizationsListResponse$inboundSchema;
    /** @deprecated use `OrganizationsListResponse$outboundSchema` instead. */
    OrganizationsListResponse$.outboundSchema = exports.OrganizationsListResponse$outboundSchema;
})(OrganizationsListResponse$ || (exports.OrganizationsListResponse$ = OrganizationsListResponse$ = {}));
