// @polar-sh/sdk/models/operations/productslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductsListResponse$ = exports.ProductsListResponse$outboundSchema = exports.ProductsListResponse$inboundSchema = exports.ProductsListRequest$ = exports.ProductsListRequest$outboundSchema = exports.ProductsListRequest$inboundSchema = exports.QueryParamBenefitIDFilter$ = exports.QueryParamBenefitIDFilter$outboundSchema = exports.QueryParamBenefitIDFilter$inboundSchema = exports.ProductsListQueryParamOrganizationIDFilter$ = exports.ProductsListQueryParamOrganizationIDFilter$outboundSchema = exports.ProductsListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.ProductsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.ProductsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductsListQueryParamOrganizationIDFilter$;
(function (ProductsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `ProductsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    ProductsListQueryParamOrganizationIDFilter$.inboundSchema = exports.ProductsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `ProductsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    ProductsListQueryParamOrganizationIDFilter$.outboundSchema = exports.ProductsListQueryParamOrganizationIDFilter$outboundSchema;
})(ProductsListQueryParamOrganizationIDFilter$ || (exports.ProductsListQueryParamOrganizationIDFilter$ = ProductsListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.QueryParamBenefitIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.QueryParamBenefitIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamBenefitIDFilter$;
(function (QueryParamBenefitIDFilter$) {
    /** @deprecated use `QueryParamBenefitIDFilter$inboundSchema` instead. */
    QueryParamBenefitIDFilter$.inboundSchema = exports.QueryParamBenefitIDFilter$inboundSchema;
    /** @deprecated use `QueryParamBenefitIDFilter$outboundSchema` instead. */
    QueryParamBenefitIDFilter$.outboundSchema = exports.QueryParamBenefitIDFilter$outboundSchema;
})(QueryParamBenefitIDFilter$ || (exports.QueryParamBenefitIDFilter$ = QueryParamBenefitIDFilter$ = {}));
/** @internal */
exports.ProductsListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    is_archived: z.nullable(z.boolean()).optional(),
    is_recurring: z.nullable(z.boolean()).optional(),
    benefit_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.ProductSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "is_archived": "isArchived",
        "is_recurring": "isRecurring",
        "benefit_id": "benefitId",
    });
});
/** @internal */
exports.ProductsListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    isArchived: z.nullable(z.boolean()).optional(),
    isRecurring: z.nullable(z.boolean()).optional(),
    benefitId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.ProductSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        isArchived: "is_archived",
        isRecurring: "is_recurring",
        benefitId: "benefit_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductsListRequest$;
(function (ProductsListRequest$) {
    /** @deprecated use `ProductsListRequest$inboundSchema` instead. */
    ProductsListRequest$.inboundSchema = exports.ProductsListRequest$inboundSchema;
    /** @deprecated use `ProductsListRequest$outboundSchema` instead. */
    ProductsListRequest$.outboundSchema = exports.ProductsListRequest$outboundSchema;
})(ProductsListRequest$ || (exports.ProductsListRequest$ = ProductsListRequest$ = {}));
/** @internal */
exports.ProductsListResponse$inboundSchema = z.object({
    Result: components.ListResourceProduct$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.ProductsListResponse$outboundSchema = z.object({
    result: components.ListResourceProduct$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductsListResponse$;
(function (ProductsListResponse$) {
    /** @deprecated use `ProductsListResponse$inboundSchema` instead. */
    ProductsListResponse$.inboundSchema = exports.ProductsListResponse$inboundSchema;
    /** @deprecated use `ProductsListResponse$outboundSchema` instead. */
    ProductsListResponse$.outboundSchema = exports.ProductsListResponse$outboundSchema;
})(ProductsListResponse$ || (exports.ProductsListResponse$ = ProductsListResponse$ = {}));
