// @polar-sh/sdk/models/operations/subscriptionsexport.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionsExportRequest$ = exports.SubscriptionsExportRequest$outboundSchema = exports.SubscriptionsExportRequest$inboundSchema = exports.OrganizationId$ = exports.OrganizationId$outboundSchema = exports.OrganizationId$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.OrganizationId$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrganizationId$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationId$;
(function (OrganizationId$) {
    /** @deprecated use `OrganizationId$inboundSchema` instead. */
    OrganizationId$.inboundSchema = exports.OrganizationId$inboundSchema;
    /** @deprecated use `OrganizationId$outboundSchema` instead. */
    OrganizationId$.outboundSchema = exports.OrganizationId$outboundSchema;
})(OrganizationId$ || (exports.OrganizationId$ = OrganizationId$ = {}));
/** @internal */
exports.SubscriptionsExportRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.SubscriptionsExportRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionsExportRequest$;
(function (SubscriptionsExportRequest$) {
    /** @deprecated use `SubscriptionsExportRequest$inboundSchema` instead. */
    SubscriptionsExportRequest$.inboundSchema = exports.SubscriptionsExportRequest$inboundSchema;
    /** @deprecated use `SubscriptionsExportRequest$outboundSchema` instead. */
    SubscriptionsExportRequest$.outboundSchema = exports.SubscriptionsExportRequest$outboundSchema;
})(SubscriptionsExportRequest$ || (exports.SubscriptionsExportRequest$ = SubscriptionsExportRequest$ = {}));
