// @polar-sh/sdk/models/operations/userslicensekeyslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersLicenseKeysListResponse$ = exports.UsersLicenseKeysListResponse$outboundSchema = exports.UsersLicenseKeysListResponse$inboundSchema = exports.UsersLicenseKeysListRequest$ = exports.UsersLicenseKeysListRequest$outboundSchema = exports.UsersLicenseKeysListRequest$inboundSchema = exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$ = exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$outboundSchema = exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersLicenseKeysListQueryParamOrganizationIDFilter$;
(function (UsersLicenseKeysListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `UsersLicenseKeysListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    UsersLicenseKeysListQueryParamOrganizationIDFilter$.inboundSchema = exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `UsersLicenseKeysListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    UsersLicenseKeysListQueryParamOrganizationIDFilter$.outboundSchema = exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$outboundSchema;
})(UsersLicenseKeysListQueryParamOrganizationIDFilter$ || (exports.UsersLicenseKeysListQueryParamOrganizationIDFilter$ = UsersLicenseKeysListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.UsersLicenseKeysListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    benefit_id: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "benefit_id": "benefitId",
    });
});
/** @internal */
exports.UsersLicenseKeysListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    benefitId: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        benefitId: "benefit_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersLicenseKeysListRequest$;
(function (UsersLicenseKeysListRequest$) {
    /** @deprecated use `UsersLicenseKeysListRequest$inboundSchema` instead. */
    UsersLicenseKeysListRequest$.inboundSchema = exports.UsersLicenseKeysListRequest$inboundSchema;
    /** @deprecated use `UsersLicenseKeysListRequest$outboundSchema` instead. */
    UsersLicenseKeysListRequest$.outboundSchema = exports.UsersLicenseKeysListRequest$outboundSchema;
})(UsersLicenseKeysListRequest$ || (exports.UsersLicenseKeysListRequest$ = UsersLicenseKeysListRequest$ = {}));
/** @internal */
exports.UsersLicenseKeysListResponse$inboundSchema = z.object({
    Result: components.ListResourceLicenseKeyRead$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.UsersLicenseKeysListResponse$outboundSchema = z.object({
    result: components.ListResourceLicenseKeyRead$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersLicenseKeysListResponse$;
(function (UsersLicenseKeysListResponse$) {
    /** @deprecated use `UsersLicenseKeysListResponse$inboundSchema` instead. */
    UsersLicenseKeysListResponse$.inboundSchema = exports.UsersLicenseKeysListResponse$inboundSchema;
    /** @deprecated use `UsersLicenseKeysListResponse$outboundSchema` instead. */
    UsersLicenseKeysListResponse$.outboundSchema = exports.UsersLicenseKeysListResponse$outboundSchema;
})(UsersLicenseKeysListResponse$ || (exports.UsersLicenseKeysListResponse$ = UsersLicenseKeysListResponse$ = {}));
