// @polar-sh/sdk/models/operations/userssubscriptionsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersSubscriptionsUpdateRequest$ = exports.UsersSubscriptionsUpdateRequest$outboundSchema = exports.UsersSubscriptionsUpdateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UsersSubscriptionsUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    UserSubscriptionUpdate: components.UserSubscriptionUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "UserSubscriptionUpdate": "userSubscriptionUpdate",
    });
});
/** @internal */
exports.UsersSubscriptionsUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    userSubscriptionUpdate: components.UserSubscriptionUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userSubscriptionUpdate: "UserSubscriptionUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersSubscriptionsUpdateRequest$;
(function (UsersSubscriptionsUpdateRequest$) {
    /** @deprecated use `UsersSubscriptionsUpdateRequest$inboundSchema` instead. */
    UsersSubscriptionsUpdateRequest$.inboundSchema = exports.UsersSubscriptionsUpdateRequest$inboundSchema;
    /** @deprecated use `UsersSubscriptionsUpdateRequest$outboundSchema` instead. */
    UsersSubscriptionsUpdateRequest$.outboundSchema = exports.UsersSubscriptionsUpdateRequest$outboundSchema;
})(UsersSubscriptionsUpdateRequest$ || (exports.UsersSubscriptionsUpdateRequest$ = UsersSubscriptionsUpdateRequest$ = {}));
