// @polar-sh/sdk/models/operations/usersadvertisementslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersAdvertisementsListResponse$ = exports.UsersAdvertisementsListResponse$outboundSchema = exports.UsersAdvertisementsListResponse$inboundSchema = exports.UsersAdvertisementsListRequest$ = exports.UsersAdvertisementsListRequest$outboundSchema = exports.UsersAdvertisementsListRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UsersAdvertisementsListRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserAdvertisementSortProperty$inboundSchema)).optional(),
});
/** @internal */
exports.UsersAdvertisementsListRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserAdvertisementSortProperty$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersAdvertisementsListRequest$;
(function (UsersAdvertisementsListRequest$) {
    /** @deprecated use `UsersAdvertisementsListRequest$inboundSchema` instead. */
    UsersAdvertisementsListRequest$.inboundSchema = exports.UsersAdvertisementsListRequest$inboundSchema;
    /** @deprecated use `UsersAdvertisementsListRequest$outboundSchema` instead. */
    UsersAdvertisementsListRequest$.outboundSchema = exports.UsersAdvertisementsListRequest$outboundSchema;
})(UsersAdvertisementsListRequest$ || (exports.UsersAdvertisementsListRequest$ = UsersAdvertisementsListRequest$ = {}));
/** @internal */
exports.UsersAdvertisementsListResponse$inboundSchema = z.object({
    Result: components.ListResourceUserAdvertisementCampaign$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.UsersAdvertisementsListResponse$outboundSchema = z.object({
    result: components.ListResourceUserAdvertisementCampaign$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersAdvertisementsListResponse$;
(function (UsersAdvertisementsListResponse$) {
    /** @deprecated use `UsersAdvertisementsListResponse$inboundSchema` instead. */
    UsersAdvertisementsListResponse$.inboundSchema = exports.UsersAdvertisementsListResponse$inboundSchema;
    /** @deprecated use `UsersAdvertisementsListResponse$outboundSchema` instead. */
    UsersAdvertisementsListResponse$.outboundSchema = exports.UsersAdvertisementsListResponse$outboundSchema;
})(UsersAdvertisementsListResponse$ || (exports.UsersAdvertisementsListResponse$ = UsersAdvertisementsListResponse$ = {}));
