// @polar-sh/sdk/models/operations/usersorderslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersOrdersListResponse$ = exports.UsersOrdersListResponse$outboundSchema = exports.UsersOrdersListResponse$inboundSchema = exports.UsersOrdersListRequest$ = exports.UsersOrdersListRequest$outboundSchema = exports.UsersOrdersListRequest$inboundSchema = exports.QueryParamSubscriptionIDFilter$ = exports.QueryParamSubscriptionIDFilter$outboundSchema = exports.QueryParamSubscriptionIDFilter$inboundSchema = exports.ProductPriceTypeFilter$ = exports.ProductPriceTypeFilter$outboundSchema = exports.ProductPriceTypeFilter$inboundSchema = exports.ProductIDFilter$ = exports.ProductIDFilter$outboundSchema = exports.ProductIDFilter$inboundSchema = exports.QueryParamOrganizationIDFilter$ = exports.QueryParamOrganizationIDFilter$outboundSchema = exports.QueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.QueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.QueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamOrganizationIDFilter$;
(function (QueryParamOrganizationIDFilter$) {
    /** @deprecated use `QueryParamOrganizationIDFilter$inboundSchema` instead. */
    QueryParamOrganizationIDFilter$.inboundSchema = exports.QueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `QueryParamOrganizationIDFilter$outboundSchema` instead. */
    QueryParamOrganizationIDFilter$.outboundSchema = exports.QueryParamOrganizationIDFilter$outboundSchema;
})(QueryParamOrganizationIDFilter$ || (exports.QueryParamOrganizationIDFilter$ = QueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.ProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.ProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductIDFilter$;
(function (ProductIDFilter$) {
    /** @deprecated use `ProductIDFilter$inboundSchema` instead. */
    ProductIDFilter$.inboundSchema = exports.ProductIDFilter$inboundSchema;
    /** @deprecated use `ProductIDFilter$outboundSchema` instead. */
    ProductIDFilter$.outboundSchema = exports.ProductIDFilter$outboundSchema;
})(ProductIDFilter$ || (exports.ProductIDFilter$ = ProductIDFilter$ = {}));
/** @internal */
exports.ProductPriceTypeFilter$inboundSchema = z.union([
    components.ProductPriceType$inboundSchema,
    z.array(components.ProductPriceType$inboundSchema),
]);
/** @internal */
exports.ProductPriceTypeFilter$outboundSchema = z.union([
    components.ProductPriceType$outboundSchema,
    z.array(components.ProductPriceType$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductPriceTypeFilter$;
(function (ProductPriceTypeFilter$) {
    /** @deprecated use `ProductPriceTypeFilter$inboundSchema` instead. */
    ProductPriceTypeFilter$.inboundSchema = exports.ProductPriceTypeFilter$inboundSchema;
    /** @deprecated use `ProductPriceTypeFilter$outboundSchema` instead. */
    ProductPriceTypeFilter$.outboundSchema = exports.ProductPriceTypeFilter$outboundSchema;
})(ProductPriceTypeFilter$ || (exports.ProductPriceTypeFilter$ = ProductPriceTypeFilter$ = {}));
/** @internal */
exports.QueryParamSubscriptionIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.QueryParamSubscriptionIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamSubscriptionIDFilter$;
(function (QueryParamSubscriptionIDFilter$) {
    /** @deprecated use `QueryParamSubscriptionIDFilter$inboundSchema` instead. */
    QueryParamSubscriptionIDFilter$.inboundSchema = exports.QueryParamSubscriptionIDFilter$inboundSchema;
    /** @deprecated use `QueryParamSubscriptionIDFilter$outboundSchema` instead. */
    QueryParamSubscriptionIDFilter$.outboundSchema = exports.QueryParamSubscriptionIDFilter$outboundSchema;
})(QueryParamSubscriptionIDFilter$ || (exports.QueryParamSubscriptionIDFilter$ = QueryParamSubscriptionIDFilter$ = {}));
/** @internal */
exports.UsersOrdersListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    product_price_type: z.nullable(z.union([
        components.ProductPriceType$inboundSchema,
        z.array(components.ProductPriceType$inboundSchema),
    ])).optional(),
    subscription_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserOrderSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_price_type": "productPriceType",
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
exports.UsersOrdersListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    productPriceType: z.nullable(z.union([
        components.ProductPriceType$outboundSchema,
        z.array(components.ProductPriceType$outboundSchema),
    ])).optional(),
    subscriptionId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserOrderSortProperty$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
        productPriceType: "product_price_type",
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersOrdersListRequest$;
(function (UsersOrdersListRequest$) {
    /** @deprecated use `UsersOrdersListRequest$inboundSchema` instead. */
    UsersOrdersListRequest$.inboundSchema = exports.UsersOrdersListRequest$inboundSchema;
    /** @deprecated use `UsersOrdersListRequest$outboundSchema` instead. */
    UsersOrdersListRequest$.outboundSchema = exports.UsersOrdersListRequest$outboundSchema;
})(UsersOrdersListRequest$ || (exports.UsersOrdersListRequest$ = UsersOrdersListRequest$ = {}));
/** @internal */
exports.UsersOrdersListResponse$inboundSchema = z.object({
    Result: components.ListResourceUserOrder$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.UsersOrdersListResponse$outboundSchema = z.object({
    result: components.ListResourceUserOrder$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersOrdersListResponse$;
(function (UsersOrdersListResponse$) {
    /** @deprecated use `UsersOrdersListResponse$inboundSchema` instead. */
    UsersOrdersListResponse$.inboundSchema = exports.UsersOrdersListResponse$inboundSchema;
    /** @deprecated use `UsersOrdersListResponse$outboundSchema` instead. */
    UsersOrdersListResponse$.outboundSchema = exports.UsersOrdersListResponse$outboundSchema;
})(UsersOrdersListResponse$ || (exports.UsersOrdersListResponse$ = UsersOrdersListResponse$ = {}));
