// @polar-sh/sdk/models/operations/usersdownloadableslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersDownloadablesListResponse$ = exports.UsersDownloadablesListResponse$outboundSchema = exports.UsersDownloadablesListResponse$inboundSchema = exports.UsersDownloadablesListRequest$ = exports.UsersDownloadablesListRequest$outboundSchema = exports.UsersDownloadablesListRequest$inboundSchema = exports.BenefitIDFilter$ = exports.BenefitIDFilter$outboundSchema = exports.BenefitIDFilter$inboundSchema = exports.UsersDownloadablesListQueryParamOrganizationIDFilter$ = exports.UsersDownloadablesListQueryParamOrganizationIDFilter$outboundSchema = exports.UsersDownloadablesListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UsersDownloadablesListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.UsersDownloadablesListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersDownloadablesListQueryParamOrganizationIDFilter$;
(function (UsersDownloadablesListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `UsersDownloadablesListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    UsersDownloadablesListQueryParamOrganizationIDFilter$.inboundSchema = exports.UsersDownloadablesListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `UsersDownloadablesListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    UsersDownloadablesListQueryParamOrganizationIDFilter$.outboundSchema = exports.UsersDownloadablesListQueryParamOrganizationIDFilter$outboundSchema;
})(UsersDownloadablesListQueryParamOrganizationIDFilter$ || (exports.UsersDownloadablesListQueryParamOrganizationIDFilter$ = UsersDownloadablesListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.BenefitIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.BenefitIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitIDFilter$;
(function (BenefitIDFilter$) {
    /** @deprecated use `BenefitIDFilter$inboundSchema` instead. */
    BenefitIDFilter$.inboundSchema = exports.BenefitIDFilter$inboundSchema;
    /** @deprecated use `BenefitIDFilter$outboundSchema` instead. */
    BenefitIDFilter$.outboundSchema = exports.BenefitIDFilter$outboundSchema;
})(BenefitIDFilter$ || (exports.BenefitIDFilter$ = BenefitIDFilter$ = {}));
/** @internal */
exports.UsersDownloadablesListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    benefit_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "benefit_id": "benefitId",
    });
});
/** @internal */
exports.UsersDownloadablesListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    benefitId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        benefitId: "benefit_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersDownloadablesListRequest$;
(function (UsersDownloadablesListRequest$) {
    /** @deprecated use `UsersDownloadablesListRequest$inboundSchema` instead. */
    UsersDownloadablesListRequest$.inboundSchema = exports.UsersDownloadablesListRequest$inboundSchema;
    /** @deprecated use `UsersDownloadablesListRequest$outboundSchema` instead. */
    UsersDownloadablesListRequest$.outboundSchema = exports.UsersDownloadablesListRequest$outboundSchema;
})(UsersDownloadablesListRequest$ || (exports.UsersDownloadablesListRequest$ = UsersDownloadablesListRequest$ = {}));
/** @internal */
exports.UsersDownloadablesListResponse$inboundSchema = z.object({
    Result: components.ListResourceDownloadableRead$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.UsersDownloadablesListResponse$outboundSchema = z.object({
    result: components.ListResourceDownloadableRead$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersDownloadablesListResponse$;
(function (UsersDownloadablesListResponse$) {
    /** @deprecated use `UsersDownloadablesListResponse$inboundSchema` instead. */
    UsersDownloadablesListResponse$.inboundSchema = exports.UsersDownloadablesListResponse$inboundSchema;
    /** @deprecated use `UsersDownloadablesListResponse$outboundSchema` instead. */
    UsersDownloadablesListResponse$.outboundSchema = exports.UsersDownloadablesListResponse$outboundSchema;
})(UsersDownloadablesListResponse$ || (exports.UsersDownloadablesListResponse$ = UsersDownloadablesListResponse$ = {}));
