// @polar-sh/sdk/models/operations/userssubscriptionslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersSubscriptionsListResponse$ = exports.UsersSubscriptionsListResponse$outboundSchema = exports.UsersSubscriptionsListResponse$inboundSchema = exports.UsersSubscriptionsListRequest$ = exports.UsersSubscriptionsListRequest$outboundSchema = exports.UsersSubscriptionsListRequest$inboundSchema = exports.QueryParamProductIDFilter$ = exports.QueryParamProductIDFilter$outboundSchema = exports.QueryParamProductIDFilter$inboundSchema = exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$ = exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$outboundSchema = exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersSubscriptionsListQueryParamOrganizationIDFilter$;
(function (UsersSubscriptionsListQueryParamOrganizationIDFilter$) {
    /** @deprecated use `UsersSubscriptionsListQueryParamOrganizationIDFilter$inboundSchema` instead. */
    UsersSubscriptionsListQueryParamOrganizationIDFilter$.inboundSchema = exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `UsersSubscriptionsListQueryParamOrganizationIDFilter$outboundSchema` instead. */
    UsersSubscriptionsListQueryParamOrganizationIDFilter$.outboundSchema = exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$outboundSchema;
})(UsersSubscriptionsListQueryParamOrganizationIDFilter$ || (exports.UsersSubscriptionsListQueryParamOrganizationIDFilter$ = UsersSubscriptionsListQueryParamOrganizationIDFilter$ = {}));
/** @internal */
exports.QueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.QueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamProductIDFilter$;
(function (QueryParamProductIDFilter$) {
    /** @deprecated use `QueryParamProductIDFilter$inboundSchema` instead. */
    QueryParamProductIDFilter$.inboundSchema = exports.QueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `QueryParamProductIDFilter$outboundSchema` instead. */
    QueryParamProductIDFilter$.outboundSchema = exports.QueryParamProductIDFilter$outboundSchema;
})(QueryParamProductIDFilter$ || (exports.QueryParamProductIDFilter$ = QueryParamProductIDFilter$ = {}));
/** @internal */
exports.UsersSubscriptionsListRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    active: z.nullable(z.boolean()).optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserSubscriptionSortProperty$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
    });
});
/** @internal */
exports.UsersSubscriptionsListRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    active: z.nullable(z.boolean()).optional(),
    query: z.nullable(z.string()).optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserSubscriptionSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersSubscriptionsListRequest$;
(function (UsersSubscriptionsListRequest$) {
    /** @deprecated use `UsersSubscriptionsListRequest$inboundSchema` instead. */
    UsersSubscriptionsListRequest$.inboundSchema = exports.UsersSubscriptionsListRequest$inboundSchema;
    /** @deprecated use `UsersSubscriptionsListRequest$outboundSchema` instead. */
    UsersSubscriptionsListRequest$.outboundSchema = exports.UsersSubscriptionsListRequest$outboundSchema;
})(UsersSubscriptionsListRequest$ || (exports.UsersSubscriptionsListRequest$ = UsersSubscriptionsListRequest$ = {}));
/** @internal */
exports.UsersSubscriptionsListResponse$inboundSchema = z.object({
    Result: components.ListResourceUserSubscription$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.UsersSubscriptionsListResponse$outboundSchema = z.object({
    result: components.ListResourceUserSubscription$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersSubscriptionsListResponse$;
(function (UsersSubscriptionsListResponse$) {
    /** @deprecated use `UsersSubscriptionsListResponse$inboundSchema` instead. */
    UsersSubscriptionsListResponse$.inboundSchema = exports.UsersSubscriptionsListResponse$inboundSchema;
    /** @deprecated use `UsersSubscriptionsListResponse$outboundSchema` instead. */
    UsersSubscriptionsListResponse$.outboundSchema = exports.UsersSubscriptionsListResponse$outboundSchema;
})(UsersSubscriptionsListResponse$ || (exports.UsersSubscriptionsListResponse$ = UsersSubscriptionsListResponse$ = {}));
